<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interspecies gene mapping • orthogene</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Interspecies gene mapping">
<meta property="og:description" content="
    `orthogene` is an R package for easy mapping of orthologous genes 
    across hundreds of species. It pulls up-to-date gene ortholog mappings 
    across **700+ organisms**. 
    It also provides various utility functions to aggregate/expand 
    common objects (e.g. data.frames, gene expression matrices, lists) 
    using **1:1**, **many:1**, **1:many** or **many:many** gene mappings, 
    both within- and between-species.  ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="articles/infer_species.html">Infer species</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="orthogene-interspecies-gene-mapping">
<code>orthogene</code>: Interspecies gene mapping<a class="anchor" aria-label="anchor" href="#orthogene-interspecies-gene-mapping"></a>
</h1></div>
<img src="https://github.com/neurogenomics/orthogene/raw/main/inst/hex/hex.png" height="600"><br><br><br><br><a href="https://github.com/neurogenomics/orthogene" class="external-link"><img src="https://img.shields.io/badge/devel%20version-1.5.3-black.svg"></a> <a href="https://github.com/neurogenomics/orthogene/actions" class="external-link"><img src="https://github.com/neurogenomics/orthogene/workflows/rworkflows/badge.svg" alt="R build status"></a> <a href="https://github.com/neurogenomics/orthogene/commits/main" class="external-link"><img src="https://img.shields.io/github/last-commit/neurogenomics/orthogene.svg"></a> <a href="https://github.com/neurogenomics/orthogene" class="external-link"><img src="https://img.shields.io/github/languages/code-size/neurogenomics/orthogene.svg"></a> <a href="https://codecov.io/gh/neurogenomics/orthogene" class="external-link"><img src="https://codecov.io/gh/neurogenomics/orthogene/branch/main/graph/badge.svg"></a> <a href="https://cran.r-project.org/web/licenses/GPL-3" class="external-link"><img src="https://img.shields.io/badge/license-GPL--3-blue.svg" alt="License: GPL-3"></a> <a href="https://www.bioconductor.org/packages/orthogene" class="external-link"><img src="https://img.shields.io/badge/release%20version-1.5.2-green.svg"></a> <a href="https://bioconductor.org/packages/stats/bioc/orthogene" class="external-link"><img src="https://img.shields.io/badge/download-271/month-green.svg"></a> <a href="https://bioconductor.org/packages/stats/bioc/orthogene" class="external-link"><img src="https://img.shields.io/badge/download-2809/total-green.svg"></a> <a href="https://bioconductor.org/packages/stats/bioc/orthogene" class="external-link"><img src="http://www.bioconductor.org/shields/downloads/release/orthogene.svg" alt="download"></a> ¶
<h4 id="orthogene-interspecies-gene-mapping-1">
¶ Authors: <i>Brian Schilder</i> ¶
<a class="anchor" aria-label="anchor" href="#orthogene-interspecies-gene-mapping-1"></a>
</h4>
<h4 id="orthogene-interspecies-gene-mapping-2">
¶ README updated: <i>May-09-2023</i> ¶
<a class="anchor" aria-label="anchor" href="#orthogene-interspecies-gene-mapping-2"></a>
</h4>
</div>
<div class="section level1">
<h1 id="intro">Intro<a class="anchor" aria-label="anchor" href="#intro"></a>
</h1>
<p><code>orthogene</code> is an R package for easy mapping of orthologous genes across hundreds of species. It pulls up-to-date gene ortholog mappings across <strong>700+ organisms</strong>. It also provides various utility functions to aggregate/expand common objects (e.g. data.frames, gene expression matrices, lists) using <strong>1:1</strong>, <strong>many:1</strong>, <strong>1:many</strong> or <strong>many:many</strong> gene mappings, both within- and between-species.</p>
<p>In brief, <code>orthogene</code> lets you easily:</p>
<ul>
<li><a href="https://neurogenomics.github.io/orthogene/articles/orthogene#convert-orthologs" class="external-link"><strong><code>convert_orthologs</code></strong> between any two species.</a></li>
<li>
<a href="https://neurogenomics.github.io/orthogene/articles/orthogene#map-species" class="external-link"><strong><code>map_species</code></strong> names onto standard taxonomic ontologies.</a><br>
</li>
<li><a href="https://neurogenomics.github.io/orthogene/articles/orthogene#report-orthologs" class="external-link"><strong><code>report_orthologs</code></strong> between any two species.</a></li>
<li><a href="https://neurogenomics.github.io/orthogene/articles/orthogene#map-genes" class="external-link"><strong><code>map_genes</code></strong> onto standard ontologies</a></li>
<li>
<a href="https://neurogenomics.github.io/orthogene/articles/orthogene#aggregate-mapped-genes" class="external-link"><strong><code>aggregate_mapped_genes</code></strong> in a matrix.</a><br>
</li>
<li><a href="https://neurogenomics.github.io/orthogene/articles/orthogene#get-all-genes" class="external-link">get <strong><code>all_genes</code></strong> from any species.</a></li>
<li>
<a href="https://neurogenomics.github.io/orthogene/articles/infer_species.html" class="external-link"><strong><code>infer_species</code></strong> from gene names.</a><br>
</li>
<li>
<a href="https://neurogenomics.github.io/orthogene/reference/create_background.html" class="external-link"><strong><code>create_background</code></strong> gene lists based one, two, or more species.</a><br>
</li>
<li>
<a href="https://neurogenomics.github.io/orthogene/reference/get_silhouettes.html" class="external-link"><strong><code>get_silhouettes</code></strong> of each species from phylopic.</a><br>
</li>
<li><a href="https://neurogenomics.github.io/orthogene/reference/prepare_tree.html" class="external-link"><strong><code>prepare_tree</code></strong> with evolutionary divergence times across &gt;147,000 species.</a></li>
</ul>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use <code>orthogene</code>, please cite:</p>
<!-- Modify this by editing the file: inst/CITATION  -->
<blockquote>
<p>Brian M. Schilder, Nathan G. Skene (2022). orthogene: Interspecies gene mapping. R package version 1.4.0, <a href="https://doi.org/doi:10.18129/B9.bioc.orthogene" class="external-link uri">https://doi.org/doi:10.18129/B9.bioc.orthogene</a></p>
</blockquote>
</div>
<div class="section level2">
<h2 id="documentation-website">
<a href="https://neurogenomics.github.io/orthogene/" class="external-link">Documentation website</a><a class="anchor" aria-label="anchor" href="#documentation-website"></a>
</h2>
</div>
<div class="section level2">
<h2 id="pdf-manual">
<a href="https://github.com/neurogenomics/orthogene/blob/main/inst/orthogene_1.5.1.pdf" class="external-link">PDF manual</a><a class="anchor" aria-label="anchor" href="#pdf-manual"></a>
</h2>
</div>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="co"># orthogene is only available on Bioconductor&gt;=3.14</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/version.html" class="external-link">version</a></span><span class="op">(</span><span class="op">)</span><span class="op">&lt;</span><span class="st">"3.14"</span><span class="op">)</span> <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span>update <span class="op">=</span> <span class="cn">TRUE</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"orthogene"</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="docker">Docker<a class="anchor" aria-label="anchor" href="#docker"></a>
</h2>
<p><code>orthogene</code> can also be installed via a <a href="https://hub.docker.com/repository/docker/neurogenomicslab/orthogene" class="external-link">Docker</a> or <a href="https://sylabs.io/guides/2.6/user-guide/singularity_and_docker.html" class="external-link">Singularity</a> container with Rstudio pre-installed. Further <a href="https://neurogenomics.github.io/orthogene/articles/docker" class="external-link">instructions provided here</a>.</p>
</div>
</div>
<div class="section level1">
<h1 id="methods">Methods<a class="anchor" aria-label="anchor" href="#methods"></a>
</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/orthogene" class="external-link">orthogene</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"exp_mouse"</span><span class="op">)</span></span>
<span><span class="co"># Setting to "homologene" for the purposes of quick demonstration.</span></span>
<span><span class="co"># We generally recommend using method="gprofiler" (default).</span></span>
<span><span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"homologene"</span>  </span></code></pre></div>
<p>For most functions, <code>orthogene</code> lets users choose between different methods, each with complementary strengths and weaknesses: <code>"gprofiler"</code>, <code>"homologene"</code>, and <code>"babelgene"</code></p>
<p>In general, we recommend you use <code>"gprofiler"</code> when possible, as it tends to be more comprehensive.</p>
<p>While <code>"babelgene"</code> contains less species, it queries a wide variety of orthology databases and can return a column “support_n” that tells you how many databases support each ortholog gene mapping. This can be helpful when you need a semi-quantitative measure of mapping quality.</p>
<p>It’s also worth noting that for smaller gene sets, the speed difference between these methods becomes negligible.</p>
<table class="table">
<colgroup>
<col width="7%">
<col width="10%">
<col width="6%">
<col width="75%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">gprofiler</th>
<th align="left">homologene</th>
<th align="left">babelgene</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Reference organisms</td>
<td align="left">700+</td>
<td align="left">20+</td>
<td align="left">19 (but cannot convert between pairs of non-human species)</td>
</tr>
<tr class="even">
<td align="left">Gene mappings</td>
<td align="left">More comprehensive</td>
<td align="left">Less comprehensive</td>
<td align="left">More comprehensive</td>
</tr>
<tr class="odd">
<td align="left">Updates</td>
<td align="left">Frequent</td>
<td align="left">Less frequent</td>
<td align="left">Less frequent</td>
</tr>
<tr class="even">
<td align="left">Orthology databases</td>
<td align="left">Ensembl, HomoloGene, WormBase</td>
<td align="left">HomoloGene</td>
<td align="left">HGNC Comparison of Orthology Predictions (HCOP), which includes predictions from eggNOG, Ensembl Compara, HGNC, HomoloGene, Inparanoid, NCBI Gene Orthology, OMA, OrthoDB, OrthoMCL, Panther, PhylomeDB, TreeFam and ZFIN</td>
</tr>
<tr class="odd">
<td align="left">Data location</td>
<td align="left">Remote</td>
<td align="left">Local</td>
<td align="left">Local</td>
</tr>
<tr class="even">
<td align="left">Internet connection</td>
<td align="left">Required</td>
<td align="left">Not required</td>
<td align="left">Not required</td>
</tr>
<tr class="odd">
<td align="left">Speed</td>
<td align="left">Slower</td>
<td align="left">Faster</td>
<td align="left">Medium</td>
</tr>
</tbody>
</table>
</div>
<div class="section level1">
<h1 id="quick-example">Quick example<a class="anchor" aria-label="anchor" href="#quick-example"></a>
</h1>
<div class="section level2">
<h2 id="convert-orthologs">Convert orthologs<a class="anchor" aria-label="anchor" href="#convert-orthologs"></a>
</h2>
<p><a href="https://neurogenomics.github.io/orthogene/reference/convert_orthologs.html" class="external-link"><code>convert_orthologs</code></a> is very flexible with what users can supply as <code>gene_df</code>, and can take a <code>data.frame</code>/<code>data.table</code>/<code>tibble</code>, (sparse) <code>matrix</code>, or <code>list</code>/<code>vector</code> containing genes.</p>
<p>Genes, transcripts, proteins, SNPs, or genomic ranges will be recognised in most formats (HGNC, Ensembl, RefSeq, UniProt, etc.) and can even be a mixture of different formats.</p>
<p>All genes will be mapped to gene symbols, unless specified otherwise with the <code>...</code> arguments (see <code><a href="reference/convert_orthologs.html">?orthogene::convert_orthologs</a></code> or <a href="https://neurogenomics.github.io/orthogene/reference/convert_orthologs.html" class="external-link">here</a> for details).</p>
<div class="section level3">
<h3 id="note-on-non-11-orthologs">Note on non-1:1 orthologs<a class="anchor" aria-label="anchor" href="#note-on-non-11-orthologs"></a>
</h3>
<p>A key feature of <a href="https://neurogenomics.github.io/orthogene/reference/convert_orthologs.html" class="external-link"><code>convert_orthologs</code></a> is that it handles the issue of genes with many-to-many mappings across species. This can occur due to evolutionary divergence, and the function of these genes tend to be less conserved and less translatable. Users can address this using different strategies via <code>non121_strategy=</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_df</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="reference/convert_orthologs.html">convert_orthologs</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse</span>,</span>
<span>                                        gene_input <span class="op">=</span> <span class="st">"rownames"</span>, </span>
<span>                                        gene_output <span class="op">=</span> <span class="st">"rownames"</span>, </span>
<span>                                        input_species <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>                                        output_species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>                                        non121_strategy <span class="op">=</span> <span class="st">"drop_both_species"</span>,</span>
<span>                                        method <span class="op">=</span> <span class="va">method</span><span class="op">)</span> </span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Preparing gene_df.</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="do">## sparseMatrix format detected.</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from rownames.</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="do">## 15,259 genes extracted.</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Converting mouse ==&gt; human orthologs using: homologene</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Retrieving all organisms available in homologene.</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: mouse</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for mouse</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: 10090</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Retrieving all organisms available in homologene.</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Mapping species name: human</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Common name mapping found for human</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="do">## 1 organism identified from search: 9606</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking for genes without orthologs in human.</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from input_gene.</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="do">## 13,416 genes extracted.</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a><span class="do">## Extracting genes from ortholog_gene.</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a><span class="do">## 13,416 genes extracted.</span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking for genes without 1:1 orthologs.</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Dropping 46 genes that have multiple input_gene per ortholog_gene (many:1).</span></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a><span class="do">## Dropping 56 genes that have multiple ortholog_gene per input_gene (1:many).</span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a><span class="do">## Filtering gene_df with gene_map</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a><span class="do">## Setting ortholog_gene to rownames.</span></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a><span class="do">## Loading required namespace: DelayedArray</span></span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a><span class="do">## =========== REPORT SUMMARY ===========</span></span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a><span class="do">## Total genes dropped after convert_orthologs :</span></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a><span class="do">##    2,016 / 15,259 (13%)</span></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a><span class="do">## Total genes remaining after convert_orthologs :</span></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a><span class="do">##    13,243 / 15,259 (87%)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_df</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="8%">
<col width="17%">
<col width="15%">
<col width="11%">
<col width="8%">
<col width="14%">
<col width="12%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">astrocytes_ependymal</th>
<th align="right">endothelial-mural</th>
<th align="right">interneurons</th>
<th align="right">microglia</th>
<th align="right">oligodendrocytes</th>
<th align="right">pyramidal CA1</th>
<th align="right">pyramidal SS</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">TSPAN12</td>
<td align="right">0.3303571</td>
<td align="right">0.5872340</td>
<td align="right">0.6413793</td>
<td align="right">0.1428571</td>
<td align="right">0.1207317</td>
<td align="right">0.2864750</td>
<td align="right">0.1453634</td>
</tr>
<tr class="even">
<td align="left">TSHZ1</td>
<td align="right">0.4285714</td>
<td align="right">0.4468085</td>
<td align="right">1.1551724</td>
<td align="right">0.4387755</td>
<td align="right">0.3621951</td>
<td align="right">0.0692226</td>
<td align="right">0.8320802</td>
</tr>
<tr class="odd">
<td align="left">ADAMTS15</td>
<td align="right">0.0089286</td>
<td align="right">0.0978723</td>
<td align="right">0.2206897</td>
<td align="right">0.0000000</td>
<td align="right">0.0231707</td>
<td align="right">0.0117146</td>
<td align="right">0.0375940</td>
</tr>
<tr class="even">
<td align="left">CLDN12</td>
<td align="right">0.2232143</td>
<td align="right">0.1148936</td>
<td align="right">0.5517241</td>
<td align="right">0.0510204</td>
<td align="right">0.2609756</td>
<td align="right">0.4376997</td>
<td align="right">0.6842105</td>
</tr>
<tr class="odd">
<td align="left">RXFP1</td>
<td align="right">0.0000000</td>
<td align="right">0.0127660</td>
<td align="right">0.2551724</td>
<td align="right">0.0000000</td>
<td align="right">0.0158537</td>
<td align="right">0.0511182</td>
<td align="right">0.0751880</td>
</tr>
<tr class="even">
<td align="left">SEMA3C</td>
<td align="right">0.1964286</td>
<td align="right">0.9957447</td>
<td align="right">8.6379310</td>
<td align="right">0.2040816</td>
<td align="right">0.1853659</td>
<td align="right">0.1608094</td>
<td align="right">0.2280702</td>
</tr>
</tbody>
</table>
<p><code>convert_orthologs</code> is just one of the many useful functions in <code>orthogene</code>. Please see the <a href="https://neurogenomics.github.io/orthogene/articles/orthogene" class="external-link">documentation website</a> for the full vignette.</p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="additional-resources">Additional resources<a class="anchor" aria-label="anchor" href="#additional-resources"></a>
</h1>
<div class="section level2">
<h2 id="hex-sticker-creation">
<a href="https://github.com/neurogenomics/orthogene/blob/main/inst/hex/hexSticker.Rmd" class="external-link">Hex sticker creation</a><a class="anchor" aria-label="anchor" href="#hex-sticker-creation"></a>
</h2>
</div>
<div class="section level2">
<h2 id="benchmarking-methods">
<a href="https://github.com/neurogenomics/orthogene/blob/main/inst/benchmark/benchmarks.Rmd" class="external-link">Benchmarking methods</a><a class="anchor" aria-label="anchor" href="#benchmarking-methods"></a>
</h2>
</div>
</div>
<div class="section level1">
<h1 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h1>
<details><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">## R version 4.3.0 (2023-04-21)</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Matrix products: default</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="do">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="do">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="do">## locale:</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="do">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="do">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="do">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="do">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="do">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="do">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="do">## time zone: UTC</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="do">## tzcode source: system (glibc)</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="do">## attached base packages:</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="do">## other attached packages:</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] orthogene_1.5.3 rmarkdown_2.21 </span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="do">## loaded via a namespace (and not attached):</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="do">##  [1] gtable_0.3.3              babelgene_22.9           </span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="do">##  [3] xfun_0.39                 ggplot2_3.4.2            </span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="do">##  [5] htmlwidgets_1.6.2         rstatix_0.7.2            </span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a><span class="do">##  [7] lattice_0.21-8            vctrs_0.6.2              </span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="do">##  [9] tools_4.3.0               generics_0.1.3           </span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a><span class="do">## [11] yulab.utils_0.0.6         stats4_4.3.0             </span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="do">## [13] parallel_4.3.0            tibble_3.2.1             </span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a><span class="do">## [15] fansi_1.0.4               pkgconfig_2.0.3          </span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a><span class="do">## [17] Matrix_1.5-4              data.table_1.14.8        </span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a><span class="do">## [19] homologene_1.4.68.19.3.27 ggplotify_0.1.0          </span></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a><span class="do">## [21] RColorBrewer_1.1-3        S4Vectors_0.39.0         </span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a><span class="do">## [23] desc_1.4.2                lifecycle_1.0.3          </span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a><span class="do">## [25] compiler_4.3.0            dlstats_0.1.6            </span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a><span class="do">## [27] treeio_1.25.0             munsell_0.5.0            </span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a><span class="do">## [29] carData_3.0-5             ggtree_3.9.0             </span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a><span class="do">## [31] ggfun_0.0.9               gprofiler2_0.2.1         </span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a><span class="do">## [33] htmltools_0.5.5           yaml_2.3.7               </span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a><span class="do">## [35] lazyeval_0.2.2            plotly_4.10.1            </span></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a><span class="do">## [37] crayon_1.5.2              pillar_1.9.0             </span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a><span class="do">## [39] car_3.1-2                 ggpubr_0.6.0             </span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a><span class="do">## [41] tidyr_1.3.0               DelayedArray_0.27.1      </span></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a><span class="do">## [43] grr_0.9.5                 abind_1.4-5              </span></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a><span class="do">## [45] nlme_3.1-162              tidyselect_1.2.0         </span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a><span class="do">## [47] aplot_0.1.10              digest_0.6.31            </span></span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a><span class="do">## [49] dplyr_1.1.2               purrr_1.0.1              </span></span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a><span class="do">## [51] rprojroot_2.0.3           fastmap_1.1.1            </span></span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a><span class="do">## [53] grid_4.3.0                here_1.0.1               </span></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a><span class="do">## [55] colorspace_2.1-0          cli_3.6.1                </span></span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a><span class="do">## [57] magrittr_2.0.3            S4Arrays_1.1.1           </span></span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a><span class="do">## [59] patchwork_1.1.2           utf8_1.2.3               </span></span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a><span class="do">## [61] broom_1.0.4               ape_5.7-1                </span></span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a><span class="do">## [63] withr_2.5.0               scales_1.2.1             </span></span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a><span class="do">## [65] backports_1.4.1           httr_1.4.5               </span></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a><span class="do">## [67] matrixStats_0.63.0        rvcheck_0.2.1            </span></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a><span class="do">## [69] ggsignif_0.6.4            evaluate_0.20            </span></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a><span class="do">## [71] knitr_1.42                IRanges_2.35.0           </span></span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a><span class="do">## [73] rworkflows_0.99.5         viridisLite_0.4.1        </span></span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a><span class="do">## [75] gridGraphics_0.5-1        rlang_1.1.1              </span></span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a><span class="do">## [77] Rcpp_1.0.10               glue_1.6.2               </span></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a><span class="do">## [79] tidytree_0.4.2            BiocManager_1.30.20      </span></span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a><span class="do">## [81] BiocGenerics_0.47.0       jsonlite_1.8.4           </span></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a><span class="do">## [83] R6_2.5.1                  badger_0.2.3             </span></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a><span class="do">## [85] MatrixGenerics_1.13.0</span></span></code></pre></div>
</details>
</div>
<div class="section level1">
<h1 id="related-projects">Related projects<a class="anchor" aria-label="anchor" href="#related-projects"></a>
</h1>
<div class="section level2">
<h2 id="tools">Tools<a class="anchor" aria-label="anchor" href="#tools"></a>
</h2>
<ul>
<li><p><a href="https://cran.r-project.org/web/packages/gprofiler2/vignettes/gprofiler2.html" class="external-link"><code>gprofiler2</code></a>: <code>orthogene</code> uses this package. <code><a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gprofiler2::gorth()</a></code> pulls from <a href="https://biit.cs.ut.ee/gprofiler/page/organism-list" class="external-link">many orthology mapping databases</a>.</p></li>
<li><p><a href="https://github.com/oganm/homologene" class="external-link"><code>homologene</code></a>: <code>orthogene</code> uses this package. Provides API access to NCBI <a href="https://www.ncbi.nlm.nih.gov/homologene" class="external-link">HomoloGene</a> database.</p></li>
<li><p><a href="https://cran.r-project.org/web/packages/babelgene/vignettes/babelgene-intro.html" class="external-link"><code>babelgene</code></a>: <code>orthogene</code> uses this package. <code><a href="https://igordot.github.io/babelgene/reference/orthologs.html" class="external-link">babelgene::orthologs()</a></code> pulls from <a href="https://cran.r-project.org/web/packages/babelgene/vignettes/babelgene-intro.html" class="external-link">many orthology mapping databases</a>.</p></li>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/html/annotationTools.html" class="external-link"><code>annotationTools</code></a>: For interspecies microarray data.</p></li>
<li><p><a href="https://www.leibniz-hki.de/en/orthology-r-package.html" class="external-link"><code>orthology</code></a>: R package for ortholog mapping (deprecated?).</p></li>
<li><p><a href="https://rdrr.io/github/elsayed-lab/hpgltools/man/load_biomart_orthologs.html" class="external-link"><code>hpgltools::load_biomart_orthologs()</code></a>: Helper function to get orthologs from biomart.</p></li>
<li><p><a href="https://github.com/ridgelab/JustOrthologs/" class="external-link"><code>JustOrthologs</code></a>: Ortholog inference from multi-species genomic sequences.</p></li>
<li><p><a href="https://github.com/drostlab/orthologr" class="external-link"><code>orthologr</code></a>: Ortholog inference from multi-species genomic sequences.</p></li>
<li><p><a href="https://github.com/davidemms/OrthoFinder" class="external-link"><code>OrthoFinder</code></a>: Gene duplication event inference from multi-species genomics.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="databases">Databases<a class="anchor" aria-label="anchor" href="#databases"></a>
</h2>
<ul>
<li><p><a href="https://www.ncbi.nlm.nih.gov/homologene" class="external-link">HomoloGene</a>: NCBI database that the R package <a href="https://github.com/oganm/homologene" class="external-link">homologene</a> pulls from.</p></li>
<li><p><a href="https://biit.cs.ut.ee/gprofiler" class="external-link">gProfiler</a>: Web server for functional enrichment analysis and conversions of gene lists.</p></li>
<li><p><a href="http://orthologene.org/resources.html" class="external-link">OrtholoGene</a>: Compiled list of gene orthology resources.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h2>
<div class="section level3">
<h3 id="neurogenomics-lab">
<a href="https://www.neurogenomics.co.uk/" class="external-link">Neurogenomics Lab</a><a class="anchor" aria-label="anchor" href="#neurogenomics-lab"></a>
</h3>
<p>UK Dementia Research Institute<br>
Department of Brain Sciences<br>
Faculty of Medicine<br>
Imperial College London<br><a href="https://github.com/neurogenomics" class="external-link">GitHub</a><br><a href="https://hub.docker.com/orgs/neurogenomicslab" class="external-link">DockerHub</a></p>
<p><br></p>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/orthogene" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/neurogenomics/orthogene/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/neurogenomics/orthogene/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing orthogene</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Brian Schilder <br><small class="roles"> Maintainer </small> <a href="https://orcid.org/0000-0001-5949-2191" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
