<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • orthogene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/neurogenomics/orthogene/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.4" id="orthogene-114">orthogene 1.1.4<a class="anchor" aria-label="anchor" href="#orthogene-114"></a></h2>
<p>BUG FIXES</p>
<ul><li>Remove <code>source_all</code> as it included a <code>library</code>
call.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.3" id="orthogene-113">orthogene 1.1.3<a class="anchor" aria-label="anchor" href="#orthogene-113"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Update GHA</li>
</ul><p>BUG FIXES</p>
<ul><li>Fix failing benchmarking tests.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.2" id="orthogene-112">orthogene 1.1.2<a class="anchor" aria-label="anchor" href="#orthogene-112"></a></h2>
<p>BUG FIXES</p>
<ul><li>
<code>convert_orthologs(method="babelgene")</code> now gets gene
mappings from <code>all_genes_babelgene</code> instead
<code><a href="https://igordot.github.io/babelgene/reference/orthologs.html" class="external-link">babelgene::orthologs</a></code> (which doesn’t seem to work very well,
despite being dedicated for this purpose).<br></li>
<li>
<code>map_species</code>:
<ul><li>Avoid running this function redundantly when nested in multiple
layers of other functions.</li>
<li>
<code>common_species_names_dict</code> now return “scientific_name”
by default, instead of “taxonomy_id”</li>
<li>Match <code>map_species</code> method to whatever method is being
used in the function it’s wrapped within, to avoid dropping species due
to naming differences.<br></li>
<li>Add “id” column (e.g. “celegans”) to all org databases to enhance
their searchability.</li>
<li>Add <code>map_species_check_args</code>.</li>
</ul></li>
<li>Ensure proper method-specific <code>output_format</code> when
passing species to other functions.</li>
</ul><p>NEW FEATURES</p>
<ul><li>
<code>plot_orthotree</code>: Automated plotting of phylogenetic
trees with 1:1 ortholog report annotations. Includes several
subfunctions:
<ul><li>
<code>prepare_tree</code> (exported): Read, prune and standardise a
phylogenetic tree.</li>
<li>
<code>gather_images</code> (internal): More robust way to find and
import valid phylopic silhouettes. Will make PR requests to
<code>rphylopic</code> and <code>ggimage</code>/<code>ggtree</code> to
include this functionality.</li>
</ul></li>
<li>Added unit tests for <code>report_orthologs</code>, especially when
<code>method="babelgene"</code>.<br></li>
<li>GitHub Actions:
<ul><li>Merge both GHA workflows into one, as implemented in <a href="https://github.com/neurogenomics/templateR" class="external-link"><code>templateR</code></a>.<br></li>
</ul></li>
<li>Added citation info to README.</li>
<li>Save <code>all_genes_babelgene</code> ortholog data to
orthogene-specific cache instead of tempdir to avoid re-downloading
every R session.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.1" id="orthogene-111">orthogene 1.1.1<a class="anchor" aria-label="anchor" href="#orthogene-111"></a></h2>
<p>BUG FIXES</p>
<ul><li>Made GHA less dependent on hard-coded R/bioc versions.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.0" id="orthogene-110">orthogene 1.1.0<a class="anchor" aria-label="anchor" href="#orthogene-110"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Now on <a href="https://bioconductor.org/packages/devel/bioc/html/orthogene.html" class="external-link">Bioconductor
release 3.14</a>.<br></li>
<li>Docker containers automatically built and pushed to <a href="https://hub.docker.com/repository/docker/bschilder/orthogene" class="external-link">DockerHub</a>
via <a href="https://github.com/neurogenomics/orthogene/blob/main/.github/workflows/dockerhub.yml" class="external-link">GitHub
Actions</a>.<br></li>
<li>
<a href="https://github.com/neurogenomics/orthogene/blob/ad0b5e015805d1f154ec4ef93dd33821e68e580a/Dockerfile" class="external-link">Dockerfile</a>
provided to build and check any R package efficiently with <a href="https://bioconductor.org/packages/release/bioc/html/AnVIL.html" class="external-link">AnVil</a>.<br></li>
<li>CRAN checks and Bioc checks run via <a href="https://github.com/neurogenomics/orthogene/blob/main/.github/workflows/check-bioc-docker.yml" class="external-link">GitHub
Actions</a>.<br></li>
<li>Added documentation on using Docker container to README.<br></li>
<li>Documentation website now automatically built via <a href="https://github.com/neurogenomics/orthogene/blob/main/.github/workflows/check-bioc-docker.yml" class="external-link">GitHub
Actions</a>.<br></li>
<li>Code coverage tests now automatically run and uploaded via <a href="https://github.com/neurogenomics/orthogene/blob/main/.github/workflows/check-bioc-docker.yml" class="external-link">GitHub
Actions</a>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.9" id="orthogene-0999">orthogene 0.99.9<a class="anchor" aria-label="anchor" href="#orthogene-0999"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Replaced R-CMD GHA with bioc-check GHA.</li>
<li>Added new badges.</li>
</ul><p>BUG FIXES</p>
<ul><li>Adjusted vignette yamls to make resulting htmls smaller.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.8" id="orthogene-0998">orthogene 0.99.8<a class="anchor" aria-label="anchor" href="#orthogene-0998"></a></h2>
<p>NEW FEATURES</p>
<ul><li>
<code>orthogene</code> now supports <code>DelayedArray</code>
objects as <code>gene_df</code> input.<br></li>
<li>
<code>create_background</code> now uses <code>all_genes</code> when
all 3 species are the same.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.7" id="orthogene-0997">orthogene 0.99.7<a class="anchor" aria-label="anchor" href="#orthogene-0997"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Added new function <code>create_background</code>.<br></li>
<li>Added new function <code>infer_species</code>.<br></li>
<li>
<code>report_orthologs</code> and <code>convert_orthologs</code> can
now handle cases where <code>input_species</code> is the same as
<code>output_species</code>.</li>
<li>Add internal function <code>get_all_orgs</code> to easily list all
organisms from different packages.<br></li>
<li>Added <code>all_genes</code> method “babelgene”.</li>
</ul><p>BUG FIXES</p>
<ul><li>
<code>report_orthologs</code> no longer throws error due to not
finding <code>tar_genes</code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.6" id="orthogene-0996">orthogene 0.99.6<a class="anchor" aria-label="anchor" href="#orthogene-0996"></a></h2>
<p>BUG FIXES</p>
<ul><li>Allow all messages to be suppressed in
<code>report_orthologs</code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.3" id="orthogene-0993">orthogene 0.99.3<a class="anchor" aria-label="anchor" href="#orthogene-0993"></a></h2>
<p>NEW FEATURES</p>
<ul><li>License switched to GPL-3 (to be compliant with Bioc).<br></li>
<li>New method “babelgene” added to <code>convert_orthologs</code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.2" id="orthogene-0992">orthogene 0.99.2<a class="anchor" aria-label="anchor" href="#orthogene-0992"></a></h2>
<p>NEW FEATURES</p>
<ul><li>License switched to GPL3 (&gt;=3).</li>
</ul><p>BUG FIXES</p>
<ul><li>
<code>GenomeInfoDbData</code> now required.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.0" id="orthogene-010">orthogene 0.1.0<a class="anchor" aria-label="anchor" href="#orthogene-010"></a></h2>
<p>NEW FEATURES</p>
<ul><li>
<code>orthogene</code> released to Bioconductor.</li>
</ul></div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

