<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • orthogene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.15.02</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/neurogenomics/orthogene/blob/main/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="1.15.02" id="orthogene-11502">orthogene 1.15.02<a class="anchor" aria-label="anchor" href="#orthogene-11502"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-15-02">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-15-02"></a></h3>
<ul><li>Fix</li>
</ul></div>
<div class="section level3">
<h3 id="new-features-1-15-02">New features<a class="anchor" aria-label="anchor" href="#new-features-1-15-02"></a></h3>
<ul><li>Disable blank Issues</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.15.01" id="orthogene-11501">orthogene 1.15.01<a class="anchor" aria-label="anchor" href="#orthogene-11501"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-15-01">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-15-01"></a></h3>
<ul><li>Update ggplot2 facet calls to avoid deprecation errors.</li>
<li>Fix all roxygen note warnings.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.14.01" id="orthogene-11401">orthogene 1.14.01<a class="anchor" aria-label="anchor" href="#orthogene-11401"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-14-01">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-14-01"></a></h3>
<ul><li>Update ggplot2 facet calls to avoid deprecation errors.</li>
<li>Fix all roxygen note warnings.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.13.0" id="orthogene-1130">orthogene 1.13.0<a class="anchor" aria-label="anchor" href="#orthogene-1130"></a></h2>
<div class="section level3">
<h3 id="new-features-1-13-0">New features<a class="anchor" aria-label="anchor" href="#new-features-1-13-0"></a></h3>
<ul><li>
<code>check_bool_args</code>: Add check that boolean args are boolean, to avoid issues like <a href="https://github.com/neurogenomics/orthogene/issues/41" class="external-link uri">https://github.com/neurogenomics/orthogene/issues/41</a>
</li>
<li>Warn instead of Error when not all Suggests are installed: <a href="https://github.com/neurogenomics/orthogene/issues/40" class="external-link uri">https://github.com/neurogenomics/orthogene/issues/40</a>
</li>
<li>Bump Docker container in <code>rworkflows</code> to RELEASE_3_20</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-13-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-13-0"></a></h3>
<ul><li>
<code>format_gene_df</code>: Fix report so that the minimum number of dropped genes is 0 (not negative number), and the percentage of retained genes is &lt;=100.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.9.1" id="orthogene-191">orthogene 1.9.1<a class="anchor" aria-label="anchor" href="#orthogene-191"></a></h2>
<div class="section level3">
<h3 id="new-features-1-9-1">New features<a class="anchor" aria-label="anchor" href="#new-features-1-9-1"></a></h3>
<ul><li>Bump version for Bioc 3.19.</li>
<li>Use GHCR instead of Dockerhub, update vignette accordingly.</li>
<li>Increase code coverage.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-9-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-9-1"></a></h3>
<ul><li>Catch case where no genes are found in the species in <code><a href="../reference/map_genes.html">map_genes()</a></code> &amp; test case added.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.3" id="orthogene-173">orthogene 1.7.3<a class="anchor" aria-label="anchor" href="#orthogene-173"></a></h2>
<div class="section level3">
<h3 id="new-features-1-7-3">New features<a class="anchor" aria-label="anchor" href="#new-features-1-7-3"></a></h3>
<ul><li>Added yeast (<em>Saccharomyces cerevisiae</em>) as a valid taxonomy choice.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.2" id="orthogene-172">orthogene 1.7.2<a class="anchor" aria-label="anchor" href="#orthogene-172"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-7-2">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-7-2"></a></h3>
<ul><li>
</li></ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.1" id="orthogene-171">orthogene 1.7.1<a class="anchor" aria-label="anchor" href="#orthogene-171"></a></h2>
<div class="section level3">
<h3 id="new-features-1-7-1">New features<a class="anchor" aria-label="anchor" href="#new-features-1-7-1"></a></h3>
<ul><li>
<code>remove_all_nas</code>
<ul><li>Can handle multiple cols.</li>
</ul></li>
<li>
<code>plot_orthotree</code>
<ul><li>New arg <code>clades_rotate</code>
</li>
<li>New func: <code>rotate_clades</code>
</li>
</ul></li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-7-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-7-1"></a></h3>
<ul><li>
<code>add_columns</code>
<ul><li>Handle both vectors and columns.</li>
</ul></li>
<li>
<code>sort_rows_func</code>
<ul><li>Handle both vectors and columns.</li>
</ul></li>
<li>
<code>filter_gene_df</code>
<ul><li>Avoid coercing single-col dataframe into vector.</li>
<li>Flagged in <a href="https://github.com/neurogenomics/orthogene/issues/34" class="external-link">#34</a></li>
</ul></li>
<li>Fix <code>test-report_orthologs</code>
<ul><li>Recognize either Gene.symbol or input_gene cols depending on when ortholog conversion was done.</li>
</ul></li>
<li>Fix <code>test-convert_orthologs</code>
<ul><li>Line 99 test had wrong number of cols.</li>
</ul></li>
<li>
<code>map_genes_planosphere</code>
<ul><li>Add backup download strategy.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.7.0" id="orthogene-170">orthogene 1.7.0<a class="anchor" aria-label="anchor" href="#orthogene-170"></a></h2>
<div class="section level3">
<h3 id="new-features-1-7-0">New features<a class="anchor" aria-label="anchor" href="#new-features-1-7-0"></a></h3>
<ul><li>Bump version.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.3" id="orthogene-153">orthogene 1.5.3<a class="anchor" aria-label="anchor" href="#orthogene-153"></a></h2>
<div class="section level3">
<h3 id="new-features-1-5-3">New features<a class="anchor" aria-label="anchor" href="#new-features-1-5-3"></a></h3>
<ul><li>
<code>convert_orthologs</code>
<ul><li>Can now take a user-supplied gene_map to use instead of gprofiler/homologene/babelgene.</li>
<li>Lets user take advantage of all of the many:many ortholog mapping functions for organisms not covered in any of the databases (eg. earthworm).</li>
</ul></li>
<li>New helper function: <code>set_gprofiler</code> (to use beta version)</li>
<li>
<code>map_genes</code>
<ul><li>Can now translate planarian gene synonyms using subfunction: <code>map_genes_planosphere</code>.</li>
</ul></li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-5-3">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-5-3"></a></h3>
<ul><li>
<code>convert_orthologs</code>
<ul><li>Now returns original data when <code>input_species</code>==<code>output</code> species (after performing method-specific species mapping) AND <code>standardise_genes==FALSE</code>.</li>
</ul></li>
<li>
<code>add_synonyms</code>
<ul><li>When <code>syn_map</code> is <code>NULL</code>, will use the <code>gene_map</code> instead.</li>
</ul></li>
<li>
<code>map_orthologs_custom</code>
<ul><li>Remove <code>if(input_species==output_species) return(NULL)</code>
</li>
</ul></li>
<li>
<code>drop_non121</code>
<ul><li>Don’t run grouped slicing when “input_gene”,“ortholog_gene” are the only cols.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.2" id="orthogene-152">orthogene 1.5.2<a class="anchor" aria-label="anchor" href="#orthogene-152"></a></h2>
<div class="section level3">
<h3 id="new-features-1-5-2">New features<a class="anchor" aria-label="anchor" href="#new-features-1-5-2"></a></h3>
<ul><li>Remove unnecessary <em>Suggests</em>
</li>
<li>
<code>map_species</code>
<ul><li>
<code>get_all_orgs</code>: When <code>species=NULL</code>, now returns an extra columns called “scientific_name_formatted”.</li>
<li>
<code>format_species_name</code>: New args <code>remove_parentheses</code>, <code>remove_subspecies</code>, <code>remove_subspecies_exceptions</code>
</li>
</ul></li>
<li>
<code>report_orthologs</code>
<ul><li>Make much more efficient by only querying <code>ref_genes</code> once.</li>
<li>Added new internal func <code>report_orthologs_i</code> instead of recursion to make this easier.</li>
<li>Ensure that <code>map</code> and <code>report</code> get rbound separately and returned according to the <code>return_report</code> arg.</li>
</ul></li>
<li>
<code>format_species</code>
<ul><li>Export function that was previously named <code>format_species_name</code>.</li>
</ul></li>
<li>
<code>all_species</code>:
<ul><li>New exported function</li>
<li>Originally implemented a version of this in <code>EWCE::list_species</code>, but decided to extend it and export it here.</li>
</ul></li>
<li>
<code>get_silhouettes</code>
<ul><li>Previously was internal func: <code>gather_images</code>.</li>
<li>Now an exported function.</li>
</ul></li>
<li>
<code>plot_orthotree</code>
<ul><li>Add “Invertebrates” to default <code>clades</code>
</li>
</ul></li>
<li>Update README to showcase more functions.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-5-2">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-5-2"></a></h3>
<ul><li>
<code>drop_non121</code>
<ul><li>New arg <code>symbol_only</code> to <em>ONLY</em> consider gene symbols (<em>not</em> ensemble IDs) when identifying non-121 orthologs.</li>
<li>This make a drastic difference in the number of 1:1 orthologs that get dropped!</li>
</ul></li>
<li>
<code>gather_images</code>
<ul><li>Update to use newly released <code>rphylopic</code> 1.0.0 which uses the new phylopic API.</li>
<li>Add another <code>tryCatch</code> for when the SVG is available but not the png.</li>
</ul></li>
<li>
<code>is_human</code>
<ul><li>Add “9606” and “homo sapiens sapiens” species ID to list of options.</li>
</ul></li>
<li>
<code>all_genes_babelgene</code>
<ul><li>Don’t filter by support when speices is human, because this column will always be <code>NA</code> since it’s irrelevant for humans.</li>
</ul></li>
<li>Fix unit tests:
<ul><li><em>report_orthologs</em></li>
</ul></li>
<li>
<code>dMcast</code>
<ul><li>Fix <code>stats::pass</code> –&gt; <code><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">stats::na.pass</a></code>. Weirdly, only a problem on Linux. Did base R change a fundamental function name?</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.1" id="orthogene-151">orthogene 1.5.1<a class="anchor" aria-label="anchor" href="#orthogene-151"></a></h2>
<div class="section level3">
<h3 id="new-features-1-5-1">New features<a class="anchor" aria-label="anchor" href="#new-features-1-5-1"></a></h3>
<ul><li>Bumped to 1.5.1 for Bioc devel 3.17</li>
<li>Merged upstream devel.</li>
<li>Now using <code>rworkflows</code> for GHA.
<ul><li>Removed <em>Dockerfile</em>
</li>
</ul></li>
<li>Host <code>orthogene</code> data resources on Zenodo:
<ul><li><a href="https://doi.org/10.5281/zenodo.7315418" class="external-link uri">https://doi.org/10.5281/zenodo.7315418</a></li>
</ul></li>
<li>Upgrade TimeTree phylogeny to v5 (2022):
<ul><li>50k+ species –&gt; 137k+ species!</li>
</ul></li>
<li>Replace <code>dplyr::%&gt;%</code> usage with <code>|&gt;</code>
</li>
<li>Add <em>CITATION</em> file</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-5-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-5-1"></a></h3>
<ul><li>
<code>prepare_tree</code>:
<ul><li>Ignore species name case and trim “’” when filtering tree.</li>
</ul></li>
<li>
<code>map_species</code>:
<ul><li>Add <code>trimws</code> step to remove flanking ” ” or “’”.</li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.4" id="orthogene-134">orthogene 1.3.4<a class="anchor" aria-label="anchor" href="#orthogene-134"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-3-4">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-3-4"></a></h3>
<ul><li>Remove <code>Matrix.utils</code> since it’s now deprecated.
<ul><li>Reimplement the <code>dMcast</code> function as a new internal function within <code>orthogene</code>, since that’s the only function I use from <code>Matrix.utils</code>.<br></li>
</ul></li>
<li>Fix GHA workflow now that <em><a href="mailto:r-lib/actions@master" class="email">r-lib/actions@master</a></em> has been removed.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.3" id="orthogene-133">orthogene 1.3.3<a class="anchor" aria-label="anchor" href="#orthogene-133"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-3-3">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-3-3"></a></h3>
<ul><li>Make <code>test-map_orthologs_babelgene</code> less stringent with the number of genes expected.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.2" id="orthogene-132">orthogene 1.3.2<a class="anchor" aria-label="anchor" href="#orthogene-132"></a></h2>
<div class="section level3">
<h3 id="new-features-1-3-2">New features<a class="anchor" aria-label="anchor" href="#new-features-1-3-2"></a></h3>
<ul><li>Add <em>inst/grofiler_namespace.csv.gz</em> for documentation purposes.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-3-2">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-3-2"></a></h3>
<ul><li>
<code>create_background</code>:
<ul><li>Ensure user-supplied <code>bg</code> gets used: <a href="https://github.com/neurogenomics/orthogene/issues/22" class="external-link uri">https://github.com/neurogenomics/orthogene/issues/22</a>
</li>
</ul></li>
<li>Properly document internal data so that <code><a href="https://devtools.r-lib.org/reference/document.html" class="external-link">devtools::document</a></code> doesn’t expect them to be exported objects.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.1" id="orthogene-131">orthogene 1.3.1<a class="anchor" aria-label="anchor" href="#orthogene-131"></a></h2>
<div class="section level3">
<h3 id="new-features-1-3-1">New features<a class="anchor" aria-label="anchor" href="#new-features-1-3-1"></a></h3>
<ul><li>
<code>plot_orthotree</code>: Pass up <code>tree_source</code> arg.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.0" id="orthogene-130">orthogene 1.3.0<a class="anchor" aria-label="anchor" href="#orthogene-130"></a></h2>
<div class="section level3">
<h3 id="new-features-1-3-0">New features<a class="anchor" aria-label="anchor" href="#new-features-1-3-0"></a></h3>
<ul><li>
<code>aggregate_mapped_genes</code>:
<ul><li>Pass up additional args from <code>map_genes</code>.</li>
<li>Add <code>map_orthologs</code> as a way to create <code>gene_map</code> automatically, when <code>gene_map=NULL</code> and <code>input_species!=output_species</code>.<br></li>
<li>Split <code>species</code> into <code>input_species</code> and <code>output_species</code> args.</li>
<li>Change <code>method</code> –&gt; <code>agg_method</code>, and use <code>method</code> to pass to <code>map_orthologs</code> instead.</li>
<li>Pass up additional args from <code>map_orthologs</code>.</li>
<li>Add link to detailed explanation of matrix aggregation/expansion in <code>many2many_rows</code> docs.</li>
<li>Automatically pick best method for many:1 or many:many mapping.</li>
<li>
<code>as_integers</code>: new arg that uses <code>floor</code>.</li>
<li>Rename <code>FUN</code> to <code>agg_fun</code>.</li>
</ul></li>
<li>Add new <em>data</em> <code>gprofiler_namespace</code>. Used to validate <code>target=</code> arg in <code>gprofiler2</code> functions.</li>
<li>Upgraded <code>aggregate_mapped_genes</code>:
<ul><li>Can now used <code>gene_map</code> made by <code>map_orthologs</code> or <code>map_genes</code>.</li>
<li>Can now handle many:many relationships.</li>
<li>Will automatically pick the best method to perform aggregation and/or expansion.</li>
</ul></li>
<li>Removed internal function <code>aggregate_mapped_genes_twice</code>
</li>
<li>Extracted aggregation args from <code>non121_strategy</code> and placed them in their own new own (<code>agg_fun</code>) since these options are no longer mutually exclusive due to many:many expansion/aggregation.</li>
<li>Pass up <code>as_DelayedArray</code>
</li>
<li>Bump to v1.3.0 and R &gt;=4.2 now that we’re developing on Bioc 3.16.</li>
<li>Add <em>ISSUE_TEMPLATE</em>.</li>
<li>
<code>prepare_tree</code>:
<ul><li>Add <code>tree_source</code> options: path / URL / OmaDB / UCSC / timetree</li>
</ul></li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-3-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-3-0"></a></h3>
<ul><li>
<code>map_genes</code>: Fix report at completion.<br></li>
<li>Add safeguards against using aggregation when <code>gene_df</code> isn’t a matrix.</li>
<li>Removed <code>DelayedMatrixStats</code> <em>Import</em> (no longer needed).<br></li>
<li>Fix all unit tests and examples after making all updates.</li>
<li>Recognize sparse/dense matrix or delayedarray in <code>check_agg_args</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.2.1" id="orthogene-121">orthogene 1.2.1<a class="anchor" aria-label="anchor" href="#orthogene-121"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-2-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-2-1"></a></h3>
<ul><li>Make <code>test-map_orthologs_babelgene</code> less stringent with the number of expected genes.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.2.0" id="orthogene-120">orthogene 1.2.0<a class="anchor" aria-label="anchor" href="#orthogene-120"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-2-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-2-0"></a></h3>
<ul><li>Increment to v2.0 now that Bioc 3.15 is the RELEASE.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.5" id="orthogene-115">orthogene 1.1.5<a class="anchor" aria-label="anchor" href="#orthogene-115"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-1-5">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-1-5"></a></h3>
<ul><li>
<code>map_orthologs_babelgene</code>
<ul><li>Add “Bad credentials” check for <code>piggyback</code>.</li>
<li>Add <code>use_old</code> as an optional arg so I can switch to more recent versions of <code>babelgene::orthologs_df</code> if need be.<br></li>
<li>Use updated built-in <code>babelgene::orthologs_df</code> by default.</li>
<li>Throw error if trying to map between two non-human species.</li>
<li>Filter support==NA mappings by default, not but support&gt;=2 like <code>babelgene</code> does by default (even when <code>babelgene::orthologs(min_support = 1)</code>).</li>
<li>See here for discussion of discrepancies with babelgene maintainer: <a href="https://github.com/igordot/babelgene/issues/2" class="external-link uri">https://github.com/igordot/babelgene/issues/2</a>
</li>
</ul></li>
</ul></div>
<div class="section level3">
<h3 id="new-features-1-1-5">New features<a class="anchor" aria-label="anchor" href="#new-features-1-1-5"></a></h3>
<ul><li>Removed <code>aggregate_rows_delayedarray</code> as it wasn’t being used and was far less efficient than the other methods anyway (which are also compatible with DelayedArray matrices anyway). * New unit tests:
<ul><li>
<code>load_data</code><br></li>
<li>
<code>aggregate_mapped_genes(method='stat')</code><br></li>
<li>
<code>sparsity</code><br></li>
<li><code>infer_species</code></li>
</ul></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.4" id="orthogene-114">orthogene 1.1.4<a class="anchor" aria-label="anchor" href="#orthogene-114"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-1-4">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-1-4"></a></h3>
<ul><li>Remove <code>source_all</code> as it included a <code>library</code> call.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.3" id="orthogene-113">orthogene 1.1.3<a class="anchor" aria-label="anchor" href="#orthogene-113"></a></h2>
<div class="section level3">
<h3 id="new-features-1-1-3">New features<a class="anchor" aria-label="anchor" href="#new-features-1-1-3"></a></h3>
<ul><li>Update GHA</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-1-1-3">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-1-3"></a></h3>
<ul><li>Fix failing benchmarking tests.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.2" id="orthogene-112">orthogene 1.1.2<a class="anchor" aria-label="anchor" href="#orthogene-112"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-1-2">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-1-2"></a></h3>
<ul><li>
<code>convert_orthologs(method="babelgene")</code> now gets gene mappings from <code>all_genes_babelgene</code> instead <code><a href="https://igordot.github.io/babelgene/reference/orthologs.html" class="external-link">babelgene::orthologs</a></code> (which doesn’t seem to work very well, despite being dedicated for this purpose).<br></li>
<li>
<code>map_species</code>:
<ul><li>Avoid running this function redundantly when nested in multiple layers of other functions.</li>
<li>
<code>common_species_names_dict</code> now return “scientific_name” by default, instead of “taxonomy_id”</li>
<li>Match <code>map_species</code> method to whatever method is being used in the function it’s wrapped within, to avoid dropping species due to naming differences.<br></li>
<li>Add “id” column (e.g. “celegans”) to all org databases to enhance their searchability.</li>
<li>Add <code>map_species_check_args</code>.</li>
</ul></li>
<li>Ensure proper method-specific <code>output_format</code> when passing species to other functions.</li>
</ul></div>
<div class="section level3">
<h3 id="new-features-1-1-2">New features<a class="anchor" aria-label="anchor" href="#new-features-1-1-2"></a></h3>
<ul><li>
<code>plot_orthotree</code>: Automated plotting of phylogenetic trees with 1:1 ortholog report annotations. Includes several subfunctions:
<ul><li>
<code>prepare_tree</code> (exported): Read, prune and standardise a phylogenetic tree.</li>
<li>
<code>gather_images</code> (internal): More robust way to find and import valid phylopic silhouettes. Will make PR requests to <code>rphylopic</code> and <code>ggimage</code>/<code>ggtree</code> to include this functionality.</li>
</ul></li>
<li>Added unit tests for <code>report_orthologs</code>, especially when <code>method="babelgene"</code>.<br></li>
<li>GitHub Actions:
<ul><li>Merge both GHA workflows into one, as implemented in <a href="https://github.com/neurogenomics/templateR" class="external-link"><code>templateR</code></a>.<br></li>
</ul></li>
<li>Added citation info to README.</li>
<li>Save <code>all_genes_babelgene</code> ortholog data to orthogene-specific cache instead of tempdir to avoid re-downloading every R session.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.1" id="orthogene-111">orthogene 1.1.1<a class="anchor" aria-label="anchor" href="#orthogene-111"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-1-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-1-1"></a></h3>
<ul><li>Made GHA less dependent on hard-coded R/bioc versions.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.0" id="orthogene-110">orthogene 1.1.0<a class="anchor" aria-label="anchor" href="#orthogene-110"></a></h2>
<div class="section level3">
<h3 id="new-features-1-1-0">New features<a class="anchor" aria-label="anchor" href="#new-features-1-1-0"></a></h3>
<ul><li>Now on <a href="https://bioconductor.org/packages/devel/bioc/html/orthogene.html" class="external-link">Bioconductor release 3.14</a>.<br></li>
<li>Docker containers automatically built and pushed to <a href="https://hub.docker.com/repository/docker/bschilder/orthogene" class="external-link">DockerHub</a> via <a href="https://github.com/neurogenomics/orthogene/blob/main/.github/workflows/dockerhub.yml" class="external-link">GitHub Actions</a>.<br></li>
<li>
<a href="https://github.com/neurogenomics/orthogene/blob/ad0b5e015805d1f154ec4ef93dd33821e68e580a/Dockerfile" class="external-link">Dockerfile</a> provided to build and check any R package efficiently with <a href="https://bioconductor.org/packages/release/bioc/html/AnVIL.html" class="external-link">AnVil</a>.<br></li>
<li>CRAN checks and Bioc checks run via <a href="https://github.com/neurogenomics/orthogene/blob/main/.github/workflows/check-bioc-docker.yml" class="external-link">GitHub Actions</a>.<br></li>
<li>Added documentation on using Docker container to README.<br></li>
<li>Documentation website now automatically built via <a href="https://github.com/neurogenomics/orthogene/blob/main/.github/workflows/check-bioc-docker.yml" class="external-link">GitHub Actions</a>.<br></li>
<li>Code coverage tests now automatically run and uploaded via <a href="https://github.com/neurogenomics/orthogene/blob/main/.github/workflows/check-bioc-docker.yml" class="external-link">GitHub Actions</a>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.9" id="orthogene-0999">orthogene 0.99.9<a class="anchor" aria-label="anchor" href="#orthogene-0999"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-9">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-9"></a></h3>
<ul><li>Replaced R-CMD GHA with bioc-check GHA.</li>
<li>Added new badges.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-99-9">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-9"></a></h3>
<ul><li>Adjusted vignette yamls to make resulting htmls smaller.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.8" id="orthogene-0998">orthogene 0.99.8<a class="anchor" aria-label="anchor" href="#orthogene-0998"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-8">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-8"></a></h3>
<ul><li>
<code>orthogene</code> now supports <code>DelayedArray</code> objects as <code>gene_df</code> input.<br></li>
<li>
<code>create_background</code> now uses <code>all_genes</code> when all 3 species are the same.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.7" id="orthogene-0997">orthogene 0.99.7<a class="anchor" aria-label="anchor" href="#orthogene-0997"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-7">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-7"></a></h3>
<ul><li>Added new function <code>create_background</code>.<br></li>
<li>Added new function <code>infer_species</code>.<br></li>
<li>
<code>report_orthologs</code> and <code>convert_orthologs</code> can now handle cases where <code>input_species</code> is the same as <code>output_species</code>.</li>
<li>Add internal function <code>get_all_orgs</code> to easily list all organisms from different packages.<br></li>
<li>Added <code>all_genes</code> method “babelgene”.</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-99-7">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-7"></a></h3>
<ul><li>
<code>report_orthologs</code> no longer throws error due to not finding <code>tar_genes</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.6" id="orthogene-0996">orthogene 0.99.6<a class="anchor" aria-label="anchor" href="#orthogene-0996"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-0-99-6">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-6"></a></h3>
<ul><li>Allow all messages to be suppressed in <code>report_orthologs</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.3" id="orthogene-0993">orthogene 0.99.3<a class="anchor" aria-label="anchor" href="#orthogene-0993"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-3">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-3"></a></h3>
<ul><li>License switched to GPL-3 (to be compliant with Bioc).<br></li>
<li>New method “babelgene” added to <code>convert_orthologs</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.2" id="orthogene-0992">orthogene 0.99.2<a class="anchor" aria-label="anchor" href="#orthogene-0992"></a></h2>
<div class="section level3">
<h3 id="new-features-0-99-2">New features<a class="anchor" aria-label="anchor" href="#new-features-0-99-2"></a></h3>
<ul><li>License switched to GPL3 (&gt;=3).</li>
</ul></div>
<div class="section level3">
<h3 id="bug-fixes-0-99-2">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-2"></a></h3>
<ul><li>
<code>GenomeInfoDbData</code> now required.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.0" id="orthogene-010">orthogene 0.1.0<a class="anchor" aria-label="anchor" href="#orthogene-010"></a></h2>
<div class="section level3">
<h3 id="new-features-0-1-0">New features<a class="anchor" aria-label="anchor" href="#new-features-0-1-0"></a></h3>
<ul><li>
<code>orthogene</code> released to Bioconductor.</li>
</ul></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

