<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Aggregate/expand a gene matrix by gene mappings — aggregate_mapped_genes • orthogene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Aggregate/expand a gene matrix by gene mappings — aggregate_mapped_genes"><meta property="og:description" content="Aggregate/expand a gene matrix (gene_df) using a gene mapping
data.frame (gene_map).
Importantly, mappings can be performed across a variety of scenarios that
can occur during within-species and between-species gene mapping:
1 gene : 1 gene
many genes : 1 gene
1 gene : many genes
many genes : many genes
For more details on how aggregation/expansion is performed,
please see: many2many_rows."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.01</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Aggregate/expand a gene matrix by gene mappings</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/main/R/aggregate_mapped_genes.R" class="external-link"><code>R/aggregate_mapped_genes.R</code></a></small>
    <div class="hidden name"><code>aggregate_mapped_genes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Aggregate/expand a gene matrix (<code>gene_df</code>) using a gene mapping
<a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a> (<code>gene_map</code>).
Importantly, mappings can be performed across a variety of scenarios that
can occur during within-species and between-species gene mapping:</p><ul><li><p>1 gene : 1 gene</p></li>
<li><p>many genes : 1 gene</p></li>
<li><p>1 gene : many genes</p></li>
<li><p>many genes : many genes</p></li>
</ul><p>For more details on how aggregation/expansion is performed,
please see: <a href="many2many_rows.html">many2many_rows</a>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">aggregate_mapped_genes</span><span class="op">(</span></span>
<span>  <span class="va">gene_df</span>,</span>
<span>  gene_map <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  input_col <span class="op">=</span> <span class="st">"input_gene"</span>,</span>
<span>  output_col <span class="op">=</span> <span class="st">"ortholog_gene"</span>,</span>
<span>  input_species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  output_species <span class="op">=</span> <span class="va">input_species</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gprofiler"</span>, <span class="st">"homologene"</span>, <span class="st">"babelgene"</span><span class="op">)</span>,</span>
<span>  agg_fun <span class="op">=</span> <span class="st">"sum"</span>,</span>
<span>  agg_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"monocle3"</span>, <span class="st">"stats"</span><span class="op">)</span>,</span>
<span>  aggregate_orthologs <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  transpose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  mthreshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  target <span class="op">=</span> <span class="st">"ENSG"</span>,</span>
<span>  numeric_ns <span class="op">=</span> <span class="st">""</span>,</span>
<span>  as_integers <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  as_sparse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  as_DelayedArray <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  dropNA <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sort_rows <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-gene-df">gene_df<a class="anchor" aria-label="anchor" href="#arg-gene-df"></a></dt>
<dd><p>Input matrix where row names are genes.</p></dd>


<dt id="arg-gene-map">gene_map<a class="anchor" aria-label="anchor" href="#arg-gene-map"></a></dt>
<dd><p>A <a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a> that maps the current gene names
to new gene names.
This function's behaviour will adapt to different situations as follows:</p><ul><li><p><code>gene_map=&lt;data.frame&gt;</code> :<br> When a data.frame containing the
gene key:value columns
(specified by <code>input_col</code> and <code>output_col</code>, respectively)
is provided, this will be used to perform aggregation/expansion.</p></li>
<li><p><code>gene_map=NULL</code> and <code>input_species!=output_species</code> :<br>
A <code>gene_map</code> is automatically generated by
 <a href="map_orthologs.html">map_orthologs</a> to perform inter-species
 gene aggregation/expansion.</p></li>
<li><p><code>gene_map=NULL</code> and <code>input_species==output_species</code> :<br>
A <code>gene_map</code> is automatically generated by
 <a href="map_genes.html">map_genes</a> to perform within-species
 gene gene symbol standardization and aggregation/expansion.</p></li>
</ul></dd>


<dt id="arg-input-col">input_col<a class="anchor" aria-label="anchor" href="#arg-input-col"></a></dt>
<dd><p>Column name within <code>gene_map</code> with gene names matching
the row names of <code>X</code>.</p></dd>


<dt id="arg-output-col">output_col<a class="anchor" aria-label="anchor" href="#arg-output-col"></a></dt>
<dd><p>Column name within <code>gene_map</code> with gene names
that you wish you map the row names of <code>X</code> onto.</p></dd>


<dt id="arg-input-species">input_species<a class="anchor" aria-label="anchor" href="#arg-input-species"></a></dt>
<dd><p>Name of the input species (e.g., "mouse","fly").
Use <a href="map_species.html">map_species</a> to return a full list
of available species.</p></dd>


<dt id="arg-output-species">output_species<a class="anchor" aria-label="anchor" href="#arg-output-species"></a></dt>
<dd><p>Name of the output species (e.g. "human","chicken").
Use <a href="map_species.html">map_species</a> to return a full list
 of available species.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>R package to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>


<dt id="arg-agg-fun">agg_fun<a class="anchor" aria-label="anchor" href="#arg-agg-fun"></a></dt>
<dd><p>Aggregation function.</p></dd>


<dt id="arg-agg-method">agg_method<a class="anchor" aria-label="anchor" href="#arg-agg-method"></a></dt>
<dd><p>Aggregation method.</p></dd>


<dt id="arg-aggregate-orthologs">aggregate_orthologs<a class="anchor" aria-label="anchor" href="#arg-aggregate-orthologs"></a></dt>
<dd><p>[Optional] After performing an initial round of
many:many aggregation/expansion with <a href="many2many_rows.html">many2many_rows</a>,
 ensure each orthologous gene only appears in one row by using the
<a href="aggregate_rows.html">aggregate_rows</a> function (default: <code>TRUE</code>).</p></dd>


<dt id="arg-transpose">transpose<a class="anchor" aria-label="anchor" href="#arg-transpose"></a></dt>
<dd><p>Transpose <code>gene_df</code> before mapping genes.</p></dd>


<dt id="arg-mthreshold">mthreshold<a class="anchor" aria-label="anchor" href="#arg-mthreshold"></a></dt>
<dd><p>maximum number of results per initial alias to show. Shows all by default.</p></dd>


<dt id="arg-target">target<a class="anchor" aria-label="anchor" href="#arg-target"></a></dt>
<dd><p>target namespace.</p></dd>


<dt id="arg-numeric-ns">numeric_ns<a class="anchor" aria-label="anchor" href="#arg-numeric-ns"></a></dt>
<dd><p>namespace to use for fully numeric IDs (<a href="https://biit.cs.ut.ee/gprofiler/page/namespaces-list" class="external-link">list of available namespaces</a>).</p></dd>


<dt id="arg-as-integers">as_integers<a class="anchor" aria-label="anchor" href="#arg-as-integers"></a></dt>
<dd><p>Force all values in the matrix to become integers,
by applying <a href="https://rdrr.io/r/base/Round.html" class="external-link">floor</a> (default: <code>FALSE</code>).</p></dd>


<dt id="arg-as-sparse">as_sparse<a class="anchor" aria-label="anchor" href="#arg-as-sparse"></a></dt>
<dd><p>Convert aggregated matrix to sparse matrix.</p></dd>


<dt id="arg-as-delayedarray">as_DelayedArray<a class="anchor" aria-label="anchor" href="#arg-as-delayedarray"></a></dt>
<dd><p>Convert aggregated matrix to
<a href="https://rdrr.io/pkg/DelayedArray/man/DelayedArray-class.html" class="external-link">DelayedArray</a>.</p></dd>


<dt id="arg-dropna">dropNA<a class="anchor" aria-label="anchor" href="#arg-dropna"></a></dt>
<dd><p>Drop genes assigned to <code>NA</code> in <code>groupings</code>.</p></dd>


<dt id="arg-sort-rows">sort_rows<a class="anchor" aria-label="anchor" href="#arg-sort-rows"></a></dt>
<dd><p>Sort <code>gene_df</code> rows alphanumerically.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Aggregated matrix</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#### Aggregate within species: gene synonyms ####</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"exp_mouse_enst"</span><span class="op">)</span>                                </span></span>
<span class="r-in"><span><span class="va">X_agg</span> <span class="op">&lt;-</span> <span class="fu">aggregate_mapped_genes</span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse_enst</span>, </span></span>
<span class="r-in"><span>                                input_species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span>  </span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in gprofiler.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using stored `gprofiler_orgs`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: mmusculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 473 / 482 (98.13%) genes mapped.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Aggregating rows using: monocle3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting obj to sparseMatrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Matrix aggregated:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - Input: 482 x 7 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - Output: 92 x 7</span>
<span class="r-in"><span>                                 </span></span>
<span class="r-in"><span><span class="co">#### Aggregate across species: gene orthologs ####               </span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"exp_mouse"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">X_agg2</span> <span class="op">&lt;-</span> <span class="fu">aggregate_mapped_genes</span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse</span>, </span></span>
<span class="r-in"><span>                                 input_species <span class="op">=</span> <span class="st">"mouse"</span>,</span></span>
<span class="r-in"><span>                                 output_species <span class="op">=</span> <span class="st">"human"</span>,</span></span>
<span class="r-in"><span>                                 method<span class="op">=</span><span class="st">"homologene"</span><span class="op">)</span>                                                     </span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping many:many rows.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1110012L19Rik : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2610034B18Rik : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> AA415398 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Ankhd1 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Anxa8 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Apitd1 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Arhgap8 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Asb3 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> C4a : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> C4b : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cbs : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Ccz1 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Ckmt1 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Coro7 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Cryaa : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> D10Jhu81e : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> F8a : converting 1 row(s) --&gt; 3 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Fam21 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Fcgr4 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gpr89 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gstt2 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> H3f3a : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> H3f3b : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Hspa1a : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Icosl : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Klhl23 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mrpl23 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Nbl1 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Nomo1 : converting 1 row(s) --&gt; 3 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Pmf1 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Pom121 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Pramef8 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Prodh : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Ranbp2 : converting 1 row(s) --&gt; 7 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Serf1 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sgk3 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Slx1b : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Smn1 : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Spin2d : converting 1 row(s) --&gt; 2 row(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Aggregating rows using: monocle3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting obj to sparseMatrix.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Matrix aggregated:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - Input: 15,259 x 7 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - Output: 13,316 x 7</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

