<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get all genes — all_genes • orthogene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Get all genes — all_genes"><meta property="og:description" content="Return all known genes from a given species."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.17.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get all genes</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/main/R/all_genes.R" class="external-link"><code>R/all_genes.R</code></a></small>
    <div class="hidden name"><code>all_genes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Return all known genes from a given species.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">all_genes</span><span class="op">(</span></span>
<span>  <span class="va">species</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gprofiler"</span>, <span class="st">"homologene"</span>, <span class="st">"babelgene"</span><span class="op">)</span>,</span>
<span>  ensure_filter_nas <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  run_map_species <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  force <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-species">species<a class="anchor" aria-label="anchor" href="#arg-species"></a></dt>
<dd><p>Species to get all genes for.
Will first be standardised with <code>map_species</code>.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>R package to use for gene mapping:</p><dl><dt><code>"gprofiler"</code></dt>
<dd><p>Slower but more species and genes.</p></dd>

 <dt><code>"homologene"</code></dt>
<dd><p>Faster but fewer species and genes.</p></dd>

 <dt><code>"babelgene"</code></dt>
<dd><p>Faster but fewer species and genes.
 Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></dd>


</dl></dd>


<dt id="arg-ensure-filter-nas">ensure_filter_nas<a class="anchor" aria-label="anchor" href="#arg-ensure-filter-nas"></a></dt>
<dd><p>Perform an extra check to remove
genes that are <code>NA</code>s of any kind.</p></dd>


<dt id="arg-run-map-species">run_map_species<a class="anchor" aria-label="anchor" href="#arg-run-map-species"></a></dt>
<dd><p>Standardise <code>species</code> names with
<a href="map_species.html">map_species</a> first (Default: <code>TRUE</code>).</p></dd>


<dt id="arg-force">force<a class="anchor" aria-label="anchor" href="#arg-force"></a></dt>
<dd><p>Force a new files to be generated even if a cached one exists.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Print messages.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments to be passed to
<a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gorth</a> or <a href="https://rdrr.io/pkg/homologene/man/homologene.html" class="external-link">homologene</a>.<br><br><em>NOTE</em>: To return only the most "popular"
interspecies ortholog mappings,
supply <code>mthreshold=1</code> here AND set <code>method="gprofiler"</code> above.
This procedure tends to yield a greater number of returned genes but at
 the cost of many of them not being true biological 1:1 orthologs.<br><br>
 For more details, please see
<a href="https://cran.r-project.org/web/packages/gprofiler2/vignettes/gprofiler2.html" class="external-link">
 here</a>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Table with all gene symbols
 from the given <code>species</code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>References <a href="https://rdrr.io/pkg/homologene/man/homologeneData.html" class="external-link">homologeneData</a> or
<a href="https://rdrr.io/pkg/gprofiler2/man/gconvert.html" class="external-link">gconvert</a>.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">genome_mouse</span> <span class="op">&lt;-</span> <span class="fu">all_genes</span><span class="op">(</span>species <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: gprofiler</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in gprofiler.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using stored `gprofiler_orgs`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: mmusculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using cached file: /github/home/.cache/R/orthogene/all_genes-mmusculus-gprofiler.csv.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 78,202 genes from mouse.</span>
<span class="r-in"><span><span class="va">genome_human</span> <span class="op">&lt;-</span> <span class="fu">all_genes</span><span class="op">(</span>species <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: gprofiler</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in gprofiler.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using stored `gprofiler_orgs`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: hsapiens</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using cached file: /github/home/.cache/R/orthogene/all_genes-hsapiens-gprofiler.csv.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 78,687 genes from human.</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer></div>






  </body></html>

