<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Map genes from one species to another — convert_orthologs • orthogene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Map genes from one species to another — convert_orthologs"><meta property="og:description" content="Currently supports ortholog mapping between any
pair of 700+ species. 
Use map_species to
return a full list of available organisms."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Map genes from one species to another</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/HEAD/R/convert_orthologs.R" class="external-link"><code>R/convert_orthologs.R</code></a></small>
    <div class="hidden name"><code>convert_orthologs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Currently supports ortholog mapping between any
pair of 700+ species. <br>
Use <a href="map_species.html">map_species</a> to
return a full list of available organisms.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class="sourceCode r"><code><span class="fu">convert_orthologs</span><span class="op">(</span>
  <span class="va">gene_df</span>,
  gene_input <span class="op">=</span> <span class="st">"rownames"</span>,
  gene_output <span class="op">=</span> <span class="st">"rownames"</span>,
  standardise_genes <span class="op">=</span> <span class="cn">FALSE</span>,
  <span class="va">input_species</span>,
  output_species <span class="op">=</span> <span class="st">"human"</span>,
  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gprofiler"</span>, <span class="st">"homologene"</span>, <span class="st">"babelgene"</span><span class="op">)</span>,
  drop_nonorths <span class="op">=</span> <span class="cn">TRUE</span>,
  non121_strategy <span class="op">=</span> <span class="st">"drop_both_species"</span>,
  mthreshold <span class="op">=</span> <span class="cn">Inf</span>,
  as_sparse <span class="op">=</span> <span class="cn">FALSE</span>,
  sort_rows <span class="op">=</span> <span class="cn">FALSE</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>

    <div id="ref-sections">
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>gene_df</dt>
<dd><p>Data object containing the genes
(see <code>gene_input</code> for options on how
the genes can be stored within the object).<br>
Can be one of the following formats:<br></p><ul><li><p><code>matrix</code> : <br>A sparse or dense matrix.</p></li>
<li><p><code>data.frame</code> : <br>A <code>data.frame</code>,
 <code>data.table</code>. or <code>tibble</code>.</p></li>
<li><p>codelist : <br>A <code>list</code> or character <code>vector</code>.</p></li>
</ul><p>Genes, transcripts, proteins, SNPs, or genomic ranges
 can be provided in any format
(HGNC, Ensembl, RefSeq, UniProt, etc.) and will be
automatically converted to gene symbols unless
specified otherwise with the <code>...</code> arguments.<br><em>Note</em>: If you set <code>method="homologene"</code>, you
must either supply genes in gene symbol format (e.g. "Sox2")
 OR set <code>standardise_genes=TRUE</code>.</p></dd>
<dt>gene_input</dt>
<dd><p>Which aspect of <code>gene_df</code> to
get gene names from:<br></p><ul><li><p><code>"rownames"</code> : <br>From row names of data.frame/matrix.</p></li>
<li><p><code>"colnames"</code> : <br>From column names of data.frame/matrix.</p></li>
<li><p><code>&lt;column name&gt;</code> : <br>From a column in <code>gene_df</code>,
 e.g. <code>"gene_names"</code>.</p></li>
</ul></dd>
<dt>gene_output</dt>
<dd><p>How to return genes.
Options include:<br></p><ul><li><p><code>"rownames"</code> : <br>As row names of <code>gene_df</code>.</p></li>
<li><p><code>"colnames"</code> : <br>As column names of <code>gene_df</code>.</p></li>
<li><p><code>"columns"</code> : <br>As new columns "input_gene", "ortholog_gene"
(and "input_gene_standard" if <code>standardise_genes=TRUE</code>)
in <code>gene_df</code>.</p></li>
<li><p><code>"dict"</code> : <br>As a dictionary (named list) where the names
are input_gene and the values are ortholog_gene.</p></li>
<li><p><code>"dict_rev"</code> : <br>As a reversed dictionary (named list)
where the names are ortholog_gene and the values are input_gene.</p></li>
</ul></dd>
<dt>standardise_genes</dt>
<dd><p>If <code>TRUE</code> AND
<code>gene_output="columns"</code>, a new column "input_gene_standard"
will be added to <code>gene_df</code> containing standardised HGNC symbols
identified by <a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gorth</a>.</p></dd>
<dt>input_species</dt>
<dd><p>Name of the input species (e.g., "mouse","fly").
Use <a href="map_species.html">map_species</a> to return a full list
of available species.</p></dd>
<dt>output_species</dt>
<dd><p>Name of the output species (e.g. "human","chicken").
Use <a href="map_species.html">map_species</a> to return a full list
 of available species.</p></dd>
<dt>method</dt>
<dd><p>R package to to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives</p></li>
</ul><p>(slower but more species and genes) or
<code>"homologene"</code> (faster but fewer species and genes).</p></dd>
<dt>drop_nonorths</dt>
<dd><p>Drop genes that don't have an ortholog
in the <code>output_species</code>.</p></dd>
<dt>non121_strategy</dt>
<dd><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br></p><ul><li><p><code>"drop_both_species" or "dbs" or 1</code> : <br>
Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code> <br>
(<em>DEFAULT</em>).</p></li>
<li><p><code>"drop_input_species" or "dis" or 2</code> : <br>
Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></li>
<li><p><code>"drop_output_species" or "dos" or 3</code> : <br>
Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></li>
<li><p><code>"keep_both_species" or "kbs" or 4</code> : <br>
Keep all genes regardless of whether
they have duplicate mappings in either species.</p></li>
<li><p><code>"keep_popular" or "kp" or 5</code> : <br>
Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></li>
<li><p><code>"sum","mean","median","min" or "max"</code> : <br>
 When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></li>
</ul></dd>
<dt>mthreshold</dt>
<dd><p>Maximum number of ortholog names per gene to show.
Passed to <a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gorth</a>.
Only used when <code>method="gprofiler"</code> (<em>DEFAULT : </em><code>Inf</code>).</p></dd>
<dt>as_sparse</dt>
<dd><p>Convert <code>gene_df</code> to a sparse matrix.
Only works if <code>gene_df</code> is one of the following classes:<br></p><ul><li><p><code>matrix</code></p></li>
<li><p><code>Matrix</code></p></li>
<li><p><code>data.frame</code></p></li>
<li><p><code>data.table</code></p></li>
<li><p><code>tibble</code></p></li>
</ul><p>If <code>gene_df</code> is a sparse matrix to begin with,
it will be returned as a sparse matrix
 (so long as <code>gene_output=</code> <code>"rownames"</code> or <code>"colnames"</code>).</p></dd>
<dt>sort_rows</dt>
<dd><p>Sort <code>gene_df</code> rows alphanumerically.</p></dd>
<dt>verbose</dt>
<dd><p>Print messages.</p></dd>
<dt>...</dt>
<dd><p>Additional arguments to be passed to
<a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gorth</a> or <a href="https://rdrr.io/pkg/homologene/man/homologene.html" class="external-link">homologene</a>.<br><br><em>NOTE</em>: To return only the most "popular"
interspecies ortholog mappings,
supply <code>mthreshold=1</code> here AND set <code>method="gprofiler"</code> above.
This procedure tends to yield a greater number of returned genes but at
 the cost of many of them not being true biological 1:1 orthologs.<br><br>
 For more details, please see
<a href="https://cran.r-project.org/web/packages/gprofiler2/vignettes/gprofiler2.html" class="external-link">
 here</a>.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><code>gene_df</code> with  orthologs converted to the
<code>output_species</code>.<br>Instead returned as a dictionary (named list) if
<code>gene_output="dict"</code> or <code>"dict_rev"</code>.</p>
    </div>
    </div>

    <div id="examples">
    <h2>Examples</h2>
    <div class="ref-examples sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"exp_mouse"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">gene_df</span> <span class="op">&lt;-</span> <span class="fu">convert_orthologs</span><span class="op">(</span></span>
<span class="r-in">    gene_df <span class="op">=</span> <span class="va">exp_mouse</span>,</span>
<span class="r-in">    input_species <span class="op">=</span> <span class="st">"mouse"</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> sparseMatrix format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from rownames.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 15,259 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; human orthologs using: gprofiler</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in gprofiler.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using stored `gprofiler_orgs`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: mmusculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in gprofiler.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using stored `gprofiler_orgs`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: hsapiens</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,021 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,021 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 2,769 NAs of all kinds from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 478 genes that have multiple input_gene per ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 299 genes that have multiple ortholog_gene per input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting ortholog_gene to rownames.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    3,025 / 15,259 (20%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    12,234 / 15,259 (80%)</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.9000.9000.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

