<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Expand/aggregate rows of matrix for many:many mappings — many2many_rows • orthogene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Expand/aggregate rows of matrix for many:many mappings — many2many_rows"><meta property="og:description" content="Expand/aggregate rows of a matrix with any combination of 
 many:many mappings. 
This method ensures that total counts per gene remain the
same regardless of how many genes it has split/condensed into.
This allows for many:many mappings that are otherwise not possible
using standard aggregation functions,
since they all require many:1 scenarios.
Internally, this is done as follows:
Identify genes that appear more than once
 in gene_map[[input_col]].
For each gene identified, split its row into multiple
 rows, where the number of new rows is equal to the number of times that
 gene appears within gene_map[[input_col]].
In the new expanded matrix, each row will be equal to the column sums
divided by the number of new rows. This means that averaged counts will be 
split equally amongst the new rows, in a column-specific manner.
Thus, the column sums of the output matrix will be equal 
to the column sums in the input matrix. 
In the case of gene expression count matrices,
 this means that the total counts will remain equal between matrices, 
 while avoiding being forced to drop genes with many:many mappings 
 (as is the case with most other aggregation methods).
Map rownames of the expanded matrix onto the orthologous gene names
from gene_map$ortholog_gene.
[Optional] : When aggregate_orthologs=TRUE, 
aggregate rows of the expanded/mapped matrix 
such that there will only be 1 row per ortholog gene, 
using aggregate_rows. 
The arguments FUN, method, 
as_sparse, as_DelayedArray, and dropNA will all
be passed to aggregate_rows if this step is selected.

"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.6.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Expand/aggregate rows of matrix for many:many mappings</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/HEAD/R/many2many_rows.R" class="external-link"><code>R/many2many_rows.R</code></a></small>
    <div class="hidden name"><code>many2many_rows.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Expand/aggregate rows of a matrix with any combination of 
 many:many mappings. 
This method ensures that total counts per gene remain the
same regardless of how many genes it has split/condensed into.
This allows for many:many mappings that are otherwise not possible
using standard aggregation functions,
since they all require many:1 scenarios.<br>
Internally, this is done as follows:</p><ol><li><p>Identify genes that appear more than once
 in <code>gene_map[[input_col]]</code>.</p></li>
<li><p>For each gene identified, split its row into multiple
 rows, where the number of new rows is equal to the number of times that
 gene appears within <code>gene_map[[input_col]]</code>.
In the new expanded matrix, each row will be equal to the column sums
divided by the number of new rows. This means that averaged counts will be 
split equally amongst the new rows, in a column-specific manner.<br>
Thus, the column sums of the output matrix will be equal 
to the column sums in the input matrix. 
In the case of gene expression count matrices,
 this means that the total counts will remain equal between matrices, 
 while avoiding being forced to drop genes with many:many mappings 
 (as is the case with most other aggregation methods).</p></li>
<li><p>Map rownames of the expanded matrix onto the orthologous gene names
from <code>gene_map$ortholog_gene</code>.</p></li>
<li><p>[Optional] : When <code>aggregate_orthologs=TRUE</code>, 
aggregate rows of the expanded/mapped matrix 
such that there will only be 1 row per ortholog gene, 
using <a href="aggregate_rows.html">aggregate_rows</a>. 
The arguments <code>FUN</code>, <code>method</code>, 
<code>as_sparse</code>, <code>as_DelayedArray</code>, and <code>dropNA</code> will all
be passed to <a href="aggregate_rows.html">aggregate_rows</a> if this step is selected.</p></li>
</ol></div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">many2many_rows</span><span class="op">(</span></span>
<span>  <span class="va">X</span>,</span>
<span>  <span class="va">gene_map</span>,</span>
<span>  input_col <span class="op">=</span> <span class="st">"input_gene"</span>,</span>
<span>  output_col <span class="op">=</span> <span class="st">"ortholog_gene"</span>,</span>
<span>  agg_fun <span class="op">=</span> <span class="st">"sum"</span>,</span>
<span>  agg_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"monocle3"</span>, <span class="st">"stats"</span><span class="op">)</span>,</span>
<span>  as_sparse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  as_DelayedArray <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  dropNA <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  aggregate_orthologs <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><code>
data("exp_mouse")
X &lt;- exp_mouse 
gene_map &lt;- orthogene:::map_orthologs(genes = rownames(exp_mouse),
                                      input_species = "mouse",
                                      method="homologene")
X_agg &lt;- orthogene:::many2many_rows(X = X,
                                    gene_map = gene_map)
sum(duplicated(rownames(exp_mouse))) # 0                                      
sum(duplicated(gene_map$input_gene)) # 46
sum(duplicated(gene_map$ortholog_gene)) # 56
sum(duplicated(rownames(X_agg))) # 56
</code></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>X</dt>
<dd><p>Input matrix.</p></dd>


<dt>gene_map</dt>
<dd><p>A <a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a> generated by
 <a href="map_orthologs.html">map_orthologs</a>, 
with columns mapping <code>input_col</code> to <code>output_col</code>.</p></dd>


<dt>input_col</dt>
<dd><p>Column name within <code>gene_map</code> with gene names matching 
the row names of <code>X</code>.</p></dd>


<dt>output_col</dt>
<dd><p>Column name within <code>gene_map</code> with gene names
that you wish you map the row names of <code>X</code> onto.</p></dd>


<dt>agg_fun</dt>
<dd><p>Aggregation function.</p></dd>


<dt>agg_method</dt>
<dd><p>Aggregation method.</p></dd>


<dt>as_sparse</dt>
<dd><p>Convert aggregated matrix to sparse matrix.</p></dd>


<dt>as_DelayedArray</dt>
<dd><p>Convert aggregated matrix to
<a href="https://rdrr.io/pkg/DelayedArray/man/DelayedArray-class.html" class="external-link">DelayedArray</a>.</p></dd>


<dt>dropNA</dt>
<dd><p>Drop genes assigned to <code>NA</code> in <code>groupings</code>.</p></dd>


<dt>aggregate_orthologs</dt>
<dd><p>[Optional] After performing an initial round of 
many:many aggregation/expansion with many2many_rows,
 ensure each orthologous gene only appears in one row by using the 
<a href="aggregate_rows.html">aggregate_rows</a> function (default: <code>TRUE</code>).</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Expanded/aggregated matrix.</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

