<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Create a phylogenetic tree of shared orthologs — plot_orthotree • orthogene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Create a phylogenetic tree of shared orthologs — plot_orthotree"><meta property="og:description" content='Automatically creates a phylogenetic tree plot annotated with metadata
describing how many orthologous genes each species shares with the
 reference_species ("human" by default).'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create a phylogenetic tree of shared orthologs</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/HEAD/R/plot_orthotree.R" class="external-link"><code>R/plot_orthotree.R</code></a></small>
    <div class="hidden name"><code>plot_orthotree.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Automatically creates a phylogenetic tree plot annotated with metadata
describing how many orthologous genes each species shares with the
 <code>reference_species</code> ("human" by default).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_orthotree</span><span class="op">(</span></span>
<span>  tree <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  orth_report <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  species <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"babelgene"</span>, <span class="st">"homologene"</span>, <span class="st">"gprofiler"</span><span class="op">)</span>,</span>
<span>  tree_source <span class="op">=</span> <span class="st">"timetree"</span>,</span>
<span>  non121_strategy <span class="op">=</span> <span class="st">"drop_both_species"</span>,</span>
<span>  reference_species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  clades <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Primates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Homo sapiens"</span>, <span class="st">"Macaca mulatta"</span><span class="op">)</span>, Eutherians <span class="op">=</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Homo sapiens"</span>, <span class="st">"Mus musculus"</span>, <span class="st">"Bos taurus"</span><span class="op">)</span>, Mammals <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Homo sapiens"</span>,</span>
<span>    <span class="st">"Mus musculus"</span>, <span class="st">"Bos taurus"</span>, <span class="st">"Ornithorhynchus anatinus"</span>, <span class="st">"Monodelphis domestica"</span><span class="op">)</span>,</span>
<span>    Tetrapods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Homo sapiens"</span>, <span class="st">"Mus musculus"</span>, <span class="st">"Gallus gallus"</span>, <span class="st">"Anolis carolinensis"</span>,</span>
<span>    <span class="st">"Xenopus tropicalis"</span><span class="op">)</span>, Vertebrates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Homo sapiens"</span>, <span class="st">"Mus musculus"</span>,</span>
<span>    <span class="st">"Gallus gallus"</span>, <span class="st">"Anolis carolinensis"</span>, <span class="st">"Xenopus tropicalis"</span>, <span class="st">"Danio rerio"</span><span class="op">)</span>,</span>
<span>    Invertebrates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Drosophila melanogaster"</span>, </span>
<span>     <span class="st">"Caenorhabditis elegans"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  clades_rotate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  scaling_factor <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  save_paths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".ggtree.pdf"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".ggtree.png"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  height <span class="op">=</span> <span class="va">width</span>,</span>
<span>  mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://yulab-smu.top/treedata-book/chapter8.html#phylopic" class="external-link">
ggtree tutorial</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>tree</dt>
<dd><p>A phylogenetic tree of class <a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">phylo</a>. If no tree 
is provided (<code>NULL</code>) a 100-way multiz tree will be imported from  
<a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/multiz100way" class="external-link">
UCSC Genome Browser</a>.</p></dd>


<dt>orth_report</dt>
<dd><p>An ortholog report from one or more species 
generated by <a href="report_orthologs.html">report_orthologs</a>.</p></dd>


<dt>species</dt>
<dd><p>Species to include in the final plot.
If <code>NULL</code>, then all species from the given database (<code>method</code>) 
will be included (via <a href="map_species.html">map_species</a>), 
so long as they also exist in the <code>tree</code>.</p></dd>


<dt>method</dt>
<dd><p>R package to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>


<dt>tree_source</dt>
<dd><p>Can be one of the following:</p><ul><li><p>"timetree2022":<br>Import and prune the
<a href="https://doi.org/10.1093/molbev/msac174" class="external-link">
TimeTree &gt;147k species</a> phylogenetic tree. Can also simply type "timetree".</p></li>
<li><p>"timetree2015":<br>Import and prune the
<a href="http://www.timetree.org/public/data/TimetreeOfLife2015.nwk" class="external-link">
TimeTree &gt;50k species</a> phylogenetic tree.</p></li>
<li><p>"OmaDB":<br>Construct a tree from <a href="omabrowser.org">OMA</a> 
(Orthologous Matrix browser) via the <a href="https://rdrr.io/pkg/OmaDB/man/getTaxonomy.html" class="external-link">getTaxonomy</a> function.
<em>NOTE: </em> Does not contain branch lengths,
 and therefore may have limited utility.</p></li>
<li><p>"UCSC":<br>Import and prune the 
<a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/multiz100way/" class="external-link">
UCSC 100-way alignment</a> phylogenetic tree (hg38 version).</p></li>
<li><p>"&lt;path&gt;":<br>Read a tree from a newick text file 
from a local or remote URL using <a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">read.tree</a>.</p></li>
</ul></dd>


<dt>non121_strategy</dt>
<dd><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br></p><ul><li><p><code>"drop_both_species" or "dbs" or 1</code> : <br>
Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code> <br>
(<em>DEFAULT</em>).</p></li>
<li><p><code>"drop_input_species" or "dis" or 2</code> : <br>
Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></li>
<li><p><code>"drop_output_species" or "dos" or 3</code> : <br>
Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></li>
<li><p><code>"keep_both_species" or "kbs" or 4</code> : <br>
Keep all genes regardless of whether
they have duplicate mappings in either species.</p></li>
<li><p><code>"keep_popular" or "kp" or 5</code> : <br>
Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></li>
<li><p><code>"sum","mean","median","min" or "max"</code> : <br>
 When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></li>
</ul></dd>


<dt>reference_species</dt>
<dd><p>Reference species.</p></dd>


<dt>clades</dt>
<dd><p>A named list of clades each containing 
a character vector of species used to define the respective clade 
using <a href="https://rdrr.io/pkg/tidytree/man/MRCA.html" class="external-link">MRCA</a>.</p></dd>


<dt>clades_rotate</dt>
<dd><p>A list of clades to rotate (via <a href="https://rdrr.io/pkg/ape/man/rotate.html" class="external-link">rotate</a>), 
each containing a character vector of species used to define the respective
 clade using <a href="https://rdrr.io/pkg/tidytree/man/MRCA.html" class="external-link">MRCA</a>.</p></dd>


<dt>scaling_factor</dt>
<dd><p>How much to scale y-axis parameters (e.g. offset) by.</p></dd>


<dt>show_plot</dt>
<dd><p>Whether to print the final tree plot.</p></dd>


<dt>save_paths</dt>
<dd><p>Paths to save plot to.</p></dd>


<dt>width</dt>
<dd><p>Saved plot width.</p></dd>


<dt>height</dt>
<dd><p>Saved plot height.</p></dd>


<dt>mc.cores</dt>
<dd><p>Number of cores to parallelise different steps with.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A list containing:</p><ul><li><p>plot : Annotated ggtree object.</p></li>
<li><p>tree : The pruned, standardised phylogenetic tree used in the plot.</p></li>
<li><p>orth_report : Ortholog reports for each <code>species</code> against
 the <code>reference_species</code>.</p></li>
<li><p>metadata : Metadata used in the plot,
 including silhouette PNG ids from <a href="http://phylopic.org/" class="external-link">phylopic</a>.</p></li>
<li><p>clades : Metadata used for highlighting clades.</p></li>
<li><p>method : <code>method</code> used.</p></li>
<li><p>reference_species : <code>reference_species</code> used.</p></li>
<li><p>save_paths : <code>save_paths</code> to plot.</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">orthotree</span> <span class="op">&lt;-</span> <span class="fu">plot_orthotree</span><span class="op">(</span>species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"human"</span>,<span class="st">"monkey"</span>,<span class="st">"mouse"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering ortholog reports.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Homo sapiens</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing babelgene::orthologs_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,492 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,492 genes from Homo sapiens.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Homo sapiens</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing babelgene::orthologs_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,492 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,492 genes from Homo sapiens.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20,206 / 20,206 (100%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20,206 / 20,206 (100%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9544</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing babelgene::orthologs_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,402 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 20,402 genes from monkey.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.frame format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from Gene.Symbol.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20,402 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting monkey ==&gt; Homo sapiens orthologs using: babelgene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for monkey</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: Macaca mulatta</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Homo sapiens</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: Homo sapiens</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Macaca mulatta</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9544</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing babelgene::orthologs_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 20,402 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Homo sapiens.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20,402 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 561 NAs of all kinds from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 19,841 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 107 NAs of all kinds from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 1,371 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 1,000 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    1,452 / 18,149 (8%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    16,697 / 18,149 (92%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,459 / 18,149 (90.69%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 16,459 / 20,206 (81.46%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing babelgene::orthologs_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 29,651 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 29,651 genes from mouse.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.frame format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from Gene.Symbol.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 29,651 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting mouse ==&gt; Homo sapiens orthologs using: babelgene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for mouse</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: Mus musculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Homo sapiens</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: Homo sapiens</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in babelgene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: Mus musculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 10090</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing babelgene::orthologs_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 29,651 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in Homo sapiens.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 29,651 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 37 NAs of all kinds from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 29,614 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 146 NAs of all kinds from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 9,286 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 10,297 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    4,602 / 20,075 (23%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    15,473 / 20,075 (77%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 15,314 / 20,075 (76.28%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 15,314 / 20,206 (75.79%) reference_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: phytools</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: TreeTools</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing tree from: TimeTree2022</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing cached tree.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising tip labels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping 3 species from `species`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping 3 species from tree.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0/3 (0%) tips dropped from tree due to inability to standardise names with `map_species`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0/3 (0%) tips dropped from tree according to overlap with selected `species`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering phylopic silhouettes.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing data for 6 clades.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning: Each clade in `clades` must contain a vector of at least 1 species. Omitting clade: Invertebrates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3 species remaining after metadata preparation.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating ggtree plot.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving plot ==&gt; /tmp/RtmpQNLDlX/file219267fc0581.ggtree.pdf</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Saving plot ==&gt; /tmp/RtmpQNLDlX/file21922635c8c0.ggtree.png</span>
<span class="r-plt img"><img src="plot_orthotree-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

