<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Prepare a phylogenetic tree — prepare_tree • orthogene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Prepare a phylogenetic tree — prepare_tree"><meta property="og:description" content="Import a phylogenetic tree  and then conduct 
a series of optional standardisation steps.
Optionally, if output_format is not NULL, species names from 
 both the tree and the species argument will first be standardised
 using map_species."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prepare a phylogenetic tree</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/HEAD/R/prepare_tree.R" class="external-link"><code>R/prepare_tree.R</code></a></small>
    <div class="hidden name"><code>prepare_tree.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Import a phylogenetic tree  and then conduct 
a series of optional standardisation steps.
Optionally, if <code>output_format</code> is not <code>NULL</code>, species names from 
 both the tree and the <code>species</code> argument will first be standardised
 using <a href="map_species.html">map_species</a>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">prepare_tree</span><span class="op">(</span></span>
<span>  tree_source <span class="op">=</span> <span class="st">"timetree"</span>,</span>
<span>  species <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  output_format <span class="op">=</span> <span class="st">"scientific_name_formatted"</span>,</span>
<span>  run_map_species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"homologene"</span>, <span class="st">"gprofiler"</span>, <span class="st">"babelgene"</span><span class="op">)</span>,</span>
<span>  force_ultrametric <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  age_max <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  save_dir <span class="op">=</span> <span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/userdir.html" class="external-link">R_user_dir</a></span><span class="op">(</span><span class="st">"orthogene"</span>, which <span class="op">=</span> <span class="st">"cache"</span><span class="op">)</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p><a href="https://doi.org/10.1093/molbev/msac174" class="external-link">
TimeTree 5: An Expanded Resource for Species Divergence Times</a></p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>tree_source</dt>
<dd><p>Can be one of the following:</p><ul><li><p>"timetree2022":<br>Import and prune the
<a href="https://doi.org/10.1093/molbev/msac174" class="external-link">
TimeTree &gt;147k species</a> phylogenetic tree. Can also simply type "timetree".</p></li>
<li><p>"timetree2015":<br>Import and prune the
<a href="http://www.timetree.org/public/data/TimetreeOfLife2015.nwk" class="external-link">
TimeTree &gt;50k species</a> phylogenetic tree.</p></li>
<li><p>"OmaDB":<br>Construct a tree from <a href="omabrowser.org">OMA</a> 
(Orthologous Matrix browser) via the <a href="https://rdrr.io/pkg/OmaDB/man/getTaxonomy.html" class="external-link">getTaxonomy</a> function.
<em>NOTE: </em> Does not contain branch lengths,
 and therefore may have limited utility.</p></li>
<li><p>"UCSC":<br>Import and prune the 
<a href="https://hgdownload.soe.ucsc.edu/goldenPath/hg38/multiz100way/" class="external-link">
UCSC 100-way alignment</a> phylogenetic tree (hg38 version).</p></li>
<li><p>"&lt;path&gt;":<br>Read a tree from a newick text file 
from a local or remote URL using <a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">read.tree</a>.</p></li>
</ul></dd>


<dt>species</dt>
<dd><p>Species names to subset the tree by 
(after <code>standardise_species</code> step).</p></dd>


<dt>output_format</dt>
<dd><p>Which column to return.</p></dd>


<dt>run_map_species</dt>
<dd><p>Whether to first standardise species names with 
<a href="map_species.html">map_species</a>.</p></dd>


<dt>method</dt>
<dd><p>R package to use for gene mapping:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>


<dt>force_ultrametric</dt>
<dd><p>Whether to force the tree to be ultrametric 
(i.e. make all tips the same date) using <a href="https://rdrr.io/pkg/phytools/man/force.ultrametric.html" class="external-link">force.ultrametric</a>.</p></dd>


<dt>age_max</dt>
<dd><p>Rescale the edges of the tree into units of 
millions of years (MY) instead than evolutionary rates (e.g. dN/dS ratios).
Only used if <code>age_max</code>, the max number , is numeric. Times are computed using
<a href="https://rdrr.io/pkg/ape/man/chronos.html" class="external-link">makeChronosCalib</a> and <a href="https://rdrr.io/pkg/ape/man/chronos.html" class="external-link">chronos</a>.</p></dd>


<dt>show_plot</dt>
<dd><p>Show a basic plot of the resulting tree.</p></dd>


<dt>save_dir</dt>
<dd><p>Directory to cache full tree in. 
Set to <code>NULL</code> to avoid using cache.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>


<dt>...</dt>
<dd><p>Additional arguments passed to <a href="https://rdrr.io/pkg/ape/man/chronos.html" class="external-link">makeChronosCalib</a>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A filtered tree of class "phylo" (with standardised species names).</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"human"</span>,<span class="st">"chimp"</span>,<span class="st">"mouse"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tr</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu">prepare_tree</span><span class="op">(</span>species <span class="op">=</span> <span class="va">species</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing tree from: TimeTree2022</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing cached tree.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Standardising tip labels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping 3 species from `species`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping 3 species from tree.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0/3 (0%) tips dropped from tree due to inability to standardise names with `map_species`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0/3 (0%) tips dropped from tree according to overlap with selected `species`.</span>
<span class="r-plt img"><img src="prepare_tree-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

