<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Report orthologs — report_orthologs • orthogene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Report orthologs — report_orthologs"><meta property="og:description" content="Identify the number of orthologous genes between two species."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.15.01</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Report orthologs</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/main/R/report_orthologs.R" class="external-link"><code>R/report_orthologs.R</code></a></small>
    <div class="hidden name"><code>report_orthologs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Identify the number of orthologous genes between two species.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">report_orthologs</span><span class="op">(</span></span>
<span>  target_species <span class="op">=</span> <span class="st">"mouse"</span>,</span>
<span>  reference_species <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  standardise_genes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  method_all_genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"homologene"</span>, <span class="st">"gprofiler"</span>, <span class="st">"babelgene"</span><span class="op">)</span>,</span>
<span>  method_convert_orthologs <span class="op">=</span> <span class="va">method_all_genes</span>,</span>
<span>  drop_nonorths <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  non121_strategy <span class="op">=</span> <span class="st">"drop_both_species"</span>,</span>
<span>  round_digits <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  return_report <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  ref_genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-target-species">target_species<a class="anchor" aria-label="anchor" href="#arg-target-species"></a></dt>
<dd><p>Target species.</p></dd>


<dt id="arg-reference-species">reference_species<a class="anchor" aria-label="anchor" href="#arg-reference-species"></a></dt>
<dd><p>Reference species.</p></dd>


<dt id="arg-standardise-genes">standardise_genes<a class="anchor" aria-label="anchor" href="#arg-standardise-genes"></a></dt>
<dd><p>If <code>TRUE</code> AND
<code>gene_output="columns"</code>, a new column "input_gene_standard"
will be added to <code>gene_df</code> containing standardised HGNC symbols
identified by <a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gorth</a>.</p></dd>


<dt id="arg-method-all-genes">method_all_genes<a class="anchor" aria-label="anchor" href="#arg-method-all-genes"></a></dt>
<dd><p>Package to use in the
<a href="all_genes.html">all_genes</a> step:</p><dl><dt><code>"gprofiler"</code></dt>
<dd><p>Slower, but covers more species and genes.</p></dd>

  <dt><code>"homologene"</code></dt>
<dd><p>Faster, but covers fewer species and genes.</p></dd>

  <dt><code>"babelgene"</code></dt>
<dd><p>Faster, fewer species/genes; also provides
    consensus scores for each mapping from multiple data sources.</p></dd>


</dl></dd>


<dt id="arg-method-convert-orthologs">method_convert_orthologs<a class="anchor" aria-label="anchor" href="#arg-method-convert-orthologs"></a></dt>
<dd><p>Package to use in the
<a href="convert_orthologs.html">convert_orthologs</a> step:</p><dl><dt><code>"gprofiler"</code></dt>
<dd><p>Slower, but covers more species and genes.</p></dd>

  <dt><code>"homologene"</code></dt>
<dd><p>Faster, but covers fewer species and genes.</p></dd>

  <dt><code>"babelgene"</code></dt>
<dd><p>Faster, fewer species/genes; also provides
    consensus scores for each mapping from multiple data sources.</p></dd>


</dl></dd>


<dt id="arg-drop-nonorths">drop_nonorths<a class="anchor" aria-label="anchor" href="#arg-drop-nonorths"></a></dt>
<dd><p>Drop genes that don't have an ortholog
in the <code>output_species</code>.</p></dd>


<dt id="arg-non-strategy">non121_strategy<a class="anchor" aria-label="anchor" href="#arg-non-strategy"></a></dt>
<dd><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br></p><dl><dt><code>"drop_both_species" or "dbs" or 1</code></dt>
<dd><p>Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code>
(<em>DEFAULT</em>).</p></dd>

<dt><code>"drop_input_species" or "dis" or 2</code></dt>
<dd><p>Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></dd>

<dt><code>"drop_output_species" or "dos" or 3</code></dt>
<dd><p>Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></dd>

<dt><code>"keep_both_species" or "kbs" or 4</code></dt>
<dd><p>Keep all genes regardless of whether
they have duplicate mappings in either species.</p></dd>

<dt><code>"keep_popular" or "kp" or 5</code></dt>
<dd><p>Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></dd>

 <dt><code>"sum","mean","median","min" or "max"</code></dt>
<dd><p>When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></dd>


</dl></dd>


<dt id="arg-round-digits">round_digits<a class="anchor" aria-label="anchor" href="#arg-round-digits"></a></dt>
<dd><p>Number of digits to round to when printing percentages.</p></dd>


<dt id="arg-return-report">return_report<a class="anchor" aria-label="anchor" href="#arg-return-report"></a></dt>
<dd><p>If <code>FALSE</code>, return only the ortholog mapping
between the two species. If <code>TRUE</code>, return both the ortholog mapping
and a <code>data.frame</code> of report statistics.</p></dd>


<dt id="arg-ref-genes">ref_genes<a class="anchor" aria-label="anchor" href="#arg-ref-genes"></a></dt>
<dd><p>A table of all genes for <code>reference_species</code>.
If <code>NULL</code> (default), this is created via
<a href="all_genes.html">all_genes</a>.</p></dd>


<dt id="arg-mc-cores">mc.cores<a class="anchor" aria-label="anchor" href="#arg-mc-cores"></a></dt>
<dd><p>Number of cores to parallelize across <code>target_species</code>.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Print messages.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Arguments passed on to <code><a href="convert_orthologs.html">convert_orthologs</a></code></p><dl><dt><code>gene_df</code></dt>
<dd><p>Data object containing the genes
(see <code>gene_input</code> for options on how
the genes can be stored within the object).<br>
Can be one of the following formats:<br></p><dl><dt><code>matrix</code></dt>
<dd><p>A sparse or dense matrix.</p></dd>

<dt><code>data.frame</code></dt>
<dd><p>A <code>data.frame</code>,
 <code>data.table</code>. or <code>tibble</code>.</p></dd>

<dt><code>list</code></dt>
<dd><p>A <code>list</code> or character <code>vector</code>.</p></dd>


</dl><p>Genes, transcripts, proteins, SNPs, or genomic ranges
 can be provided in any format
(HGNC, Ensembl, RefSeq, UniProt, etc.) and will be
automatically converted to gene symbols unless
specified otherwise with the <code>...</code> arguments.<br><em>Note</em>: If you set <code>method="homologene"</code>, you
must either supply genes in gene symbol format (e.g. "Sox2")
 OR set <code>standardise_genes=TRUE</code>.</p></dd>

    <dt><code>gene_input</code></dt>
<dd><p>Which aspect of <code>gene_df</code> to
get gene names from:<br></p><dl><dt><code>"rownames"</code></dt>
<dd><p>From row names of data.frame/matrix.</p></dd>

<dt><code>"colnames"</code></dt>
<dd><p>From column names of data.frame/matrix.</p></dd>

<dt><code>&lt;column name&gt;</code></dt>
<dd><p>From a column in <code>gene_df</code>,
 e.g. <code>"gene_names"</code>.</p></dd>


</dl></dd>

    <dt><code>gene_output</code></dt>
<dd><p>How to return genes.
Options include:<br></p><dl><dt><code>"rownames"</code></dt>
<dd><p>As row names of <code>gene_df</code>.</p></dd>

<dt><code>"colnames"</code></dt>
<dd><p>As column names of <code>gene_df</code>.</p></dd>

<dt><code>"columns"</code></dt>
<dd><p>As new columns "input_gene", "ortholog_gene"
(and "input_gene_standard" if <code>standardise_genes=TRUE</code>)
in <code>gene_df</code>.</p></dd>

<dt><code>"dict"</code></dt>
<dd><p>As a dictionary (named list) where the names
are input_gene and the values are ortholog_gene</p></dd>
.
<dt><code>"dict_rev"</code></dt>
<dd><p>As a reversed dictionary (named list)
where the names are ortholog_gene and the values are input_gene</p></dd>
.

</dl></dd>

    <dt><code>input_species</code></dt>
<dd><p>Name of the input species (e.g., "mouse","fly").
Use <a href="map_species.html">map_species</a> to return a full list
of available species.</p></dd>

    <dt><code>output_species</code></dt>
<dd><p>Name of the output species (e.g. "human","chicken").
Use <a href="map_species.html">map_species</a> to return a full list
 of available species.</p></dd>

    <dt><code>agg_fun</code></dt>
<dd><p>Aggregation function passed to
 <a href="aggregate_mapped_genes.html">aggregate_mapped_genes</a>.
Set to <code>NULL</code> to skip aggregation step (default).</p></dd>

    <dt><code>mthreshold</code></dt>
<dd><p>Maximum number of ortholog names per gene to show.
Passed to <a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gorth</a>.
Only used when <code>method="gprofiler"</code> (<em>DEFAULT : </em><code>Inf</code>).</p></dd>

    <dt><code>method</code></dt>
<dd><p>R package to use for gene mapping:</p><dl><dt><code>"gprofiler"</code></dt>
<dd><p>Slower but more species and genes.</p></dd>

 <dt><code>"homologene"</code></dt>
<dd><p>Faster but fewer species and genes.</p></dd>

 <dt><code>"babelgene"</code></dt>
<dd><p>Faster but fewer species and genes.
 Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></dd>


</dl></dd>

    <dt><code>as_sparse</code></dt>
<dd><p>Convert <code>gene_df</code> to a sparse matrix.
Only works if <code>gene_df</code> is one of the following classes:<br></p><ul><li><p><code>matrix</code></p></li>
<li><p><code>Matrix</code></p></li>
<li><p><code>data.frame</code></p></li>
<li><p><code>data.table</code></p></li>
<li><p><code>tibble</code></p></li>
</ul><p>If <code>gene_df</code> is a sparse matrix to begin with,
it will be returned as a sparse matrix
 (so long as <code>gene_output=</code> <code>"rownames"</code> or <code>"colnames"</code>).</p></dd>

    <dt><code>sort_rows</code></dt>
<dd><p>Sort <code>gene_df</code> rows alphanumerically.</p></dd>

    <dt><code>gene_map</code></dt>
<dd><p>A <a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a> that maps the current gene names
to new gene names.
This function's behaviour will adapt to different situations as follows:</p><dl><dt><code>gene_map=&lt;data.frame&gt;</code></dt>
<dd><p>When a data.frame containing the
gene key:value columns
(specified by <code>input_col</code> and <code>output_col</code>, respectively)
is provided, this will be used to perform aggregation/expansion.</p></dd>

<dt><code>gene_map=NULL</code> and <code>input_species!=output_species</code></dt>
<dd><p>A <code>gene_map</code> is automatically generated by
 <a href="map_orthologs.html">map_orthologs</a> to perform inter-species
 gene aggregation/expansion.</p></dd>

<dt><code>gene_map=NULL</code> and <code>input_species==output_species</code></dt>
<dd><p>A <code>gene_map</code> is automatically generated by
 <a href="map_genes.html">map_genes</a> to perform within-species
 gene gene symbol standardization and aggregation/expansion.</p></dd>


</dl></dd>

    <dt><code>as_DelayedArray</code></dt>
<dd><p>Convert aggregated matrix to
<a href="https://rdrr.io/pkg/DelayedArray/man/DelayedArray-class.html" class="external-link">DelayedArray</a>.</p></dd>

    <dt><code>input_col</code></dt>
<dd><p>Column name within <code>gene_map</code> with gene names matching
the row names of <code>X</code>.</p></dd>

    <dt><code>output_col</code></dt>
<dd><p>Column name within <code>gene_map</code> with gene names
that you wish you map the row names of <code>X</code> onto.</p></dd>


</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list containing:</p><dl><dt>map</dt>
<dd><p>A table of inter-species gene mappings.</p></dd>

  <dt>report</dt>
<dd><p>A list of aggregate orthology report statistics.</p></dd>


</dl><p>If more than one <code>target_species</code> is provided, the function returns a
table of aggregated <code>report</code> statistics concatenated across species.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">orth_fly</span> <span class="op">&lt;-</span> <span class="fu">report_orthologs</span><span class="op">(</span></span></span>
<span class="r-in"><span>  target_species <span class="op">=</span> <span class="st">"fly"</span>,</span></span>
<span class="r-in"><span>  reference_species <span class="op">=</span> <span class="st">"human"</span></span></span>
<span class="r-in"><span><span class="op">)</span> </span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gathering ortholog reports.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7227</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 8,438 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 8,438 genes from fly.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.frame format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from Gene.Symbol.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 8,438 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting fly ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7227</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4,500 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 1 NAs of all kinds from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4,499 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 19 genes that have multiple input_gene per ortholog_gene (many:1).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 266 genes that have multiple ortholog_gene per input_gene (1:many).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    4,380 / 8,438 (52%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    4,058 / 8,438 (48%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4,058 / 8,438 (48.09%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4,058 / 19,129 (21.21%) reference_species genes remain after ortholog conversion.</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

