<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Report orthologs — report_orthologs • orthogene</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Report orthologs — report_orthologs"><meta property="og:description" content="Identify the number of orthologous genes between two species."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Report orthologs</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/HEAD/R/report_orthologs.R" class="external-link"><code>R/report_orthologs.R</code></a></small>
    <div class="hidden name"><code>report_orthologs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Identify the number of orthologous genes between two species.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">report_orthologs</span><span class="op">(</span>
  target_species <span class="op">=</span> <span class="st">"mouse"</span>,
  reference_species <span class="op">=</span> <span class="st">"human"</span>,
  standardise_genes <span class="op">=</span> <span class="cn">FALSE</span>,
  method_all_genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"homologene"</span>, <span class="st">"gprofiler"</span>, <span class="st">"babelgene"</span><span class="op">)</span>,
  method_convert_orthologs <span class="op">=</span> <span class="va">method_all_genes</span>,
  drop_nonorths <span class="op">=</span> <span class="cn">TRUE</span>,
  non121_strategy <span class="op">=</span> <span class="st">"drop_both_species"</span>,
  round_digits <span class="op">=</span> <span class="fl">2</span>,
  return_report <span class="op">=</span> <span class="cn">TRUE</span>,
  mc.cores <span class="op">=</span> <span class="fl">1</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>target_species</dt>
<dd><p>Target species.</p></dd>
<dt>reference_species</dt>
<dd><p>Reference species.</p></dd>
<dt>standardise_genes</dt>
<dd><p>If <code>TRUE</code> AND
<code>gene_output="columns"</code>, a new column "input_gene_standard"
will be added to <code>gene_df</code> containing standardised HGNC symbols
identified by <a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gorth</a>.</p></dd>
<dt>method_all_genes</dt>
<dd><p>R package to to use in
<a href="all_genes.html">all_genes</a> step:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>
<dt>method_convert_orthologs</dt>
<dd><p>R package to to use in
<a href="convert_orthologs.html">convert_orthologs</a> step:</p><ul><li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives consensus scores for each gene mapping based on a
 several different data sources.</p></li>
</ul></dd>
<dt>drop_nonorths</dt>
<dd><p>Drop genes that don't have an ortholog
in the <code>output_species</code>.</p></dd>
<dt>non121_strategy</dt>
<dd><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br></p><ul><li><p><code>"drop_both_species" or "dbs" or 1</code> : <br>
Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code> <br>
(<em>DEFAULT</em>).</p></li>
<li><p><code>"drop_input_species" or "dis" or 2</code> : <br>
Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></li>
<li><p><code>"drop_output_species" or "dos" or 3</code> : <br>
Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></li>
<li><p><code>"keep_both_species" or "kbs" or 4</code> : <br>
Keep all genes regardless of whether
they have duplicate mappings in either species.</p></li>
<li><p><code>"keep_popular" or "kp" or 5</code> : <br>
Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></li>
<li><p><code>"sum","mean","median","min" or "max"</code> : <br>
 When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></li>
</ul></dd>
<dt>round_digits</dt>
<dd><p>Number of digits to round to when printing percentages.</p></dd>
<dt>return_report</dt>
<dd><p>Return just the ortholog mapping
between two species (<code>FALSE</code>) or return both the
 ortholog mapping as well a <code>data.frame</code>
 of the report statistics (<code>TRUE</code>).</p></dd>
<dt>mc.cores</dt>
<dd><p>Number of cores to parallelise each
<code>target_species</code> with.</p></dd>
<dt>verbose</dt>
<dd><p>Print messages.</p></dd>
<dt>...</dt>
<dd><p>Additional arguments to be passed to
<a href="https://rdrr.io/pkg/gprofiler2/man/gorth.html" class="external-link">gorth</a> or <a href="https://rdrr.io/pkg/homologene/man/homologene.html" class="external-link">homologene</a>.<br><br><em>NOTE</em>: To return only the most "popular"
interspecies ortholog mappings,
supply <code>mthreshold=1</code> here AND set <code>method="gprofiler"</code> above.
This procedure tends to yield a greater number of returned genes but at
 the cost of many of them not being true biological 1:1 orthologs.<br><br>
 For more details, please see
<a href="https://cran.r-project.org/web/packages/gprofiler2/vignettes/gprofiler2.html" class="external-link">
 here</a>.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list containing:</p><ul><li><p>map : A table of inter-species gene mappings.</p></li>
<li><p>report : A list of aggregate orthology report statistics.</p></li>
</ul><p>If &gt;1 <code>target_species</code> are provided, then a table of 
aggregated <code>report</code> statistics concatenated across species 
will be returned instead.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">orth_fly</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu">report_orthologs</span><span class="op">(</span></span>
<span class="r-in">    target_species <span class="op">=</span> <span class="st">"fly"</span>,</span>
<span class="r-in">    reference_species <span class="op">=</span> <span class="st">"human"</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7227</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 8,438 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 8,438 genes from fly.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all genes using: homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gene table with 19,129 rows retrieved.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Returning all 19,129 genes from human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> data.frame format detected.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from Gene.Symbol.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 8,438 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting fly ==&gt; human orthologs using: homologene</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for fly</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 7227</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Retrieving all organisms available in homologene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping species name: human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Common name mapping found for human</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 organism identified from search: 9606</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without orthologs in human.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4,500 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 1 NAs of all kinds from input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting genes from ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4,499 genes extracted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking for genes without 1:1 orthologs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 19 genes that have multiple input_gene per ortholog_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Dropping 266 genes that have multiple ortholog_gene per input_gene.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering gene_df with gene_map</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding input_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ortholog_gene col to gene_df.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes dropped after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    4,380 / 8,438 (52%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Total genes remaining after convert_orthologs :</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    4,058 / 8,438 (48%)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> =========== REPORT SUMMARY ===========</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4,058 / 8,438 (48.09%) target_species genes remain after ortholog conversion.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4,058 / 19,129 (21.21%) reference_species genes remain after ortholog conversion.</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

