<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>benchmarks • orthogene</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="benchmarks">
<meta property="og:description" content="orthogene">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/benchmarks.html">benchmarks</a>
    </li>
    <li>
      <a href="../articles/hexSticker.html">hexSticker</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/orthogene/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="benchmarks_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>benchmarks</h1>
                        <h4 class="author"><h4>
Author: <i>Brian M. Schilder</i>
</h4></h4>
            
            <h4 class="date"><h4>
Most recent update: <i>Aug-01-2021</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/master/vignettes/benchmarks.Rmd"><code>vignettes/benchmarks.Rmd</code></a></small>
      <div class="hidden name"><code>benchmarks.Rmd</code></div>

    </div>

    
    
<p>Benchmark the following strategies:</p>
<ol style="list-style-type: decimal">
<li>
<strong>uppercase</strong> : Simply making the genes uppercase.</li>
<li>
<strong>homologene</strong> : <code>method="homologene"</code>
</li>
<li>
<strong>gorth</strong> : <code>method="gorth"</code>
</li>
</ol>
<p>For each method, benchmark the following metrics:</p>
<ul>
<li>% genes mapped within species (<code>all_genes</code>).</li>
<li>% genes mapped to human (<code>map_orthologs</code>).<br>
</li>
<li>Speed of each metric.</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/orthogene">orthogene</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span> 
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span></code></pre></div>
<div id="define-species" class="section level1">
<h1 class="hasAnchor">
<a href="#define-species" class="anchor"></a>Define species</h1>
<p>Repeat tests across various common model organisms.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>human<span class="op">=</span><span class="st">"H sapiens"</span>,
             chimp<span class="op">=</span><span class="st">"P troglodytes"</span>,
             baboon<span class="op">=</span><span class="st">"P anubis"</span>, 
             macaque <span class="op">=</span> <span class="st">"M mulatta"</span>,
             marmoset <span class="op">=</span> <span class="st">"C jacchus"</span>,
             mouse <span class="op">=</span> <span class="st">"M musculus"</span>,
             rat <span class="op">=</span> <span class="st">"R norvegicus"</span>,
             hamster <span class="op">=</span> <span class="st">"M auratus"</span>,
             dog <span class="op">=</span> <span class="st">"C lupus familiaris"</span>,
             cat <span class="op">=</span> <span class="st">"F catus"</span>,
             cow <span class="op">=</span> <span class="st">"B taurus"</span>,
             chicken <span class="op">=</span> <span class="st">"G gallus"</span>,
             zebrafish <span class="op">=</span> <span class="st">"D rerio"</span>,
             fly <span class="op">=</span> <span class="st">"D melanogaster"</span>,
             worm <span class="op">=</span> <span class="st">"C elegans"</span>,
             rice <span class="op">=</span> <span class="st">"O sativa"</span>
             <span class="op">)</span>
<span class="va">species_mapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/map_species.html">map_species</a></span><span class="op">(</span>species <span class="op">=</span> <span class="va">species</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">`names&lt;-`</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">species</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: H sapiens</code></pre>
<pre><code>## 1 organism identified from search: hsapiens</code></pre>
<pre><code>## Mapping species name: P troglodytes</code></pre>
<pre><code>## 1 organism identified from search: ptroglodytes</code></pre>
<pre><code>## Mapping species name: P anubis</code></pre>
<pre><code>## 1 organism identified from search: panubis</code></pre>
<pre><code>## Mapping species name: M mulatta</code></pre>
<pre><code>## 1 organism identified from search: mmulatta</code></pre>
<pre><code>## Mapping species name: C jacchus</code></pre>
<pre><code>## 1 organism identified from search: cjacchus</code></pre>
<pre><code>## Mapping species name: M musculus</code></pre>
<pre><code>## 1 organism identified from search: mmusculus</code></pre>
<pre><code>## Mapping species name: R norvegicus</code></pre>
<pre><code>## 1 organism identified from search: rnorvegicus</code></pre>
<pre><code>## Mapping species name: M auratus</code></pre>
<pre><code>## 1 organism identified from search: mauratus</code></pre>
<pre><code>## Mapping species name: C lupus familiaris</code></pre>
<pre><code>## 1 organism identified from search: clfamiliaris</code></pre>
<pre><code>## Mapping species name: F catus</code></pre>
<pre><code>## 1 organism identified from search: fcatus</code></pre>
<pre><code>## Mapping species name: B taurus</code></pre>
<pre><code>## 1 organism identified from search: btaurus</code></pre>
<pre><code>## Mapping species name: G gallus</code></pre>
<pre><code>## 1 organism identified from search: ggallus</code></pre>
<pre><code>## Mapping species name: D rerio</code></pre>
<pre><code>## 1 organism identified from search: drerio</code></pre>
<pre><code>## Mapping species name: D melanogaster</code></pre>
<pre><code>## 1 organism identified from search: dmelanogaster</code></pre>
<pre><code>## Mapping species name: C elegans</code></pre>
<pre><code>## 1 organism identified from search: celegans</code></pre>
<pre><code>## Mapping species name: O sativa</code></pre>
<pre><code>## 1 organism identified from search: osativa</code></pre>
</div>
<div id="benchmark" class="section level1">
<h1 class="hasAnchor">
<a href="#benchmark" class="anchor"></a>Benchmark</h1>
<p><em>Note</em> the <code>orthogene:::function</code> notation is needed to use these benchmarking functions, as they are internal.</p>
<div id="run-benchmark" class="section level2">
<h2 class="hasAnchor">
<a href="#run-benchmark" class="anchor"></a>Run benchmark</h2>
<p><code><a href="../reference/run_benchmark.html">run_benchmark()</a></code> will run the full benchmarking pipeline.</p>
<p>You can set <code>mc.core</code> to speed this up with multi-core parallelisation.</p>
<p><strong>WARNING</strong>: This step can take a long time. For the purposes of this example, we’ll not run the full benchmark and instead provide some pre-computed results.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bench_res</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">:::</span><span class="fu"><a href="../reference/run_benchmark.html">run_benchmark</a></span><span class="op">(</span>species_mapped <span class="op">=</span> <span class="va">species_mapped</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"human"</span>,<span class="st">"mouse"</span>,<span class="st">"fly"</span><span class="op">)</span><span class="op">]</span>, 
                                       run_convert_orthologs <span class="op">=</span> <span class="cn">TRUE</span>, 
                                       mc.core <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> 
<span class="co"># write.csv(bench_res, here::here("inst/benchmark/bench_res_example.csv"), row.names = FALSE)</span></code></pre></div>
</div>
<div id="print-results" class="section level2">
<h2 class="hasAnchor">
<a href="#print-results" class="anchor"></a>Print results</h2>
<p>Load stored benchmark results.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html">exists</a></span><span class="op">(</span><span class="st">"bench_res"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">bench_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"orthogene"</span>,<span class="st">"benchmark/bench_res_example.csv"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">bench_res</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">species</th>
<th align="left">method</th>
<th align="left">test</th>
<th align="right">time</th>
<th align="right">genes</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">hsapiens</td>
<td align="left">homologene</td>
<td align="left">all_genes()</td>
<td align="right">0.4597051</td>
<td align="right">19129</td>
</tr>
<tr class="even">
<td align="left">hsapiens</td>
<td align="left">homologene</td>
<td align="left">convert_orthologs()</td>
<td align="right">0.7699881</td>
<td align="right">18713</td>
</tr>
<tr class="odd">
<td align="left">hsapiens</td>
<td align="left">gprofiler</td>
<td align="left">all_genes()</td>
<td align="right">8.6186595</td>
<td align="right">39357</td>
</tr>
<tr class="even">
<td align="left">hsapiens</td>
<td align="left">gprofiler</td>
<td align="left">convert_orthologs()</td>
<td align="right">134.0899084</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">mmusculus</td>
<td align="left">homologene</td>
<td align="left">all_genes()</td>
<td align="right">0.4631681</td>
<td align="right">21207</td>
</tr>
<tr class="even">
<td align="left">mmusculus</td>
<td align="left">homologene</td>
<td align="left">convert_orthologs()</td>
<td align="right">0.8222713</td>
<td align="right">16482</td>
</tr>
<tr class="odd">
<td align="left">mmusculus</td>
<td align="left">gprofiler</td>
<td align="left">all_genes()</td>
<td align="right">10.9749143</td>
<td align="right">53725</td>
</tr>
<tr class="even">
<td align="left">mmusculus</td>
<td align="left">gprofiler</td>
<td align="left">convert_orthologs()</td>
<td align="right">146.1204503</td>
<td align="right">16189</td>
</tr>
<tr class="odd">
<td align="left">dmelanogaster</td>
<td align="left">homologene</td>
<td align="left">all_genes()</td>
<td align="right">0.4492853</td>
<td align="right">8437</td>
</tr>
<tr class="even">
<td align="left">dmelanogaster</td>
<td align="left">homologene</td>
<td align="left">convert_orthologs()</td>
<td align="right">0.7071683</td>
<td align="right">4059</td>
</tr>
<tr class="odd">
<td align="left">dmelanogaster</td>
<td align="left">gprofiler</td>
<td align="left">all_genes()</td>
<td align="right">6.1289411</td>
<td align="right">4493</td>
</tr>
<tr class="even">
<td align="left">dmelanogaster</td>
<td align="left">gprofiler</td>
<td align="left">convert_orthologs()</td>
<td align="right">5.2325366</td>
<td align="right">638</td>
</tr>
</tbody>
</table>
</div>
<div id="plot-results" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-results" class="anchor"></a>Plot results</h2>
<div id="bar-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#bar-plot" class="anchor"></a>Bar plot</h3>
<p>For each method, plot the run time (a) and the number of genes returned (b).</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bench_barplot</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">:::</span><span class="fu"><a href="../reference/plot_benchmark_bar.html">plot_benchmark_bar</a></span><span class="op">(</span>bench_res <span class="op">=</span> <span class="va">bench_res</span><span class="op">)</span></code></pre></div>
<p><img src="benchmarks_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ggsave(here::here("inst/benchmark/bench_barplot.pdf"),bench_barplot, height = 8)</span></code></pre></div>
</div>
</div>
<div id="scatter-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#scatter-plot" class="anchor"></a>Scatter plot</h2>
<p>For each method, plot the relationship between number of genes returned and run time.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bench_scatterplot</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">:::</span><span class="fu"><a href="../reference/plot_benchmark_scatter.html">plot_benchmark_scatter</a></span><span class="op">(</span>bench_res <span class="op">=</span> <span class="va">bench_res</span><span class="op">)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula 'y ~ x'</code></pre>
<p><img src="benchmarks_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ggsave(here::here("inst/benchmark/bench_scatterplot.pdf"),bench_scatterplot)</span></code></pre></div>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<details><div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] patchwork_1.1.1 ggplot2_3.3.5   dplyr_1.0.7     orthogene_0.1.0
## 
## loaded via a namespace (and not attached):
##  [1] httr_1.4.2                sass_0.4.0               
##  [3] tidyr_1.1.3               splines_4.1.0            
##  [5] jsonlite_1.7.2            viridisLite_0.4.0        
##  [7] carData_3.0-4             here_1.0.1               
##  [9] gprofiler2_0.2.0          bslib_0.2.5.1            
## [11] assertthat_0.2.1          highr_0.9                
## [13] cellranger_1.1.0          yaml_2.2.1               
## [15] pillar_1.6.1              backports_1.2.1          
## [17] lattice_0.20-44           glue_1.4.2               
## [19] digest_0.6.27             ggsignif_0.6.2           
## [21] colorspace_2.0-2          htmltools_0.5.1.1        
## [23] Matrix_1.3-4              pkgconfig_2.0.3          
## [25] broom_0.7.8               haven_2.4.1              
## [27] purrr_0.3.4               scales_1.1.1             
## [29] openxlsx_4.2.4            rio_0.5.27               
## [31] tibble_3.1.3              mgcv_1.8-36              
## [33] farver_2.1.0              generics_0.1.0           
## [35] car_3.0-11                ellipsis_0.3.2           
## [37] ggpubr_0.4.0              withr_2.4.2              
## [39] cachem_1.0.5              lazyeval_0.2.2           
## [41] magrittr_2.0.1            crayon_1.4.1             
## [43] readxl_1.3.1              memoise_2.0.0            
## [45] evaluate_0.14             fs_1.5.0                 
## [47] fansi_0.5.0               nlme_3.1-152             
## [49] rstatix_0.7.0             homologene_1.4.68.19.3.27
## [51] forcats_0.5.1             foreign_0.8-81           
## [53] textshaping_0.3.5         tools_4.1.0              
## [55] data.table_1.14.0         hms_1.1.0                
## [57] lifecycle_1.0.0           stringr_1.4.0            
## [59] plotly_4.9.4.9000         munsell_0.5.0            
## [61] zip_2.2.0                 compiler_4.1.0           
## [63] pkgdown_1.6.1             jquerylib_0.1.4          
## [65] systemfonts_1.0.2         rlang_0.4.11             
## [67] grid_4.1.0                htmlwidgets_1.5.3        
## [69] labeling_0.4.2            rmarkdown_2.9            
## [71] gtable_0.3.0              abind_1.4-5              
## [73] DBI_1.1.1                 curl_4.3.2               
## [75] R6_2.5.0                  knitr_1.33               
## [77] fastmap_1.1.0             utf8_1.2.2               
## [79] rprojroot_2.0.2           ragg_1.1.3               
## [81] desc_1.3.0                stringi_1.7.3            
## [83] parallel_4.1.0            Rcpp_1.0.7               
## [85] vctrs_0.3.8               tidyselect_1.1.1         
## [87] xfun_0.24</code></pre>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
