<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Create gene background — create_background • orthogene</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Create gene background — create_background" />
<meta property="og:description" content="Create a gene background as the union/intersect of 
all orthologs between input species (species1 and species2), 
and the output_species. 
This can be useful when generating random lists of background genes 
to test against in analyses with data from multiple species
 (e.g. enrichment of mouse cell-type markers gene sets in
  human GWAS-derived gene sets)." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/neurogenomics/orthogene/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create gene background</h1>
    <small class="dont-index">Source: <a href='https://github.com/neurogenomics/orthogene/blob/master/R/create_background.R'><code>R/create_background.R</code></a></small>
    <div class="hidden name"><code>create_background.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Create a gene background as the union/intersect of 
all orthologs between input species (<code>species1</code> and <code>species2</code>), 
and the <code>output_species</code>. 
This can be useful when generating random lists of background genes 
to test against in analyses with data from multiple species
 (e.g. enrichment of mouse cell-type markers gene sets in
  human GWAS-derived gene sets).</p>
    </div>

    <pre class="usage"><span class='fu'>create_background</span><span class='op'>(</span>
  <span class='va'>species1</span>,
  <span class='va'>species2</span>,
  output_species <span class='op'>=</span> <span class='st'>"human"</span>,
  as_output_species <span class='op'>=</span> <span class='cn'>TRUE</span>,
  use_intersect <span class='op'>=</span> <span class='cn'>TRUE</span>,
  bg <span class='op'>=</span> <span class='cn'>NULL</span>,
  gene_map <span class='op'>=</span> <span class='cn'>NULL</span>,
  method <span class='op'>=</span> <span class='st'>"homologene"</span>,
  non121_strategy <span class='op'>=</span> <span class='st'>"drop_both_species"</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>species1</th>
      <td><p>First species.</p></td>
    </tr>
    <tr>
      <th>species2</th>
      <td><p>Second species.</p></td>
    </tr>
    <tr>
      <th>output_species</th>
      <td><p>Species to convert all genes from 
<code>species1</code> and <code>species2</code> to first.
 <code>Default="human"</code>, but can be to either any species
  supported by <span class="pkg">orthogene</span>, including 
 <code>species1</code> or <code>species2</code>.</p></td>
    </tr>
    <tr>
      <th>as_output_species</th>
      <td><p>Return background gene list as 
<code>output_species</code> orthologs, instead of the
 gene names of the original input species.</p></td>
    </tr>
    <tr>
      <th>use_intersect</th>
      <td><p>When <code>species1</code> and <code>species2</code> are both
different from <code>output_species</code>, this argument will determine whether 
to use the intersect (<code>TRUE</code>) or union (<code>FALSE</code>) of all genes
from <code>species1</code> and <code>species2</code>.</p></td>
    </tr>
    <tr>
      <th>bg</th>
      <td><p>User supplied background list that will be returned to the 
user after removing duplicate genes.</p></td>
    </tr>
    <tr>
      <th>gene_map</th>
      <td><p>User-supplied <code>gene_map</code> data table from
<a href='map_orthologs.html'>map_orthologs</a> or <a href='map_genes.html'>map_genes</a>.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>R package to to use for gene mapping:</p><ul>
<li><p><code>"gprofiler"</code> : Slower but more species and genes.</p></li>
<li><p><code>"homologene"</code> : Faster but fewer species and genes.</p></li>
<li><p><code>"babelgene"</code> : Faster but fewer species and genes.
Also gives</p></li>
</ul><p>(slower but more species and genes) or
<code>"homologene"</code> (faster but fewer species and genes).</p></td>
    </tr>
    <tr>
      <th>non121_strategy</th>
      <td><p>How to handle genes that don't have
1:1 mappings between <code>input_species</code>:<code>output_species</code>.
Options include:<br /></p><ul>
<li><p><code>"drop_both_species" or "dbs" or 1</code> : <br />
Drop genes that have duplicate
mappings in either the <code>input_species</code> or <code>output_species</code> <br />
(<em>DEFAULT</em>).</p></li>
<li><p><code>"drop_input_species" or "dis" or 2</code> : <br />
Only drop genes that have duplicate
mappings in the <code>input_species</code>.</p></li>
<li><p><code>"drop_output_species" or "dos" or 3</code> : <br />
Only drop genes that have duplicate
mappings in the <code>output_species</code>.</p></li>
<li><p><code>"keep_both_species" or "kbs" or 4</code> : <br />
Keep all genes regardless of whether
they have duplicate mappings in either species.</p></li>
<li><p><code>"keep_popular" or "kp" or 5</code> : <br />
Return only the most "popular" interspecies ortholog mappings.
 This procedure tends to yield a greater number of returned genes
 but at the cost of many of them not being true biological 1:1 orthologs.</p></li>
<li><p><code>"sum","mean","median","min" or "max"</code> : <br />
 When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>,
  these options will aggregate many-to-one gene mappings
  (<code>input_species</code>-to-<code>output_species</code>)
  after dropping any duplicate genes in the <code>output_species</code>.</p></li>
</ul></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Print messages.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Background gene list.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>bg</span> <span class='op'>&lt;-</span> <span class='fu'>orthogene</span><span class='fu'>::</span><span class='fu'>create_background</span><span class='op'>(</span>species1 <span class='op'>=</span> <span class='st'>"mouse"</span>, 
                                   species2 <span class='op'>=</span> <span class='st'>"rat"</span>,
                                   output_species <span class='op'>=</span> <span class='st'>"human"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Generating gene background for mouse x rat ==&gt; human</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: mouse</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for mouse</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: mmusculus</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: mmusculus</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 10090</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 21,207 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from Gene.Symbol.</span></div><div class='output co'>#&gt; <span class='message'>21,207 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 21,207 genes from mmusculus.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from Gene.Symbol.</span></div><div class='output co'>#&gt; <span class='message'>19,129 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from hsapiens.</span></div><div class='output co'>#&gt; <span class='message'>Preparing gene_df.</span></div><div class='output co'>#&gt; <span class='message'>data.frame format detected.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from Gene.Symbol.</span></div><div class='output co'>#&gt; <span class='message'>21,207 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Converting mmusculus ==&gt; hsapiens orthologs using: homologene</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: mmusculus</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 10090</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Checking for genes without orthologs in hsapiens.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from input_gene.</span></div><div class='output co'>#&gt; <span class='message'>17,355 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from ortholog_gene.</span></div><div class='output co'>#&gt; <span class='message'>17,355 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Checking for genes without 1:1 orthologs.</span></div><div class='output co'>#&gt; <span class='message'>Dropping 131 genes that have multiple input_gene per ortholog_gene.</span></div><div class='output co'>#&gt; <span class='message'>Dropping 498 genes that have multiple ortholog_gene per input_gene.</span></div><div class='output co'>#&gt; <span class='message'>Filtering gene_df with gene_map</span></div><div class='output co'>#&gt; <span class='message'>Adding input_gene col to gene_df.</span></div><div class='output co'>#&gt; <span class='message'>Adding ortholog_gene col to gene_df.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>=========== REPORT SUMMARY ===========</span></div><div class='output co'>#&gt; <span class='message'>Total genes dropped after convert_orthologs :</span>
#&gt; <span class='message'>   4,725 / 21,207 (22%)</span></div><div class='output co'>#&gt; <span class='message'>Total genes remaining after convert_orthologs :</span>
#&gt; <span class='message'>   16,482 / 21,207 (78%)</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>=========== REPORT SUMMARY ===========</span></div><div class='output co'>#&gt; <span class='message'>16,482 / 21,207 (77.72%) target_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>16,482 / 19,129 (86.16%) reference_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: rat</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for rat</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: rnorvegicus</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: human</span></div><div class='output co'>#&gt; <span class='message'>Common name mapping found for human</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: rnorvegicus</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 10116</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 20,616 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from Gene.Symbol.</span></div><div class='output co'>#&gt; <span class='message'>20,616 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 20,616 genes from rnorvegicus.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all genes using: homologene.</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Gene table with 19,129 rows retrieved.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from Gene.Symbol.</span></div><div class='output co'>#&gt; <span class='message'>19,129 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Returning all 19,129 genes from hsapiens.</span></div><div class='output co'>#&gt; <span class='message'>Preparing gene_df.</span></div><div class='output co'>#&gt; <span class='message'>data.frame format detected.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from Gene.Symbol.</span></div><div class='output co'>#&gt; <span class='message'>20,616 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Converting rnorvegicus ==&gt; hsapiens orthologs using: homologene</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: rnorvegicus</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 10116</span></div><div class='output co'>#&gt; <span class='message'>Retrieving all organisms available in gprofiler.</span></div><div class='output co'>#&gt; <span class='message'>Using stored `gprofiler_orgs`.</span></div><div class='output co'>#&gt; <span class='message'>Mapping species name: hsapiens</span></div><div class='output co'>#&gt; <span class='message'>1 organism identified from search: 9606</span></div><div class='output co'>#&gt; <span class='message'>Checking for genes without orthologs in hsapiens.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from input_gene.</span></div><div class='output co'>#&gt; <span class='message'>16,989 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Extracting genes from ortholog_gene.</span></div><div class='output co'>#&gt; <span class='message'>16,989 genes extracted.</span></div><div class='output co'>#&gt; <span class='message'>Checking for genes without 1:1 orthologs.</span></div><div class='output co'>#&gt; <span class='message'>Dropping 122 genes that have multiple input_gene per ortholog_gene.</span></div><div class='output co'>#&gt; <span class='message'>Dropping 607 genes that have multiple ortholog_gene per input_gene.</span></div><div class='output co'>#&gt; <span class='message'>Filtering gene_df with gene_map</span></div><div class='output co'>#&gt; <span class='message'>Adding input_gene col to gene_df.</span></div><div class='output co'>#&gt; <span class='message'>Adding ortholog_gene col to gene_df.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>=========== REPORT SUMMARY ===========</span></div><div class='output co'>#&gt; <span class='message'>Total genes dropped after convert_orthologs :</span>
#&gt; <span class='message'>   4,813 / 20,616 (23%)</span></div><div class='output co'>#&gt; <span class='message'>Total genes remaining after convert_orthologs :</span>
#&gt; <span class='message'>   15,803 / 20,616 (77%)</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>=========== REPORT SUMMARY ===========</span></div><div class='output co'>#&gt; <span class='message'>15,803 / 20,616 (76.65%) target_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>15,803 / 19,129 (82.61%) reference_species genes remain after ortholog conversion.</span></div><div class='output co'>#&gt; <span class='message'>15,450 intersect background genes used.</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


