<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Infer species • orthogene</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Infer species">
<meta property="og:description" content="orthogene">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/docker.html">Docker</a>
    </li>
    <li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Infer species</h1>
                        <h4 data-toc-skip class="author"><h4>
Author: <i>Brian M. Schilder</i>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Most recent update: <i>Oct-29-2021</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/HEAD/vignettes/infer_species.Rmd" class="external-link"><code>vignettes/infer_species.Rmd</code></a></small>
      <div class="hidden name"><code>infer_species.Rmd</code></div>

    </div>

    
    
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span> 
<span class="co"># orthogene is only available on Bioconductor&gt;=3.14</span>
<span class="kw">if</span><span class="op">(</span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/version.html" class="external-link">version</a></span><span class="op">(</span><span class="op">)</span><span class="op">&lt;</span><span class="st">"3.14"</span><span class="op">)</span> 
  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span>update <span class="op">=</span> <span class="cn">TRUE</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"orthogene"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/orthogene" class="external-link">orthogene</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"exp_mouse"</span><span class="op">)</span>
<span class="co"># Setting to "homologene" for the purposes of quick demonstration.</span>
<span class="co"># We generally recommend using method="gprofiler" (default).</span>
<span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"homologene"</span>  </code></pre></div>
</div>
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h1>
<p>It’s not always clear whether a dataset is using the original species gene names, human gene names, or some other species’ gene names.</p>
<p><code>infer_species</code> takes a list/matrix/data.frame with genes and infers the species that they best match to!</p>
<p>For the sake of speed, the genes extracted from <code>gene_df</code> are tested against genomes from only the following 6 <code>test_species</code> by default: - human - monkey - rat - mouse - zebrafish - fly</p>
<p>However, you can supply your own list of <code>test_species</code>, which will be automatically be mapped and standardised using <code>map_species</code>.</p>
</div>
<div class="section level1">
<h1 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h1>
<div class="section level2">
<h2 id="mouse-genes">Mouse genes<a class="anchor" aria-label="anchor" href="#mouse-genes"></a>
</h2>
<div class="section level3">
<h3 id="infer-the-species">Infer the species<a class="anchor" aria-label="anchor" href="#infer-the-species"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">matches</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/infer_species.html">infer_species</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse</span>, 
                                    method <span class="op">=</span> <span class="va">method</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## sparseMatrix format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from rownames.</span></code></pre>
<pre><code><span class="co">## 15,259 genes extracted.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: human</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: monkey</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: monkey</span></code></pre>
<pre><code><span class="co">## Common name mapping found for monkey</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9544</span></code></pre>
<pre><code><span class="co">## Gene table with 16,843 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 16,843 genes from monkey.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: rat</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: rat</span></code></pre>
<pre><code><span class="co">## Common name mapping found for rat</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10116</span></code></pre>
<pre><code><span class="co">## Gene table with 20,616 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 20,616 genes from rat.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: mouse</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mouse.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: zebrafish</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: zebrafish</span></code></pre>
<pre><code><span class="co">## Common name mapping found for zebrafish</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7955</span></code></pre>
<pre><code><span class="co">## Gene table with 20,897 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 20,897 genes from zebrafish.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: fly</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: fly</span></code></pre>
<pre><code><span class="co">## Common name mapping found for fly</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7227</span></code></pre>
<pre><code><span class="co">## Gene table with 8,438 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 8,438 genes from fly.</span></code></pre>
<pre><code><span class="co">## Top match:</span>
<span class="co">##   - species: mouse </span>
<span class="co">##   - percent_match: 92%</span></code></pre>
<pre><code><span class="co">## Loading required namespace: hrbrthemes</span></code></pre>
<pre><code><span class="co">## You will likely need to install these fonts on your system as well.</span>
<span class="co">## </span>
<span class="co">## You can find them in [/__w/_temp/Library/hrbrthemes/fonts/roboto-condensed]</span></code></pre>
<p><img src="infer_species_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="rat-genes">Rat genes<a class="anchor" aria-label="anchor" href="#rat-genes"></a>
</h2>
<div class="section level3">
<h3 id="create-example-data">Create example data<a class="anchor" aria-label="anchor" href="#create-example-data"></a>
</h3>
<p>To create an example dataset, turn the gene names into rat genes.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_rat</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/convert_orthologs.html">convert_orthologs</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse</span>, 
                                        input_species <span class="op">=</span> <span class="st">"mouse"</span>, 
                                        output_species <span class="op">=</span> <span class="st">"rat"</span>,
                                        method <span class="op">=</span> <span class="va">method</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## sparseMatrix format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from rownames.</span></code></pre>
<pre><code><span class="co">## 15,259 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mouse ==&gt; rat orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: rat</span></code></pre>
<pre><code><span class="co">## Common name mapping found for rat</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10116</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in rat.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 13,812 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 13,812 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 486 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 148 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Setting ortholog_gene to rownames.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    2,322 / 15,259 (15%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    12,937 / 15,259 (85%)</span></code></pre>
</div>
<div class="section level3">
<h3 id="infer-the-species-1">Infer the species<a class="anchor" aria-label="anchor" href="#infer-the-species-1"></a>
</h3>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">matches</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/infer_species.html">infer_species</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_rat</span>, 
                                    method <span class="op">=</span> <span class="va">method</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## sparseMatrix format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from rownames.</span></code></pre>
<pre><code><span class="co">## 12,937 genes extracted.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: human</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: monkey</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: monkey</span></code></pre>
<pre><code><span class="co">## Common name mapping found for monkey</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9544</span></code></pre>
<pre><code><span class="co">## Gene table with 16,843 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 16,843 genes from monkey.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: rat</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: rat</span></code></pre>
<pre><code><span class="co">## Common name mapping found for rat</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10116</span></code></pre>
<pre><code><span class="co">## Gene table with 20,616 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 20,616 genes from rat.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: mouse</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mouse.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: zebrafish</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: zebrafish</span></code></pre>
<pre><code><span class="co">## Common name mapping found for zebrafish</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7955</span></code></pre>
<pre><code><span class="co">## Gene table with 20,897 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 20,897 genes from zebrafish.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: fly</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: fly</span></code></pre>
<pre><code><span class="co">## Common name mapping found for fly</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7227</span></code></pre>
<pre><code><span class="co">## Gene table with 8,438 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 8,438 genes from fly.</span></code></pre>
<pre><code><span class="co">## Top match:</span>
<span class="co">##   - species: rat </span>
<span class="co">##   - percent_match: 100%</span></code></pre>
<pre><code><span class="co">## You will likely need to install these fonts on your system as well.</span>
<span class="co">## </span>
<span class="co">## You can find them in [/__w/_temp/Library/hrbrthemes/fonts/roboto-condensed]</span></code></pre>
<p><img src="infer_species_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="human-genes">Human genes<a class="anchor" aria-label="anchor" href="#human-genes"></a>
</h2>
<div class="section level3">
<h3 id="create-example-data-1">Create example data<a class="anchor" aria-label="anchor" href="#create-example-data-1"></a>
</h3>
<p>To create an example dataset, turn the gene names into human genes.</p>
<div class="sourceCode" id="cb155"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_human</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/convert_orthologs.html">convert_orthologs</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse</span>, 
                                          input_species <span class="op">=</span> <span class="st">"mouse"</span>, 
                                          output_species <span class="op">=</span> <span class="st">"human"</span>,
                                          method <span class="op">=</span> <span class="va">method</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## sparseMatrix format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from rownames.</span></code></pre>
<pre><code><span class="co">## 15,259 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in human.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 13,416 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 13,416 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 46 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 56 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Setting ortholog_gene to rownames.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    2,016 / 15,259 (13%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    13,243 / 15,259 (87%)</span></code></pre>
</div>
<div class="section level3">
<h3 id="infer-the-species-2">Infer the species<a class="anchor" aria-label="anchor" href="#infer-the-species-2"></a>
</h3>
<div class="sourceCode" id="cb184"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">matches</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/infer_species.html">infer_species</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_human</span>, 
                                    method <span class="op">=</span> <span class="va">method</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## sparseMatrix format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from rownames.</span></code></pre>
<pre><code><span class="co">## 13,243 genes extracted.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: human</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from human.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: monkey</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: monkey</span></code></pre>
<pre><code><span class="co">## Common name mapping found for monkey</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9544</span></code></pre>
<pre><code><span class="co">## Gene table with 16,843 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 16,843 genes from monkey.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: rat</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: rat</span></code></pre>
<pre><code><span class="co">## Common name mapping found for rat</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10116</span></code></pre>
<pre><code><span class="co">## Gene table with 20,616 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 20,616 genes from rat.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: mouse</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mouse.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: zebrafish</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: zebrafish</span></code></pre>
<pre><code><span class="co">## Common name mapping found for zebrafish</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7955</span></code></pre>
<pre><code><span class="co">## Gene table with 20,897 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 20,897 genes from zebrafish.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: fly</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: fly</span></code></pre>
<pre><code><span class="co">## Common name mapping found for fly</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7227</span></code></pre>
<pre><code><span class="co">## Gene table with 8,438 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 8,438 genes from fly.</span></code></pre>
<pre><code><span class="co">## Top match:</span>
<span class="co">##   - species: human </span>
<span class="co">##   - percent_match: 100%</span></code></pre>
<pre><code><span class="co">## You will likely need to install these fonts on your system as well.</span>
<span class="co">## </span>
<span class="co">## You can find them in [/__w/_temp/Library/hrbrthemes/fonts/roboto-condensed]</span></code></pre>
<p><img src="infer_species_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="additional-test_species">Additional <code>test_species</code><a class="anchor" aria-label="anchor" href="#additional-test_species"></a>
</h1>
<p>You can even supply <code>test_species</code> with the name of one of the R packages that <code>orthogene</code> gets orthologs from. This will test against all species available in that particular R package.</p>
<p>For example, by setting <code>test_species="homologene"</code> we automatically test for % gene matches in each of the 20+ species available in <code>homologene</code>.</p>
<div class="sourceCode" id="cb245"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">matches</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/infer_species.html">infer_species</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_human</span>, 
                                    test_species <span class="op">=</span> <span class="va">method</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Retrieving all organisms available in homologene.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## sparseMatrix format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from rownames.</span></code></pre>
<pre><code><span class="co">## 13,243 genes extracted.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Mus musculus</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Mus musculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from Mus musculus.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Rattus norvegicus</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Rattus norvegicus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10116</span></code></pre>
<pre><code><span class="co">## Gene table with 20,616 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 20,616 genes from Rattus norvegicus.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Kluyveromyces lactis</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Kluyveromyces lactis</span></code></pre>
<pre><code><span class="co">## WARNING: No organisms identified matched  'kluyveromyces lactis' Try a different query.</span></code></pre>
<pre><code><span class="co">## Gene table with 0 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 0 genes from Kluyveromyces lactis.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Magnaporthe oryzae</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Magnaporthe oryzae</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 242507</span></code></pre>
<pre><code><span class="co">## Gene table with 0 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 0 genes from Magnaporthe oryzae.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Eremothecium gossypii</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Eremothecium gossypii</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 284811</span></code></pre>
<pre><code><span class="co">## Gene table with 0 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 0 genes from Eremothecium gossypii.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Arabidopsis thaliana</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Arabidopsis thaliana</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 3702</span></code></pre>
<pre><code><span class="co">## Gene table with 19,143 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,143 genes from Arabidopsis thaliana.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Oryza sativa</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Oryza sativa</span></code></pre>
<pre><code><span class="co">## 2 organisms identified from search.</span>
<span class="co">## Selecting first:</span>
<span class="co">##    - 39946</span>
<span class="co">##    - 39947</span></code></pre>
<pre><code><span class="co">## Gene table with 0 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 0 genes from Oryza sativa.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Schizosaccharomyces pombe</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Schizosaccharomyces pombe</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 284812</span></code></pre>
<pre><code><span class="co">## Gene table with 0 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 0 genes from Schizosaccharomyces pombe.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Saccharomyces cerevisiae</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Saccharomyces cerevisiae</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 4932</span></code></pre>
<pre><code><span class="co">## Gene table with 4,579 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 4,579 genes from Saccharomyces cerevisiae.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Neurospora crassa</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Neurospora crassa</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 367110</span></code></pre>
<pre><code><span class="co">## Gene table with 0 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 0 genes from Neurospora crassa.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Caenorhabditis elegans</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Caenorhabditis elegans</span></code></pre>
<pre><code><span class="co">## 2 organisms identified from search.</span>
<span class="co">## Selecting first:</span>
<span class="co">##    - 6239</span>
<span class="co">##    - 6239</span></code></pre>
<pre><code><span class="co">## Gene table with 7,575 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 7,575 genes from Caenorhabditis elegans.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Anopheles gambiae</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Anopheles gambiae</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7165</span></code></pre>
<pre><code><span class="co">## Gene table with 8,428 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 8,428 genes from Anopheles gambiae.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Drosophila melanogaster</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Drosophila melanogaster</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7227</span></code></pre>
<pre><code><span class="co">## Gene table with 8,438 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 8,438 genes from Drosophila melanogaster.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Danio rerio</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Danio rerio</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7955</span></code></pre>
<pre><code><span class="co">## Gene table with 20,897 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 20,897 genes from Danio rerio.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Xenopus (Silurana) tropicalis</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Xenopus (Silurana) tropicalis</span></code></pre>
<pre><code><span class="co">## Warning in grepl(spec_queries, display_name, ignore.case = TRUE): TRE pattern</span>
<span class="co">## compilation error 'Missing ')''</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Gallus gallus</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Gallus gallus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9031</span></code></pre>
<pre><code><span class="co">## Gene table with 14,600 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 14,600 genes from Gallus gallus.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Macaca mulatta</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Macaca mulatta</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9544</span></code></pre>
<pre><code><span class="co">## Gene table with 16,843 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 16,843 genes from Macaca mulatta.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Pan troglodytes</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Pan troglodytes</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9598</span></code></pre>
<pre><code><span class="co">## Gene table with 18,730 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 18,730 genes from Pan troglodytes.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Homo sapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Homo sapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from Homo sapiens.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Canis lupus familiaris</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Canis lupus familiaris</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9615</span></code></pre>
<pre><code><span class="co">## Gene table with 18,117 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 18,117 genes from Canis lupus familiaris.</span></code></pre>
<pre><code><span class="co">## Testing for gene overlap with: Bos taurus</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: Bos taurus</span></code></pre>
<pre><code><span class="co">## 2 organisms identified from search.</span>
<span class="co">## Selecting first:</span>
<span class="co">##    - 30522</span>
<span class="co">##    - 9913</span></code></pre>
<pre><code><span class="co">## Gene table with 0 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 0 genes from Bos taurus.</span></code></pre>
<pre><code><span class="co">## Top match:</span>
<span class="co">##   - species: Homo sapiens </span>
<span class="co">##   - percent_match: 100%</span></code></pre>
<pre><code><span class="co">## You will likely need to install these fonts on your system as well.</span>
<span class="co">## </span>
<span class="co">## You can find them in [/__w/_temp/Library/hrbrthemes/fonts/roboto-condensed]</span></code></pre>
<p><img src="infer_species_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div class="section level1">
<h1 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h1>
<details><div class="sourceCode" id="cb419"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.1 (2021-08-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] orthogene_1.1.2  BiocStyle_2.22.0</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] fs_1.5.0                  httr_1.4.2               </span>
<span class="co">##  [3] rprojroot_2.0.2           tools_4.1.1              </span>
<span class="co">##  [5] backports_1.3.0           bslib_0.3.1              </span>
<span class="co">##  [7] utf8_1.2.2                R6_2.5.1                 </span>
<span class="co">##  [9] lazyeval_0.2.2            colorspace_2.0-2         </span>
<span class="co">## [11] tidyselect_1.1.1          curl_4.3.2               </span>
<span class="co">## [13] compiler_4.1.1            extrafontdb_1.0          </span>
<span class="co">## [15] textshaping_0.3.6         cli_3.1.0                </span>
<span class="co">## [17] desc_1.4.0                plotly_4.10.0            </span>
<span class="co">## [19] labeling_0.4.2            bookdown_0.24            </span>
<span class="co">## [21] sass_0.4.0                scales_1.1.1             </span>
<span class="co">## [23] pkgdown_1.9000.9000.9000  systemfonts_1.0.3        </span>
<span class="co">## [25] stringr_1.4.0             digest_0.6.28            </span>
<span class="co">## [27] foreign_0.8-81            rmarkdown_2.11           </span>
<span class="co">## [29] rio_0.5.27                pkgconfig_2.0.3          </span>
<span class="co">## [31] htmltools_0.5.2           extrafont_0.17           </span>
<span class="co">## [33] highr_0.9                 fastmap_1.1.0            </span>
<span class="co">## [35] htmlwidgets_1.5.4         rlang_0.99.0.9000        </span>
<span class="co">## [37] readxl_1.3.1              farver_2.1.0             </span>
<span class="co">## [39] jquerylib_0.1.4           generics_0.1.1           </span>
<span class="co">## [41] jsonlite_1.7.2            dplyr_1.0.7              </span>
<span class="co">## [43] zip_2.2.0                 car_3.0-11               </span>
<span class="co">## [45] homologene_1.4.68.19.3.27 magrittr_2.0.1           </span>
<span class="co">## [47] GenomeInfoDbData_1.2.7    patchwork_1.1.1          </span>
<span class="co">## [49] Matrix_1.3-4              Rcpp_1.0.7               </span>
<span class="co">## [51] munsell_0.5.0             fansi_0.5.0              </span>
<span class="co">## [53] gdtools_0.2.3             abind_1.4-5              </span>
<span class="co">## [55] babelgene_21.4            lifecycle_1.0.1          </span>
<span class="co">## [57] stringi_1.7.5             yaml_2.2.1               </span>
<span class="co">## [59] carData_3.0-4             grid_4.1.1               </span>
<span class="co">## [61] hrbrthemes_0.8.0          parallel_4.1.1           </span>
<span class="co">## [63] forcats_0.5.1             crayon_1.4.1             </span>
<span class="co">## [65] lattice_0.20-45           haven_2.4.3              </span>
<span class="co">## [67] hms_1.1.1                 knitr_1.36               </span>
<span class="co">## [69] pillar_1.6.4              ggpubr_0.4.0             </span>
<span class="co">## [71] ggsignif_0.6.3            glue_1.4.2               </span>
<span class="co">## [73] gprofiler2_0.2.1          evaluate_0.14            </span>
<span class="co">## [75] data.table_1.14.2         BiocManager_1.30.16      </span>
<span class="co">## [77] vctrs_0.3.8               Rttf2pt1_1.3.9           </span>
<span class="co">## [79] cellranger_1.1.0          gtable_0.3.0             </span>
<span class="co">## [81] purrr_0.3.4               tidyr_1.1.4              </span>
<span class="co">## [83] cachem_1.0.6              ggplot2_3.3.5            </span>
<span class="co">## [85] xfun_0.27                 openxlsx_4.2.4           </span>
<span class="co">## [87] broom_0.7.9               rstatix_0.7.0            </span>
<span class="co">## [89] ragg_1.1.3                viridisLite_0.4.0        </span>
<span class="co">## [91] tibble_3.1.5              memoise_2.0.0            </span>
<span class="co">## [93] ellipsis_0.3.2</span></code></pre>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.9000.9000.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
