<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Infer species • orthogene</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Infer species">
<meta property="og:description" content="orthogene">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/orthogene/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="infer_species_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Infer species</h1>
                        <h4 class="author"><h4>
Author: <i>Brian M. Schilder</i>
</h4></h4>
            
            <h4 class="date"><h4>
Most recent update: <i>Oct-23-2021</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/master/vignettes/infer_species.Rmd"><code>vignettes/infer_species.Rmd</code></a></small>
      <div class="hidden name"><code>infer_species.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span> 
<span class="co"># orthogene is only available on Bioconductor&gt;=3.14</span>
<span class="kw">if</span><span class="op">(</span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/version.html">version</a></span><span class="op">(</span><span class="op">)</span><span class="op">&lt;</span><span class="st">"3.14"</span><span class="op">)</span> <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span>version <span class="op">=</span> <span class="st">"devel"</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"orthogene"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/orthogene">orthogene</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"exp_mouse"</span><span class="op">)</span>
<span class="co"># Setting to "homologene" for the purposes of quick demonstration.</span>
<span class="co"># We generally recommend using method="gprofiler" (default).</span>
<span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"homologene"</span>  </code></pre></div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>It’s not always clear whether a dataset is using the original species gene names, human gene names, or some other species’ gene names.</p>
<p><code>infer_species</code> takes a list/matrix/data.frame with genes and infers the species that they best match to!</p>
<p>For the sake of speed, the genes extracted from <code>gene_df</code> are tested against genomes from only the following 6 <code>test_species</code> by default: - human - monkey - rat - mouse - zebrafish - fly</p>
<p>However, you can supply your own list of <code>test_species</code>, which will be automatically be mapped and standardised using <code>map_species</code>.</p>
</div>
<div id="examples" class="section level1">
<h1 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h1>
<div id="mouse-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#mouse-genes" class="anchor"></a>Mouse genes</h2>
<div id="infer-the-species" class="section level3">
<h3 class="hasAnchor">
<a href="#infer-the-species" class="anchor"></a>Infer the species</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">matches</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/infer_species.html">infer_species</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse</span><span class="op">)</span></code></pre></div>
<pre><code>## Preparing gene_df.</code></pre>
<pre><code>## sparseMatrix format detected.</code></pre>
<pre><code>## Extracting genes from rownames.</code></pre>
<pre><code>## 15,259 genes extracted.</code></pre>
<pre><code>## Testing for gene overlap with: human</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: human</code></pre>
<pre><code>## Common name mapping found for human</code></pre>
<pre><code>## 1 organism identified from search: 9606</code></pre>
<pre><code>## Gene table with 19,129 rows retrieved.</code></pre>
<pre><code>## Returning all 19,129 genes from human.</code></pre>
<pre><code>## Testing for gene overlap with: monkey</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: monkey</code></pre>
<pre><code>## Common name mapping found for monkey</code></pre>
<pre><code>## 1 organism identified from search: 9544</code></pre>
<pre><code>## Gene table with 16,843 rows retrieved.</code></pre>
<pre><code>## Returning all 16,843 genes from monkey.</code></pre>
<pre><code>## Testing for gene overlap with: rat</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: rat</code></pre>
<pre><code>## Common name mapping found for rat</code></pre>
<pre><code>## 1 organism identified from search: 10116</code></pre>
<pre><code>## Gene table with 20,616 rows retrieved.</code></pre>
<pre><code>## Returning all 20,616 genes from rat.</code></pre>
<pre><code>## Testing for gene overlap with: mouse</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: mouse</code></pre>
<pre><code>## Common name mapping found for mouse</code></pre>
<pre><code>## 1 organism identified from search: 10090</code></pre>
<pre><code>## Gene table with 21,207 rows retrieved.</code></pre>
<pre><code>## Returning all 21,207 genes from mouse.</code></pre>
<pre><code>## Testing for gene overlap with: zebrafish</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: zebrafish</code></pre>
<pre><code>## Common name mapping found for zebrafish</code></pre>
<pre><code>## 1 organism identified from search: 7955</code></pre>
<pre><code>## Gene table with 20,897 rows retrieved.</code></pre>
<pre><code>## Returning all 20,897 genes from zebrafish.</code></pre>
<pre><code>## Testing for gene overlap with: fly</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: fly</code></pre>
<pre><code>## Common name mapping found for fly</code></pre>
<pre><code>## 1 organism identified from search: 7227</code></pre>
<pre><code>## Gene table with 8,438 rows retrieved.</code></pre>
<pre><code>## Returning all 8,438 genes from fly.</code></pre>
<pre><code>## Top match:
##   - species: mouse 
##   - percent_match: 92%</code></pre>
<p><img src="infer_species_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
</div>
</div>
<div id="rat-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#rat-genes" class="anchor"></a>Rat genes</h2>
<div id="create-example-data" class="section level3">
<h3 class="hasAnchor">
<a href="#create-example-data" class="anchor"></a>Create example data</h3>
<p>To create an example dataset, turn the gene names into rat genes.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_rat</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/convert_orthologs.html">convert_orthologs</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse</span>, 
                                        input_species <span class="op">=</span> <span class="st">"mouse"</span>, 
                                        output_species <span class="op">=</span> <span class="st">"rat"</span>,
                                        method <span class="op">=</span> <span class="st">"homologene"</span><span class="op">)</span></code></pre></div>
<pre><code>## Preparing gene_df.</code></pre>
<pre><code>## sparseMatrix format detected.</code></pre>
<pre><code>## Extracting genes from rownames.</code></pre>
<pre><code>## 15,259 genes extracted.</code></pre>
<pre><code>## Converting mouse ==&gt; rat orthologs using: homologene</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: mouse</code></pre>
<pre><code>## Common name mapping found for mouse</code></pre>
<pre><code>## 1 organism identified from search: 10090</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: rat</code></pre>
<pre><code>## Common name mapping found for rat</code></pre>
<pre><code>## 1 organism identified from search: 10116</code></pre>
<pre><code>## Checking for genes without orthologs in rat.</code></pre>
<pre><code>## Extracting genes from input_gene.</code></pre>
<pre><code>## 13,812 genes extracted.</code></pre>
<pre><code>## Extracting genes from ortholog_gene.</code></pre>
<pre><code>## 13,812 genes extracted.</code></pre>
<pre><code>## Checking for genes without 1:1 orthologs.</code></pre>
<pre><code>## Dropping 486 genes that have multiple input_gene per ortholog_gene.</code></pre>
<pre><code>## Dropping 148 genes that have multiple ortholog_gene per input_gene.</code></pre>
<pre><code>## Filtering gene_df with gene_map</code></pre>
<pre><code>## Setting ortholog_gene to rownames.</code></pre>
<pre><code>## 
## =========== REPORT SUMMARY ===========</code></pre>
<pre><code>## Total genes dropped after convert_orthologs :
##    2,322 / 15,259 (15%)</code></pre>
<pre><code>## Total genes remaining after convert_orthologs :
##    12,937 / 15,259 (85%)</code></pre>
</div>
<div id="infer-the-species-1" class="section level3">
<h3 class="hasAnchor">
<a href="#infer-the-species-1" class="anchor"></a>Infer the species</h3>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">matches</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/infer_species.html">infer_species</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_rat</span><span class="op">)</span></code></pre></div>
<pre><code>## Preparing gene_df.</code></pre>
<pre><code>## sparseMatrix format detected.</code></pre>
<pre><code>## Extracting genes from rownames.</code></pre>
<pre><code>## 12,937 genes extracted.</code></pre>
<pre><code>## Testing for gene overlap with: human</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: human</code></pre>
<pre><code>## Common name mapping found for human</code></pre>
<pre><code>## 1 organism identified from search: 9606</code></pre>
<pre><code>## Gene table with 19,129 rows retrieved.</code></pre>
<pre><code>## Returning all 19,129 genes from human.</code></pre>
<pre><code>## Testing for gene overlap with: monkey</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: monkey</code></pre>
<pre><code>## Common name mapping found for monkey</code></pre>
<pre><code>## 1 organism identified from search: 9544</code></pre>
<pre><code>## Gene table with 16,843 rows retrieved.</code></pre>
<pre><code>## Returning all 16,843 genes from monkey.</code></pre>
<pre><code>## Testing for gene overlap with: rat</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: rat</code></pre>
<pre><code>## Common name mapping found for rat</code></pre>
<pre><code>## 1 organism identified from search: 10116</code></pre>
<pre><code>## Gene table with 20,616 rows retrieved.</code></pre>
<pre><code>## Returning all 20,616 genes from rat.</code></pre>
<pre><code>## Testing for gene overlap with: mouse</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: mouse</code></pre>
<pre><code>## Common name mapping found for mouse</code></pre>
<pre><code>## 1 organism identified from search: 10090</code></pre>
<pre><code>## Gene table with 21,207 rows retrieved.</code></pre>
<pre><code>## Returning all 21,207 genes from mouse.</code></pre>
<pre><code>## Testing for gene overlap with: zebrafish</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: zebrafish</code></pre>
<pre><code>## Common name mapping found for zebrafish</code></pre>
<pre><code>## 1 organism identified from search: 7955</code></pre>
<pre><code>## Gene table with 20,897 rows retrieved.</code></pre>
<pre><code>## Returning all 20,897 genes from zebrafish.</code></pre>
<pre><code>## Testing for gene overlap with: fly</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: fly</code></pre>
<pre><code>## Common name mapping found for fly</code></pre>
<pre><code>## 1 organism identified from search: 7227</code></pre>
<pre><code>## Gene table with 8,438 rows retrieved.</code></pre>
<pre><code>## Returning all 8,438 genes from fly.</code></pre>
<pre><code>## Top match:
##   - species: rat 
##   - percent_match: 100%</code></pre>
<p><img src="infer_species_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
</div>
<div id="human-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#human-genes" class="anchor"></a>Human genes</h2>
<div id="create-example-data-1" class="section level3">
<h3 class="hasAnchor">
<a href="#create-example-data-1" class="anchor"></a>Create example data</h3>
<p>To create an example dataset, turn the gene names into human genes.</p>
<div class="sourceCode" id="cb152"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_human</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/convert_orthologs.html">convert_orthologs</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse</span>, 
                                          input_species <span class="op">=</span> <span class="st">"mouse"</span>, 
                                          output_species <span class="op">=</span> <span class="st">"human"</span>,
                                          method <span class="op">=</span> <span class="st">"homologene"</span><span class="op">)</span></code></pre></div>
<pre><code>## Preparing gene_df.</code></pre>
<pre><code>## sparseMatrix format detected.</code></pre>
<pre><code>## Extracting genes from rownames.</code></pre>
<pre><code>## 15,259 genes extracted.</code></pre>
<pre><code>## Converting mouse ==&gt; human orthologs using: homologene</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: mouse</code></pre>
<pre><code>## Common name mapping found for mouse</code></pre>
<pre><code>## 1 organism identified from search: 10090</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: human</code></pre>
<pre><code>## Common name mapping found for human</code></pre>
<pre><code>## 1 organism identified from search: 9606</code></pre>
<pre><code>## Checking for genes without orthologs in human.</code></pre>
<pre><code>## Extracting genes from input_gene.</code></pre>
<pre><code>## 13,416 genes extracted.</code></pre>
<pre><code>## Extracting genes from ortholog_gene.</code></pre>
<pre><code>## 13,416 genes extracted.</code></pre>
<pre><code>## Checking for genes without 1:1 orthologs.</code></pre>
<pre><code>## Dropping 46 genes that have multiple input_gene per ortholog_gene.</code></pre>
<pre><code>## Dropping 56 genes that have multiple ortholog_gene per input_gene.</code></pre>
<pre><code>## Filtering gene_df with gene_map</code></pre>
<pre><code>## Setting ortholog_gene to rownames.</code></pre>
<pre><code>## 
## =========== REPORT SUMMARY ===========</code></pre>
<pre><code>## Total genes dropped after convert_orthologs :
##    2,016 / 15,259 (13%)</code></pre>
<pre><code>## Total genes remaining after convert_orthologs :
##    13,243 / 15,259 (87%)</code></pre>
</div>
<div id="infer-the-species-2" class="section level3">
<h3 class="hasAnchor">
<a href="#infer-the-species-2" class="anchor"></a>Infer the species</h3>
<div class="sourceCode" id="cb181"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">matches</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/infer_species.html">infer_species</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_human</span><span class="op">)</span></code></pre></div>
<pre><code>## Preparing gene_df.</code></pre>
<pre><code>## sparseMatrix format detected.</code></pre>
<pre><code>## Extracting genes from rownames.</code></pre>
<pre><code>## 13,243 genes extracted.</code></pre>
<pre><code>## Testing for gene overlap with: human</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: human</code></pre>
<pre><code>## Common name mapping found for human</code></pre>
<pre><code>## 1 organism identified from search: 9606</code></pre>
<pre><code>## Gene table with 19,129 rows retrieved.</code></pre>
<pre><code>## Returning all 19,129 genes from human.</code></pre>
<pre><code>## Testing for gene overlap with: monkey</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: monkey</code></pre>
<pre><code>## Common name mapping found for monkey</code></pre>
<pre><code>## 1 organism identified from search: 9544</code></pre>
<pre><code>## Gene table with 16,843 rows retrieved.</code></pre>
<pre><code>## Returning all 16,843 genes from monkey.</code></pre>
<pre><code>## Testing for gene overlap with: rat</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: rat</code></pre>
<pre><code>## Common name mapping found for rat</code></pre>
<pre><code>## 1 organism identified from search: 10116</code></pre>
<pre><code>## Gene table with 20,616 rows retrieved.</code></pre>
<pre><code>## Returning all 20,616 genes from rat.</code></pre>
<pre><code>## Testing for gene overlap with: mouse</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: mouse</code></pre>
<pre><code>## Common name mapping found for mouse</code></pre>
<pre><code>## 1 organism identified from search: 10090</code></pre>
<pre><code>## Gene table with 21,207 rows retrieved.</code></pre>
<pre><code>## Returning all 21,207 genes from mouse.</code></pre>
<pre><code>## Testing for gene overlap with: zebrafish</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: zebrafish</code></pre>
<pre><code>## Common name mapping found for zebrafish</code></pre>
<pre><code>## 1 organism identified from search: 7955</code></pre>
<pre><code>## Gene table with 20,897 rows retrieved.</code></pre>
<pre><code>## Returning all 20,897 genes from zebrafish.</code></pre>
<pre><code>## Testing for gene overlap with: fly</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: fly</code></pre>
<pre><code>## Common name mapping found for fly</code></pre>
<pre><code>## 1 organism identified from search: 7227</code></pre>
<pre><code>## Gene table with 8,438 rows retrieved.</code></pre>
<pre><code>## Returning all 8,438 genes from fly.</code></pre>
<pre><code>## Top match:
##   - species: human 
##   - percent_match: 100%</code></pre>
<p><img src="infer_species_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
</div>
</div>
<div id="additional-test_species" class="section level1">
<h1 class="hasAnchor">
<a href="#additional-test_species" class="anchor"></a>Additional <code>test_species</code>
</h1>
<p>You can even supply <code>test_species</code> with the name of one of the R packages that <code>orthogene</code> gets orthologs from. This will test against all species available in that particular R package.</p>
<p>For example, by setting <code>test_species="homologene"</code> we automatically test for % gene matches in each of the 20+ species available in <code>homologene</code>.</p>
<div class="sourceCode" id="cb241"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">matches</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/infer_species.html">infer_species</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_human</span>, 
                                    test_species <span class="op">=</span> <span class="st">"homologene"</span><span class="op">)</span></code></pre></div>
<pre><code>## Retrieving all organisms available in homologene.</code></pre>
<pre><code>## Preparing gene_df.</code></pre>
<pre><code>## sparseMatrix format detected.</code></pre>
<pre><code>## Extracting genes from rownames.</code></pre>
<pre><code>## 13,243 genes extracted.</code></pre>
<pre><code>## Testing for gene overlap with: Mus musculus</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Mus musculus</code></pre>
<pre><code>## 1 organism identified from search: 10090</code></pre>
<pre><code>## Gene table with 21,207 rows retrieved.</code></pre>
<pre><code>## Returning all 21,207 genes from Mus musculus.</code></pre>
<pre><code>## Testing for gene overlap with: Rattus norvegicus</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Rattus norvegicus</code></pre>
<pre><code>## 1 organism identified from search: 10116</code></pre>
<pre><code>## Gene table with 20,616 rows retrieved.</code></pre>
<pre><code>## Returning all 20,616 genes from Rattus norvegicus.</code></pre>
<pre><code>## Testing for gene overlap with: Kluyveromyces lactis</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Kluyveromyces lactis</code></pre>
<pre><code>## WARNING: No organisms identified matched  'kluyveromyces lactis' Try a different query.</code></pre>
<pre><code>## Gene table with 0 rows retrieved.</code></pre>
<pre><code>## Returning all 0 genes from Kluyveromyces lactis.</code></pre>
<pre><code>## Testing for gene overlap with: Magnaporthe oryzae</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Magnaporthe oryzae</code></pre>
<pre><code>## 1 organism identified from search: 242507</code></pre>
<pre><code>## Gene table with 0 rows retrieved.</code></pre>
<pre><code>## Returning all 0 genes from Magnaporthe oryzae.</code></pre>
<pre><code>## Testing for gene overlap with: Eremothecium gossypii</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Eremothecium gossypii</code></pre>
<pre><code>## 1 organism identified from search: 284811</code></pre>
<pre><code>## Gene table with 0 rows retrieved.</code></pre>
<pre><code>## Returning all 0 genes from Eremothecium gossypii.</code></pre>
<pre><code>## Testing for gene overlap with: Arabidopsis thaliana</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Arabidopsis thaliana</code></pre>
<pre><code>## 1 organism identified from search: 3702</code></pre>
<pre><code>## Gene table with 19,143 rows retrieved.</code></pre>
<pre><code>## Returning all 19,143 genes from Arabidopsis thaliana.</code></pre>
<pre><code>## Testing for gene overlap with: Oryza sativa</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Oryza sativa</code></pre>
<pre><code>## 2 organisms identified from search.
## Selecting first:
##    - 39946
##    - 39947</code></pre>
<pre><code>## Gene table with 0 rows retrieved.</code></pre>
<pre><code>## Returning all 0 genes from Oryza sativa.</code></pre>
<pre><code>## Testing for gene overlap with: Schizosaccharomyces pombe</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Schizosaccharomyces pombe</code></pre>
<pre><code>## 1 organism identified from search: 284812</code></pre>
<pre><code>## Gene table with 0 rows retrieved.</code></pre>
<pre><code>## Returning all 0 genes from Schizosaccharomyces pombe.</code></pre>
<pre><code>## Testing for gene overlap with: Saccharomyces cerevisiae</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Saccharomyces cerevisiae</code></pre>
<pre><code>## 1 organism identified from search: 4932</code></pre>
<pre><code>## Gene table with 4,579 rows retrieved.</code></pre>
<pre><code>## Returning all 4,579 genes from Saccharomyces cerevisiae.</code></pre>
<pre><code>## Testing for gene overlap with: Neurospora crassa</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Neurospora crassa</code></pre>
<pre><code>## 1 organism identified from search: 367110</code></pre>
<pre><code>## Gene table with 0 rows retrieved.</code></pre>
<pre><code>## Returning all 0 genes from Neurospora crassa.</code></pre>
<pre><code>## Testing for gene overlap with: Caenorhabditis elegans</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Caenorhabditis elegans</code></pre>
<pre><code>## 2 organisms identified from search.
## Selecting first:
##    - 6239
##    - 6239</code></pre>
<pre><code>## Gene table with 7,575 rows retrieved.</code></pre>
<pre><code>## Returning all 7,575 genes from Caenorhabditis elegans.</code></pre>
<pre><code>## Testing for gene overlap with: Anopheles gambiae</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Anopheles gambiae</code></pre>
<pre><code>## 1 organism identified from search: 7165</code></pre>
<pre><code>## Gene table with 8,428 rows retrieved.</code></pre>
<pre><code>## Returning all 8,428 genes from Anopheles gambiae.</code></pre>
<pre><code>## Testing for gene overlap with: Drosophila melanogaster</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Drosophila melanogaster</code></pre>
<pre><code>## 1 organism identified from search: 7227</code></pre>
<pre><code>## Gene table with 8,438 rows retrieved.</code></pre>
<pre><code>## Returning all 8,438 genes from Drosophila melanogaster.</code></pre>
<pre><code>## Testing for gene overlap with: Danio rerio</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Danio rerio</code></pre>
<pre><code>## 1 organism identified from search: 7955</code></pre>
<pre><code>## Gene table with 20,897 rows retrieved.</code></pre>
<pre><code>## Returning all 20,897 genes from Danio rerio.</code></pre>
<pre><code>## Testing for gene overlap with: Xenopus (Silurana) tropicalis</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Xenopus (Silurana) tropicalis</code></pre>
<pre><code>## Warning in grepl(spec_queries, display_name, ignore.case = TRUE): TRE pattern
## compilation error 'Missing ')''</code></pre>
<pre><code>## Testing for gene overlap with: Gallus gallus</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Gallus gallus</code></pre>
<pre><code>## 1 organism identified from search: 9031</code></pre>
<pre><code>## Gene table with 14,600 rows retrieved.</code></pre>
<pre><code>## Returning all 14,600 genes from Gallus gallus.</code></pre>
<pre><code>## Testing for gene overlap with: Macaca mulatta</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Macaca mulatta</code></pre>
<pre><code>## 1 organism identified from search: 9544</code></pre>
<pre><code>## Gene table with 16,843 rows retrieved.</code></pre>
<pre><code>## Returning all 16,843 genes from Macaca mulatta.</code></pre>
<pre><code>## Testing for gene overlap with: Pan troglodytes</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Pan troglodytes</code></pre>
<pre><code>## 1 organism identified from search: 9598</code></pre>
<pre><code>## Gene table with 18,730 rows retrieved.</code></pre>
<pre><code>## Returning all 18,730 genes from Pan troglodytes.</code></pre>
<pre><code>## Testing for gene overlap with: Homo sapiens</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Homo sapiens</code></pre>
<pre><code>## 1 organism identified from search: 9606</code></pre>
<pre><code>## Gene table with 19,129 rows retrieved.</code></pre>
<pre><code>## Returning all 19,129 genes from Homo sapiens.</code></pre>
<pre><code>## Testing for gene overlap with: Canis lupus familiaris</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Canis lupus familiaris</code></pre>
<pre><code>## 1 organism identified from search: 9615</code></pre>
<pre><code>## Gene table with 18,117 rows retrieved.</code></pre>
<pre><code>## Returning all 18,117 genes from Canis lupus familiaris.</code></pre>
<pre><code>## Testing for gene overlap with: Bos taurus</code></pre>
<pre><code>## Retrieving all genes using: homologene.</code></pre>
<pre><code>## Retrieving all organisms available in gprofiler.</code></pre>
<pre><code>## Using stored `gprofiler_orgs`.</code></pre>
<pre><code>## Mapping species name: Bos taurus</code></pre>
<pre><code>## 2 organisms identified from search.
## Selecting first:
##    - 30522
##    - 9913</code></pre>
<pre><code>## Gene table with 0 rows retrieved.</code></pre>
<pre><code>## Returning all 0 genes from Bos taurus.</code></pre>
<pre><code>## Top match:
##   - species: Homo sapiens 
##   - percent_match: 100%</code></pre>
<p><img src="infer_species_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<details><div class="sourceCode" id="cb414"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Big Sur 10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] orthogene_0.99.8 BiocStyle_2.21.4
## 
## loaded via a namespace (and not attached):
##  [1] fs_1.5.0                  httr_1.4.2               
##  [3] rprojroot_2.0.2           tools_4.1.0              
##  [5] backports_1.2.1           bslib_0.3.1              
##  [7] utf8_1.2.2                R6_2.5.1                 
##  [9] DBI_1.1.1                 lazyeval_0.2.2           
## [11] colorspace_2.0-2          tidyselect_1.1.1         
## [13] curl_4.3.2                compiler_4.1.0           
## [15] textshaping_0.3.6         cli_3.0.1                
## [17] desc_1.4.0                plotly_4.10.0            
## [19] labeling_0.4.2            bookdown_0.24            
## [21] sass_0.4.0                scales_1.1.1             
## [23] pkgdown_1.6.1             systemfonts_1.0.3        
## [25] stringr_1.4.0             digest_0.6.28            
## [27] foreign_0.8-81            rmarkdown_2.11           
## [29] rio_0.5.27                pkgconfig_2.0.3          
## [31] htmltools_0.5.2           fastmap_1.1.0            
## [33] highr_0.9                 htmlwidgets_1.5.4        
## [35] rlang_0.4.12              readxl_1.3.1             
## [37] rstudioapi_0.13           jquerylib_0.1.4          
## [39] generics_0.1.0            farver_2.1.0             
## [41] jsonlite_1.7.2            dplyr_1.0.7              
## [43] zip_2.2.0                 car_3.0-11               
## [45] homologene_1.4.68.19.3.27 magrittr_2.0.1           
## [47] GenomeInfoDbData_1.2.7    patchwork_1.1.1          
## [49] Matrix_1.3-4              Rcpp_1.0.7               
## [51] munsell_0.5.0             fansi_0.5.0              
## [53] abind_1.4-5               babelgene_21.4           
## [55] lifecycle_1.0.1           stringi_1.7.5            
## [57] yaml_2.2.1                carData_3.0-4            
## [59] grid_4.1.0                parallel_4.1.0           
## [61] forcats_0.5.1             crayon_1.4.1             
## [63] lattice_0.20-45           haven_2.4.3              
## [65] hms_1.1.1                 knitr_1.36               
## [67] pillar_1.6.4              ggpubr_0.4.0             
## [69] ggsignif_0.6.3            glue_1.4.2               
## [71] gprofiler2_0.2.1          evaluate_0.14            
## [73] data.table_1.14.2         BiocManager_1.30.16      
## [75] vctrs_0.3.8               cellranger_1.1.0         
## [77] gtable_0.3.0              purrr_0.3.4              
## [79] tidyr_1.1.4               assertthat_0.2.1         
## [81] cachem_1.0.6              ggplot2_3.3.5            
## [83] xfun_0.27                 openxlsx_4.2.4           
## [85] broom_0.7.9               rstatix_0.7.0            
## [87] ragg_1.1.3                viridisLite_0.4.0        
## [89] tibble_3.1.5              memoise_2.0.0            
## [91] ellipsis_0.3.2</code></pre>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
