<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>orthogene: Getting Started • orthogene</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="orthogene: Getting Started">
<meta property="og:description" content="orthogene">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthogene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/orthogene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/infer_species.html">Infer species</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/orthogene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>orthogene: Getting Started</h1>
                        <h4 data-toc-skip class="author"><h4>
Author: <i>Brian M. Schilder</i>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
Most recent update: <i>Oct-27-2021</i>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/orthogene/blob/HEAD/vignettes/orthogene.Rmd" class="external-link"><code>vignettes/orthogene.Rmd</code></a></small>
      <div class="hidden name"><code>orthogene.Rmd</code></div>

    </div>

    
    
<!-- badges: start -->
<!-- badger::badge_codecov() -->
<!-- copied from MungeSumstats README.Rmd -->
<!-- badger::badge_lifecycle("stable", "green") -->
<!-- badger::badge_last_commit()  -->
<!-- badger::badge_license() -->
<p><a href="https://codecov.io/gh/neurogenomics/orthogene" class="external-link"><img src="https://codecov.io/gh/neurogenomics/orthogene/branch/main/graph/badge.svg"></a> <a href="https://github.com/neurogenomics/orthogene/actions" class="external-link"><img src="https://github.com/neurogenomics/orthogene/workflows/R-full/badge.svg" alt="R-CMD-check"></a> <a href="https://lifecycle.r-lib.org/articles/stages.html#stable" class="external-link"><img src="https://img.shields.io/badge/lifecycle-stable-green.svg"></a> <a href="https://github.com/neurogenomics/orthogene/commits/main" class="external-link"><img src="https://img.shields.io/github/last-commit/neurogenomics/orthogene.svg"></a> <a href="https://cran.r-project.org/web/licenses/GPL-3" class="external-link"><img src="https://img.shields.io/badge/license-GPL--3-blue.svg" alt="License: GPL-3"></a> <!-- badges: end --></p>
<div class="section level2">
<h2 id="orthogene-interspecies-gene-mapping">
<code>orthogene</code>: Interspecies gene mapping<a class="anchor" aria-label="anchor" href="#orthogene-interspecies-gene-mapping"></a>
</h2>
<p><code>orthogene</code> is an R package for easy mapping of orthologous genes across hundreds of species.<br>
It pulls up-to-date interspecies gene ortholog mappings across 700+ organisms.</p>
<p>It also provides various utility functions to map common objects (e.g. data.frames, gene expression matrices, lists) onto 1:1 gene orthologs from any other species.</p>
<p>In brief, <code>orthogene</code> lets you easily:</p>
<ul>
<li><a href="https://neurogenomics.github.io/orthogene/articles/orthogene#convert-orthologs" class="external-link"><strong><code>convert_orthologs</code></strong> between any two species</a></li>
<li>
<a href="https://neurogenomics.github.io/orthogene/articles/orthogene#map-species" class="external-link"><strong><code>map_species</code></strong> names onto standard taxonomic ontologies</a><br>
</li>
<li><a href="https://neurogenomics.github.io/orthogene/articles/orthogene#report-orthologs" class="external-link"><strong><code>report_orthologs</code></strong> between any two species</a></li>
<li><a href="https://neurogenomics.github.io/orthogene/articles/orthogene#map-genes" class="external-link"><strong><code>map_genes</code></strong> onto standard ontologies</a></li>
<li>
<a href="https://neurogenomics.github.io/orthogene/articles/orthogene#aggregate-mapped-genes" class="external-link"><strong><code>aggregate_mapped_genes</code></strong> in a matrix.</a><br>
</li>
<li><a href="https://neurogenomics.github.io/orthogene/articles/orthogene#get-all-genes" class="external-link">get <strong><code>all_genes</code></strong> from any species</a></li>
</ul>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="co"># orthogene is only available on Bioconductor&gt;=3.14</span>
<span class="kw">if</span><span class="op">(</span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/version.html" class="external-link">version</a></span><span class="op">(</span><span class="op">)</span><span class="op">&lt;</span><span class="st">"3.14"</span><span class="op">)</span> <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span>version <span class="op">=</span> <span class="st">"3.14"</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"orthogene"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/neurogenomics/orthogene" class="external-link">orthogene</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"exp_mouse"</span><span class="op">)</span>
<span class="co"># Setting to "homologene" for the purposes of quick demonstration.</span>
<span class="co"># We generally recommend using method="gprofiler" (default).</span>
<span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"homologene"</span>  </code></pre></div>
</div>
<div class="section level1">
<h1 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h1>
<div class="section level2">
<h2 id="convert-orthologs">Convert orthologs<a class="anchor" aria-label="anchor" href="#convert-orthologs"></a>
</h2>
<p><a href="https://neurogenomics.github.io/orthogene/reference/convert_orthologs.html" class="external-link"><code>convert_orthologs</code></a> is very flexible with what users can supply as <code>gene_df</code>, and can take a <code>data.frame</code>/<code>data.table</code>/<code>tibble</code>, (sparse) <code>matrix</code>, or <code>list</code>/<code>vector</code> containing genes.</p>
<p>Genes, transcripts, proteins, SNPs, or genomic ranges will be recognised in most formats (HGNC, Ensembl, RefSeq, UniProt, etc.) and can even be a mixture of different formats.</p>
<p>All genes will be mapped to gene symbols, unless specified otherwise with the <code>...</code> arguments (see <code><a href="../reference/convert_orthologs.html">?orthogene::convert_orthologs</a></code> or <a href="https://neurogenomics.github.io/orthogene/reference/convert_orthologs.html" class="external-link">here</a> for details).</p>
<div class="section level3">
<h3 id="note-on-non-11-orthologs">Note on non-1:1 orthologs<a class="anchor" aria-label="anchor" href="#note-on-non-11-orthologs"></a>
</h3>
<p>A key feature of <a href="https://neurogenomics.github.io/orthogene/reference/convert_orthologs.html" class="external-link"><code>convert_orthologs</code></a> is that it handles the issue of genes with many-to-many mappings across species. This can occur due to evolutionary divergence, and the function of these genes tends to be less conserved and less translatable. Users can address this using different strategies via <code>non121_strategy=</code>:</p>
<ol style="list-style-type: decimal">
<li>
<code>"drop_both_species"</code> : Drop genes that have duplicate mappings in either the input_species or output_species, (<em>DEFAULT</em>).</li>
<li>
<code>"drop_input_species"</code> : Only drop genes that have duplicate mappings in <code>input_species</code>.<br>
</li>
<li>
<code>"drop_output_species"</code> : Only drop genes that have duplicate mappings in the <code>output_species</code>.</li>
<li>
<code>"keep_both_species"</code> : Keep all genes regardless of whether they have duplicate mappings in either species.<br>
</li>
<li>
<code>"keep_popular"</code> : Return only the most “popular” interspecies ortholog mappings. This procedure tends to yield a greater number of returned genes but at the cost of many of them not being true biological 1:1 orthologs.</li>
<li>
<code>"sum"</code>,<code>"mean"</code>,<code>"median"</code>,<code>"min"</code> or <code>"max"</code>: When <code>gene_df</code> is a matrix and <code>gene_output="rownames"</code>, these options will aggregate many-to-one gene mappings (<code>input_species</code>-to-<code>output_species</code>) after dropping any duplicate genes in the <code>output_species</code>.</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene_df</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/convert_orthologs.html">convert_orthologs</a></span><span class="op">(</span>gene_df <span class="op">=</span> <span class="va">exp_mouse</span>,
                                        gene_input <span class="op">=</span> <span class="st">"rownames"</span>, 
                                        gene_output <span class="op">=</span> <span class="st">"rownames"</span>, 
                                        input_species <span class="op">=</span> <span class="st">"mouse"</span>,
                                        output_species <span class="op">=</span> <span class="st">"human"</span>,
                                        non121_strategy <span class="op">=</span> <span class="st">"drop_both_species"</span>,
                                        method <span class="op">=</span> <span class="va">method</span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## sparseMatrix format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from rownames.</span></code></pre>
<pre><code><span class="co">## 15,259 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting mouse ==&gt; human orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in human.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 13,416 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 13,416 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 46 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 56 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Setting ortholog_gene to rownames.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    2,016 / 15,259 (13%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    13,243 / 15,259 (87%)</span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_df</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="18%">
<col width="15%">
<col width="11%">
<col width="8%">
<col width="14%">
<col width="12%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">astrocytes_ependymal</th>
<th align="right">endothelial-mural</th>
<th align="right">interneurons</th>
<th align="right">microglia</th>
<th align="right">oligodendrocytes</th>
<th align="right">pyramidal CA1</th>
<th align="right">pyramidal SS</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">TSPAN12</td>
<td align="right">0.3303571</td>
<td align="right">0.5872340</td>
<td align="right">0.6413793</td>
<td align="right">0.1428571</td>
<td align="right">0.1207317</td>
<td align="right">0.2864750</td>
<td align="right">0.1453634</td>
</tr>
<tr class="even">
<td align="left">TSHZ1</td>
<td align="right">0.4285714</td>
<td align="right">0.4468085</td>
<td align="right">1.1551724</td>
<td align="right">0.4387755</td>
<td align="right">0.3621951</td>
<td align="right">0.0692226</td>
<td align="right">0.8320802</td>
</tr>
<tr class="odd">
<td align="left">ADAMTS15</td>
<td align="right">0.0089286</td>
<td align="right">0.0978723</td>
<td align="right">0.2206897</td>
<td align="right">0.0000000</td>
<td align="right">0.0231707</td>
<td align="right">0.0117146</td>
<td align="right">0.0375940</td>
</tr>
<tr class="even">
<td align="left">CLDN12</td>
<td align="right">0.2232143</td>
<td align="right">0.1148936</td>
<td align="right">0.5517241</td>
<td align="right">0.0510204</td>
<td align="right">0.2609756</td>
<td align="right">0.4376997</td>
<td align="right">0.6842105</td>
</tr>
<tr class="odd">
<td align="left">RXFP1</td>
<td align="right">0.0000000</td>
<td align="right">0.0127660</td>
<td align="right">0.2551724</td>
<td align="right">0.0000000</td>
<td align="right">0.0158537</td>
<td align="right">0.0511182</td>
<td align="right">0.0751880</td>
</tr>
<tr class="even">
<td align="left">SEMA3C</td>
<td align="right">0.1964286</td>
<td align="right">0.9957447</td>
<td align="right">8.6379310</td>
<td align="right">0.2040816</td>
<td align="right">0.1853659</td>
<td align="right">0.1608094</td>
<td align="right">0.2280702</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="map-species">Map species<a class="anchor" aria-label="anchor" href="#map-species"></a>
</h2>
<p><a href="https://neurogenomics.github.io/orthogene/reference/map_species.html" class="external-link"><code>map_species</code></a> lets you standardise species names from a wide variety of identifiers (e.g. common name, taxonomy ID, Latin name, partial match).</p>
<p>All exposed <code>orthogene</code> functions (including <a href="https://neurogenomics.github.io/orthogene/reference/convert_orthologs.html" class="external-link"><code>convert_orthologs</code></a>) use <code>map_species</code> under the hood, so you don’t have to worry about getting species names exactly right.</p>
<p>You can check the full list of available species by simply running <code><a href="../reference/map_species.html">map_species()</a></code> with no arguments, or checking <a href="https://biit.cs.ut.ee/gprofiler/page/organism-list" class="external-link">here</a>.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/map_species.html">map_species</a></span><span class="op">(</span>species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"human"</span>,<span class="fl">9544</span>,<span class="st">"mus musculus"</span>,
                                              <span class="st">"fruit fly"</span>,<span class="st">"Celegans"</span><span class="op">)</span>, 
                                  output_format <span class="op">=</span> <span class="st">"scientific_name"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: Homo sapiens</span></code></pre>
<pre><code><span class="co">## Mapping species name: 9544</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: Macaca mulatta</span></code></pre>
<pre><code><span class="co">## Mapping species name: mus musculus</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: Mus musculus</span></code></pre>
<pre><code><span class="co">## Mapping species name: fruit fly</span></code></pre>
<pre><code><span class="co">## Common name mapping found for fruit fly</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: Drosophila melanogaster</span></code></pre>
<pre><code><span class="co">## Mapping species name: Celegans</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: Caenorhabditis elegans</span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">species</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                     human                      9544              mus musculus </span>
<span class="co">##            "Homo sapiens"          "Macaca mulatta"            "Mus musculus" </span>
<span class="co">##                 fruit fly                  Celegans </span>
<span class="co">## "Drosophila melanogaster"  "Caenorhabditis elegans"</span></code></pre>
</div>
<div class="section level2">
<h2 id="report-orthologs">Report orthologs<a class="anchor" aria-label="anchor" href="#report-orthologs"></a>
</h2>
<p>It may be helpful to know the maximum expected number of orthologous gene mappings from one species to another.</p>
<p><a href="https://neurogenomics.github.io/orthogene/reference/report_orthologs.html" class="external-link"><code>ortholog_report</code></a> generates a report that tells you this information genome-wide.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">orth_zeb</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/report_orthologs.html">report_orthologs</a></span><span class="op">(</span>target_species <span class="op">=</span> <span class="st">"zebrafish"</span>,
                                        reference_species <span class="op">=</span> <span class="st">"human"</span>,
                                        method_all_genes <span class="op">=</span> <span class="va">method</span>,
                                        method_convert_orthologs <span class="op">=</span> <span class="va">method</span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: zebrafish</span></code></pre>
<pre><code><span class="co">## Common name mapping found for zebrafish</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: drerio</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: human</span></code></pre>
<pre><code><span class="co">## Common name mapping found for human</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: hsapiens</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: drerio</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7955</span></code></pre>
<pre><code><span class="co">## Gene table with 20,897 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 20,897 genes from drerio.</span></code></pre>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Gene table with 19,129 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 19,129 genes from hsapiens.</span></code></pre>
<pre><code><span class="co">## Preparing gene_df.</span></code></pre>
<pre><code><span class="co">## data.frame format detected.</span></code></pre>
<pre><code><span class="co">## Extracting genes from Gene.Symbol.</span></code></pre>
<pre><code><span class="co">## 20,897 genes extracted.</span></code></pre>
<pre><code><span class="co">## Converting drerio ==&gt; hsapiens orthologs using: homologene</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: drerio</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 7955</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: hsapiens</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 9606</span></code></pre>
<pre><code><span class="co">## Checking for genes without orthologs in hsapiens.</span></code></pre>
<pre><code><span class="co">## Extracting genes from input_gene.</span></code></pre>
<pre><code><span class="co">## 14,768 genes extracted.</span></code></pre>
<pre><code><span class="co">## Extracting genes from ortholog_gene.</span></code></pre>
<pre><code><span class="co">## 14,768 genes extracted.</span></code></pre>
<pre><code><span class="co">## Checking for genes without 1:1 orthologs.</span></code></pre>
<pre><code><span class="co">## Dropping 47 genes that have multiple input_gene per ortholog_gene.</span></code></pre>
<pre><code><span class="co">## Dropping 2,708 genes that have multiple ortholog_gene per input_gene.</span></code></pre>
<pre><code><span class="co">## Filtering gene_df with gene_map</span></code></pre>
<pre><code><span class="co">## Adding input_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## Adding ortholog_gene col to gene_df.</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## Total genes dropped after convert_orthologs :</span>
<span class="co">##    10,338 / 20,895 (49%)</span></code></pre>
<pre><code><span class="co">## Total genes remaining after convert_orthologs :</span>
<span class="co">##    10,557 / 20,895 (51%)</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## =========== REPORT SUMMARY ===========</span></code></pre>
<pre><code><span class="co">## 10,556 / 20,895 (50.52%) target_species genes remain after ortholog conversion.</span></code></pre>
<pre><code><span class="co">## 10,556 / 19,129 (55.18%) reference_species genes remain after ortholog conversion.</span></code></pre>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">orth_zeb</span><span class="op">$</span><span class="va">map</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">HID</th>
<th align="right">Gene.ID</th>
<th align="left">Gene.Symbol</th>
<th align="right">taxonomy_id</th>
<th align="left">input_gene</th>
<th align="left">ortholog_gene</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">8</td>
<td align="right">3</td>
<td align="right">406283</td>
<td align="left">acadm</td>
<td align="right">7955</td>
<td align="left">acadm</td>
<td align="left">ACADM</td>
</tr>
<tr class="even">
<td align="left">20</td>
<td align="right">5</td>
<td align="right">573723</td>
<td align="left">acadvl</td>
<td align="right">7955</td>
<td align="left">acadvl</td>
<td align="left">ACADVL</td>
</tr>
<tr class="odd">
<td align="left">32</td>
<td align="right">6</td>
<td align="right">445290</td>
<td align="left">acat1</td>
<td align="right">7955</td>
<td align="left">acat1</td>
<td align="left">ACAT1</td>
</tr>
<tr class="even">
<td align="left">55</td>
<td align="right">7</td>
<td align="right">30615</td>
<td align="left">acvr1l</td>
<td align="right">7955</td>
<td align="left">acvr1l</td>
<td align="left">ACVR1</td>
</tr>
<tr class="odd">
<td align="left">74</td>
<td align="right">12</td>
<td align="right">334431</td>
<td align="left">adsl</td>
<td align="right">7955</td>
<td align="left">adsl</td>
<td align="left">ADSL</td>
</tr>
<tr class="even">
<td align="left">92</td>
<td align="right">13</td>
<td align="right">566517</td>
<td align="left">aga</td>
<td align="right">7955</td>
<td align="left">aga</td>
<td align="left">AGA</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">orth_zeb</span><span class="op">$</span><span class="va">report</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="12%">
<col width="15%">
<col width="14%">
<col width="17%">
<col width="14%">
<col width="12%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th align="left">target_species</th>
<th align="right">target_total_genes</th>
<th align="left">reference_species</th>
<th align="right">reference_total_genes</th>
<th align="right">one2one_orthologs</th>
<th align="right">target_percent</th>
<th align="right">reference_percent</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">drerio</td>
<td align="right">20895</td>
<td align="left">hsapiens</td>
<td align="right">19129</td>
<td align="right">10556</td>
<td align="right">50.52</td>
<td align="right">55.18</td>
</tr></tbody>
</table>
</div>
<div class="section level2">
<h2 id="map-genes">Map genes<a class="anchor" aria-label="anchor" href="#map-genes"></a>
</h2>
<p><a href="https://neurogenomics.github.io/orthogene/reference/map_genes.html" class="external-link"><code>map_genes</code></a> finds matching <em>within-species</em> synonyms across a wide variety of gene naming conventions (HGNC, Ensembl, RefSeq, UniProt, etc.) and returns a table with standardised gene symbols (or whatever output format you prefer).</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Klf4"</span>, <span class="st">"Sox2"</span>, <span class="st">"TSPAN12"</span>,<span class="st">"NM_173007"</span>,<span class="st">"Q8BKT6"</span>,<span class="fl">9999</span>,
             <span class="st">"ENSMUSG00000012396"</span>,<span class="st">"ENSMUSG00000074637"</span><span class="op">)</span>
<span class="va">mapped_genes</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/map_genes.html">map_genes</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="va">genes</span>,
                                     species <span class="op">=</span> <span class="st">"mouse"</span>, 
                                     drop_na <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## 8 / 7 (87.5%) genes mapped.</span></code></pre>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mapped_genes</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="5%">
<col width="8%">
<col width="11%">
<col width="4%">
<col width="41%">
<col width="21%">
</colgroup>
<thead><tr class="header">
<th align="right">input_number</th>
<th align="left">input</th>
<th align="left">target_number</th>
<th align="left">target</th>
<th align="left">name</th>
<th align="left">description</th>
<th align="left">namespace</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">Klf4</td>
<td align="left">1.1</td>
<td align="left">ENSMUSG00000003032</td>
<td align="left">Klf4</td>
<td align="left">Kruppel-like factor 4 (gut) [Source:MGI Symbol;Acc:MGI:1342287]</td>
<td align="left">ENTREZGENE,MGI,UNIPROT_GN,WIKIGENE</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="left">Sox2</td>
<td align="left">2.1</td>
<td align="left">ENSMUSG00000074637</td>
<td align="left">Sox2</td>
<td align="left">SRY (sex determining region Y)-box 2 [Source:MGI Symbol;Acc:MGI:98364]</td>
<td align="left">ENTREZGENE,MGI,UNIPROT_GN,WIKIGENE</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="left">TSPAN12</td>
<td align="left">3.1</td>
<td align="left">ENSMUSG00000029669</td>
<td align="left">Tspan12</td>
<td align="left">tetraspanin 12 [Source:MGI Symbol;Acc:MGI:1889818]</td>
<td align="left">ENTREZGENE,MGI,UNIPROT_GN,WIKIGENE</td>
</tr>
<tr class="even">
<td align="right">4</td>
<td align="left">NM_173007</td>
<td align="left">4.1</td>
<td align="left">ENSMUSG00000029669</td>
<td align="left">Tspan12</td>
<td align="left">tetraspanin 12 [Source:MGI Symbol;Acc:MGI:1889818]</td>
<td align="left">REFSEQ_MRNA_ACC</td>
</tr>
<tr class="odd">
<td align="right">5</td>
<td align="left">Q8BKT6</td>
<td align="left">5.1</td>
<td align="left">ENSMUSG00000029669</td>
<td align="left">Tspan12</td>
<td align="left">tetraspanin 12 [Source:MGI Symbol;Acc:MGI:1889818]</td>
<td align="left">UNIPROTSWISSPROT_ACC,UNIPROT_GN_ACC</td>
</tr>
<tr class="even">
<td align="right">6</td>
<td align="left">9999</td>
<td align="left">6.1</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left"></td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="aggregate-mapped-genes">Aggregate mapped genes<a class="anchor" aria-label="anchor" href="#aggregate-mapped-genes"></a>
</h2>
<p><code>aggregate_mapped_genes</code> does the following:</p>
<ol style="list-style-type: decimal">
<li>Uses <code>map_genes</code> to identify <em>within-species</em> many-to-one gene mappings (e.g. Ensembl transcript IDs ==&gt; gene symbols). Alternatively, can map <em>across species</em> if output from <code>map_orthologs</code> is supplied to <code>gene_map</code> argument (and <code>gene_map_col="ortholog_gene"</code>).<br>
</li>
<li>Drops all non-mappable genes.<br>
</li>
<li>Aggregates the values of matrix <code>gene_df</code> using <code>"sum"</code>,<code>"mean"</code>,<code>"median"</code>,<code>"min"</code> or <code>"max"</code>.</li>
</ol>
<p>Note, this only works when the input data (<code>gene_df</code>) is a sparse or dense matrix, and the genes are row names.</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"exp_mouse_enst"</span><span class="op">)</span> 
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">exp_mouse_enst</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="15%">
<col width="16%">
<col width="14%">
<col width="10%">
<col width="8%">
<col width="13%">
<col width="11%">
<col width="10%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">astrocytes_ependymal</th>
<th align="right">endothelial-mural</th>
<th align="right">interneurons</th>
<th align="right">microglia</th>
<th align="right">oligodendrocytes</th>
<th align="right">pyramidal CA1</th>
<th align="right">pyramidal SS</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSMUST00000102875</td>
<td align="right">2.8258910</td>
<td align="right">0.4041560</td>
<td align="right">1.3171987</td>
<td align="right">0.3774840</td>
<td align="right">1.3426606</td>
<td align="right">1.0403481</td>
<td align="right">1.0876508</td>
</tr>
<tr class="even">
<td align="left">ENSMUST00000133343</td>
<td align="right">2.8259032</td>
<td align="right">0.4042189</td>
<td align="right">1.3171312</td>
<td align="right">0.3774038</td>
<td align="right">1.3425772</td>
<td align="right">1.0403432</td>
<td align="right">1.0876385</td>
</tr>
<tr class="odd">
<td align="left">ENSMUST00000143890</td>
<td align="right">2.8258554</td>
<td align="right">0.4041963</td>
<td align="right">1.3171145</td>
<td align="right">0.3774192</td>
<td align="right">1.3426119</td>
<td align="right">1.0403496</td>
<td align="right">1.0876334</td>
</tr>
<tr class="even">
<td align="left">ENSMUST00000005053</td>
<td align="right">0.4597978</td>
<td align="right">0.3403299</td>
<td align="right">0.9067953</td>
<td align="right">0.2958589</td>
<td align="right">0.7254482</td>
<td align="right">0.4813420</td>
<td align="right">0.7418000</td>
</tr>
<tr class="odd">
<td align="left">ENSMUST00000185896</td>
<td align="right">0.4596631</td>
<td align="right">0.3403637</td>
<td align="right">0.9067538</td>
<td align="right">0.2958896</td>
<td align="right">0.7255006</td>
<td align="right">0.4812783</td>
<td align="right">0.7417918</td>
</tr>
<tr class="even">
<td align="left">ENSMUST00000188282</td>
<td align="right">0.4597399</td>
<td align="right">0.3403441</td>
<td align="right">0.9067727</td>
<td align="right">0.2957819</td>
<td align="right">0.7255681</td>
<td align="right">0.4811978</td>
<td align="right">0.7417924</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp_agg</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/aggregate_mapped_genes.html">aggregate_mapped_genes</a></span><span class="op">(</span>gene_df<span class="op">=</span><span class="va">exp_mouse_enst</span>,
                                             species<span class="op">=</span><span class="st">"mouse"</span>,
                                             FUN <span class="op">=</span> <span class="st">"sum"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: mmusculus</span></code></pre>
<pre><code><span class="co">## 482 / 482 (100%) genes mapped.</span></code></pre>
<pre><code><span class="co">## Aggregating rows using: monocle3</span></code></pre>
<pre><code><span class="co">## Matrix aggregated:</span>
<span class="co">##   - Input: 482 x 7 </span>
<span class="co">##   - Output: 92 x 7</span></code></pre>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">exp_agg</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="7%">
<col width="18%">
<col width="15%">
<col width="11%">
<col width="8%">
<col width="14%">
<col width="12%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">astrocytes_ependymal</th>
<th align="right">endothelial-mural</th>
<th align="right">interneurons</th>
<th align="right">microglia</th>
<th align="right">oligodendrocytes</th>
<th align="right">pyramidal CA1</th>
<th align="right">pyramidal SS</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Tshz1</td>
<td align="right">1.713936</td>
<td align="right">1.7869498</td>
<td align="right">4.620366</td>
<td align="right">1.7545487</td>
<td align="right">1.4483505</td>
<td align="right">0.2764327</td>
<td align="right">3.3280256</td>
</tr>
<tr class="even">
<td align="left">Tspan12</td>
<td align="right">1.981594</td>
<td align="right">3.5228954</td>
<td align="right">3.847706</td>
<td align="right">0.8565873</td>
<td align="right">0.7237384</td>
<td align="right">1.7184690</td>
<td align="right">0.8716624</td>
</tr>
<tr class="odd">
<td align="left">Ugp2</td>
<td align="right">11.303434</td>
<td align="right">1.6167531</td>
<td align="right">5.268556</td>
<td align="right">1.5096830</td>
<td align="right">5.3705113</td>
<td align="right">4.1614945</td>
<td align="right">4.3505804</td>
</tr>
<tr class="even">
<td align="left">Usp28</td>
<td align="right">1.561545</td>
<td align="right">1.4885072</td>
<td align="right">12.481956</td>
<td align="right">0.9176950</td>
<td align="right">1.0237324</td>
<td align="right">5.5261972</td>
<td align="right">6.4652509</td>
</tr>
<tr class="odd">
<td align="left">Vat1l</td>
<td align="right">0.178117</td>
<td align="right">0.0337314</td>
<td align="right">1.199619</td>
<td align="right">0.0812187</td>
<td align="right">0.1165772</td>
<td align="right">0.2339571</td>
<td align="right">0.4006268</td>
</tr>
<tr class="even">
<td align="left">Wtap</td>
<td align="right">2.691383</td>
<td align="right">2.4118074</td>
<td align="right">12.289111</td>
<td align="right">3.5809075</td>
<td align="right">3.2808114</td>
<td align="right">9.3443456</td>
<td align="right">8.6384533</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="get-all-genes">Get all genes<a class="anchor" aria-label="anchor" href="#get-all-genes"></a>
</h2>
<p>You can also quickly get all known genes from the genome of a given species with <a href="https://neurogenomics.github.io/orthogene/reference/all_genes.html" class="external-link"><code>all_genes</code></a>.</p>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genome_mouse</span> <span class="op">&lt;-</span> <span class="fu">orthogene</span><span class="fu">::</span><span class="fu"><a href="../reference/all_genes.html">all_genes</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"mouse"</span>, 
                                     method <span class="op">=</span> <span class="va">method</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Retrieving all genes using: homologene.</span></code></pre>
<pre><code><span class="co">## Retrieving all organisms available in gprofiler.</span></code></pre>
<pre><code><span class="co">## Using stored `gprofiler_orgs`.</span></code></pre>
<pre><code><span class="co">## Mapping species name: mouse</span></code></pre>
<pre><code><span class="co">## Common name mapping found for mouse</span></code></pre>
<pre><code><span class="co">## 1 organism identified from search: 10090</span></code></pre>
<pre><code><span class="co">## Gene table with 21,207 rows retrieved.</span></code></pre>
<pre><code><span class="co">## Returning all 21,207 genes from mouse.</span></code></pre>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genome_mouse</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">HID</th>
<th align="right">Gene.ID</th>
<th align="left">Gene.Symbol</th>
<th align="right">taxonomy_id</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">6</td>
<td align="right">3</td>
<td align="right">11364</td>
<td align="left">Acadm</td>
<td align="right">10090</td>
</tr>
<tr class="even">
<td align="left">18</td>
<td align="right">5</td>
<td align="right">11370</td>
<td align="left">Acadvl</td>
<td align="right">10090</td>
</tr>
<tr class="odd">
<td align="left">29</td>
<td align="right">6</td>
<td align="right">110446</td>
<td align="left">Acat1</td>
<td align="right">10090</td>
</tr>
<tr class="even">
<td align="left">52</td>
<td align="right">7</td>
<td align="right">11477</td>
<td align="left">Acvr1</td>
<td align="right">10090</td>
</tr>
<tr class="odd">
<td align="left">64</td>
<td align="right">9</td>
<td align="right">20391</td>
<td align="left">Sgca</td>
<td align="right">10090</td>
</tr>
<tr class="even">
<td align="left">71</td>
<td align="right">12</td>
<td align="right">11564</td>
<td align="left">Adsl</td>
<td align="right">10090</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level1">
<h1 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h1>
<details><div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.1 (2021-08-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] orthogene_0.99.9 BiocStyle_2.21.4</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] bitops_1.0-7              fs_1.5.0                 </span>
<span class="co">##  [3] httr_1.4.2                rprojroot_2.0.2          </span>
<span class="co">##  [5] tools_4.1.1               backports_1.2.1          </span>
<span class="co">##  [7] bslib_0.3.1               utf8_1.2.2               </span>
<span class="co">##  [9] R6_2.5.1                  lazyeval_0.2.2           </span>
<span class="co">## [11] colorspace_2.0-2          tidyselect_1.1.1         </span>
<span class="co">## [13] curl_4.3.2                compiler_4.1.1           </span>
<span class="co">## [15] textshaping_0.3.6         cli_3.0.1                </span>
<span class="co">## [17] desc_1.4.0                plotly_4.10.0            </span>
<span class="co">## [19] bookdown_0.24             sass_0.4.0               </span>
<span class="co">## [21] scales_1.1.1              pkgdown_1.9000.9000.9000 </span>
<span class="co">## [23] systemfonts_1.0.3         stringr_1.4.0            </span>
<span class="co">## [25] digest_0.6.28             foreign_0.8-81           </span>
<span class="co">## [27] rmarkdown_2.11            rio_0.5.27               </span>
<span class="co">## [29] pkgconfig_2.0.3           htmltools_0.5.2          </span>
<span class="co">## [31] fastmap_1.1.0             highr_0.9                </span>
<span class="co">## [33] htmlwidgets_1.5.4         rlang_0.99.0.9000        </span>
<span class="co">## [35] readxl_1.3.1              jquerylib_0.1.4          </span>
<span class="co">## [37] generics_0.1.1            jsonlite_1.7.2           </span>
<span class="co">## [39] dplyr_1.0.7               zip_2.2.0                </span>
<span class="co">## [41] car_3.0-11                RCurl_1.98-1.5           </span>
<span class="co">## [43] homologene_1.4.68.19.3.27 magrittr_2.0.1           </span>
<span class="co">## [45] GenomeInfoDbData_1.2.7    patchwork_1.1.1          </span>
<span class="co">## [47] Matrix_1.3-4              Rcpp_1.0.7               </span>
<span class="co">## [49] munsell_0.5.0             fansi_0.5.0              </span>
<span class="co">## [51] abind_1.4-5               babelgene_21.4           </span>
<span class="co">## [53] lifecycle_1.0.1           stringi_1.7.5            </span>
<span class="co">## [55] yaml_2.2.1                carData_3.0-4            </span>
<span class="co">## [57] grid_4.1.1                parallel_4.1.1           </span>
<span class="co">## [59] forcats_0.5.1             crayon_1.4.1             </span>
<span class="co">## [61] lattice_0.20-45           haven_2.4.3              </span>
<span class="co">## [63] hms_1.1.1                 knitr_1.36               </span>
<span class="co">## [65] pillar_1.6.4              ggpubr_0.4.0             </span>
<span class="co">## [67] ggsignif_0.6.3            glue_1.4.2               </span>
<span class="co">## [69] gprofiler2_0.2.1          evaluate_0.14            </span>
<span class="co">## [71] data.table_1.14.2         BiocManager_1.30.16      </span>
<span class="co">## [73] vctrs_0.3.8               cellranger_1.1.0         </span>
<span class="co">## [75] grr_0.9.5                 gtable_0.3.0             </span>
<span class="co">## [77] purrr_0.3.4               tidyr_1.1.4              </span>
<span class="co">## [79] cachem_1.0.6              ggplot2_3.3.5            </span>
<span class="co">## [81] xfun_0.27                 openxlsx_4.2.4           </span>
<span class="co">## [83] broom_0.7.9               rstatix_0.7.0            </span>
<span class="co">## [85] ragg_1.1.3                viridisLite_0.4.0        </span>
<span class="co">## [87] tibble_3.1.5              Matrix.utils_0.9.8       </span>
<span class="co">## [89] memoise_2.0.0             ellipsis_0.3.2</span></code></pre>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.9000.9000.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
